// aidp-typescript-client/dist/lib/model/adw-external-catalog-connection-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdwExternalCatalogConnectionPropertyKey = void 0;
/**
 * ADW external catalog connection property keys.
**/
var AdwExternalCatalogConnectionPropertyKey;
(function (AdwExternalCatalogConnectionPropertyKey) {
    AdwExternalCatalogConnectionPropertyKey["AdwWalletContentBase64"] = "ADW_WALLET_CONTENT_BASE64";
    AdwExternalCatalogConnectionPropertyKey["AdwWalletPassword"] = "ADW_WALLET_PASSWORD";
    AdwExternalCatalogConnectionPropertyKey["AdwUsername"] = "ADW_USERNAME";
    AdwExternalCatalogConnectionPropertyKey["AdwPassword"] = "ADW_PASSWORD";
    AdwExternalCatalogConnectionPropertyKey["AdwTnsAlias"] = "ADW_TNS_ALIAS";
    AdwExternalCatalogConnectionPropertyKey["AdwStagingTenancyOcid"] = "ADW_STAGING_TENANCY_OCID";
    AdwExternalCatalogConnectionPropertyKey["AdwStagingRegion"] = "ADW_STAGING_REGION";
    AdwExternalCatalogConnectionPropertyKey["AdwStagingNamespace"] = "ADW_STAGING_NAMESPACE";
    AdwExternalCatalogConnectionPropertyKey["AdwStagingBucket"] = "ADW_STAGING_BUCKET";
    AdwExternalCatalogConnectionPropertyKey["AdwStagingFilePrefix"] = "ADW_STAGING_FILE_PREFIX";
})(AdwExternalCatalogConnectionPropertyKey || (exports.AdwExternalCatalogConnectionPropertyKey = AdwExternalCatalogConnectionPropertyKey = {}));
(function (AdwExternalCatalogConnectionPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    AdwExternalCatalogConnectionPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AdwExternalCatalogConnectionPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(AdwExternalCatalogConnectionPropertyKey || (exports.AdwExternalCatalogConnectionPropertyKey = AdwExternalCatalogConnectionPropertyKey = {}));
