// aidp-typescript-client/dist/lib/model/agent-flow-privilege-for-manage.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentFlowPrivilegeForManage = void 0;
/**
 * The privilege for a Agent flow.
**/
var AgentFlowPrivilegeForManage;
(function (AgentFlowPrivilegeForManage) {
    AgentFlowPrivilegeForManage["Read"] = "READ";
    AgentFlowPrivilegeForManage["Manage"] = "MANAGE";
    AgentFlowPrivilegeForManage["Admin"] = "ADMIN";
    AgentFlowPrivilegeForManage["Use"] = "USE";
})(AgentFlowPrivilegeForManage || (exports.AgentFlowPrivilegeForManage = AgentFlowPrivilegeForManage = {}));
(function (AgentFlowPrivilegeForManage) {
    function getJsonObj(obj) {
        return obj;
    }
    AgentFlowPrivilegeForManage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AgentFlowPrivilegeForManage.getDeserializedJsonObj = getDeserializedJsonObj;
})(AgentFlowPrivilegeForManage || (exports.AgentFlowPrivilegeForManage = AgentFlowPrivilegeForManage = {}));
