// aidp-typescript-client/dist/lib/model/agent-flow-validation-issue.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentFlowValidationIssue = void 0;
var AgentFlowValidationIssue;
(function (AgentFlowValidationIssue) {
    let Severity;
    (function (Severity) {
        Severity["Error"] = "ERROR";
        Severity["Warning"] = "WARNING";
        Severity["Info"] = "INFO";
    })(Severity = AgentFlowValidationIssue.Severity || (AgentFlowValidationIssue.Severity = {}));
    let Phase;
    (function (Phase) {
        Phase["Model"] = "MODEL";
        Phase["Graph"] = "GRAPH";
        Phase["Config"] = "CONFIG";
        Phase["Reference"] = "REFERENCE";
        Phase["Codegen"] = "CODEGEN";
    })(Phase = AgentFlowValidationIssue.Phase || (AgentFlowValidationIssue.Phase = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AgentFlowValidationIssue.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AgentFlowValidationIssue.getDeserializedJsonObj = getDeserializedJsonObj;
})(AgentFlowValidationIssue || (exports.AgentFlowValidationIssue = AgentFlowValidationIssue = {}));
