// aidp-typescript-client/dist/lib/model/agent-flow-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentFlowPermissionSummary = void 0;
var AgentFlowPermissionSummary;
(function (AgentFlowPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["Admin"] = "ADMIN";
    })(GranteePermissions = AgentFlowPermissionSummary.GranteePermissions || (AgentFlowPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AgentFlowPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AgentFlowPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(AgentFlowPermissionSummary || (exports.AgentFlowPermissionSummary = AgentFlowPermissionSummary = {}));
