// aidp-typescript-client/dist/lib/model/agent-flow.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentFlow = void 0;
const model = __importStar(require("../model"));
var AgentFlow;
(function (AgentFlow) {
    let Type;
    (function (Type) {
        Type["Canvas"] = "CANVAS";
        Type["Code"] = "CODE";
    })(Type = AgentFlow.Type || (AgentFlow.Type = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Draft"] = "DRAFT";
        LifecycleState["Deployed"] = "DEPLOYED";
    })(LifecycleState = AgentFlow.LifecycleState || (AgentFlow.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'diagram': obj.diagram ?
                model.AgentFlowDiagram.getJsonObj(obj.diagram) : undefined,
            'guardrails': obj.guardrails ?
                model.GuardrailsConfiguration.getJsonObj(obj.guardrails) : undefined,
            'sessionConfig': obj.sessionConfig ?
                model.SessionConfiguration.getJsonObj(obj.sessionConfig) : undefined,
            'agentCardConfig': obj.agentCardConfig ?
                model.AgentCardConfigDetail.getJsonObj(obj.agentCardConfig) : undefined,
        });
        return jsonObj;
    }
    AgentFlow.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'diagram': obj.diagram ?
                model.AgentFlowDiagram.getDeserializedJsonObj(obj.diagram) : undefined,
            'guardrails': obj.guardrails ?
                model.GuardrailsConfiguration.getDeserializedJsonObj(obj.guardrails) : undefined,
            'sessionConfig': obj.sessionConfig ?
                model.SessionConfiguration.getDeserializedJsonObj(obj.sessionConfig) : undefined,
            'agentCardConfig': obj.agentCardConfig ?
                model.AgentCardConfigDetail.getDeserializedJsonObj(obj.agentCardConfig) : undefined,
        });
        return jsonObj;
    }
    AgentFlow.getDeserializedJsonObj = getDeserializedJsonObj;
})(AgentFlow || (exports.AgentFlow = AgentFlow = {}));
