// aidp-typescript-client/dist/lib/model/agent-flow-validation-metadata.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentFlowValidationMetadata = void 0;
var AgentFlowValidationMetadata;
(function (AgentFlowValidationMetadata) {
    let SkippedPhases;
    (function (SkippedPhases) {
        SkippedPhases["Model"] = "MODEL";
        SkippedPhases["Graph"] = "GRAPH";
        SkippedPhases["Config"] = "CONFIG";
        SkippedPhases["Reference"] = "REFERENCE";
        SkippedPhases["Codegen"] = "CODEGEN";
    })(SkippedPhases = AgentFlowValidationMetadata.SkippedPhases || (AgentFlowValidationMetadata.SkippedPhases = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AgentFlowValidationMetadata.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AgentFlowValidationMetadata.getDeserializedJsonObj = getDeserializedJsonObj;
})(AgentFlowValidationMetadata || (exports.AgentFlowValidationMetadata = AgentFlowValidationMetadata = {}));
