// aidp-typescript-client/dist/lib/model/alh-external-catalog-connection-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlhExternalCatalogConnectionPropertyKey = void 0;
/**
 * ALH external catalog connection property keys.
**/
var AlhExternalCatalogConnectionPropertyKey;
(function (AlhExternalCatalogConnectionPropertyKey) {
    AlhExternalCatalogConnectionPropertyKey["AlhWalletContentBase64"] = "ALH_WALLET_CONTENT_BASE64";
    AlhExternalCatalogConnectionPropertyKey["AlhWalletPassword"] = "ALH_WALLET_PASSWORD";
    AlhExternalCatalogConnectionPropertyKey["AlhUsername"] = "ALH_USERNAME";
    AlhExternalCatalogConnectionPropertyKey["AlhPassword"] = "ALH_PASSWORD";
    AlhExternalCatalogConnectionPropertyKey["AlhTnsAlias"] = "ALH_TNS_ALIAS";
    AlhExternalCatalogConnectionPropertyKey["AlhStagingTenancyOcid"] = "ALH_STAGING_TENANCY_OCID";
    AlhExternalCatalogConnectionPropertyKey["AlhStagingRegion"] = "ALH_STAGING_REGION";
    AlhExternalCatalogConnectionPropertyKey["AlhStagingNamespace"] = "ALH_STAGING_NAMESPACE";
    AlhExternalCatalogConnectionPropertyKey["AlhStagingBucket"] = "ALH_STAGING_BUCKET";
    AlhExternalCatalogConnectionPropertyKey["AlhStagingFilePrefix"] = "ALH_STAGING_FILE_PREFIX";
})(AlhExternalCatalogConnectionPropertyKey || (exports.AlhExternalCatalogConnectionPropertyKey = AlhExternalCatalogConnectionPropertyKey = {}));
(function (AlhExternalCatalogConnectionPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    AlhExternalCatalogConnectionPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AlhExternalCatalogConnectionPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlhExternalCatalogConnectionPropertyKey || (exports.AlhExternalCatalogConnectionPropertyKey = AlhExternalCatalogConnectionPropertyKey = {}));
