// aidp-typescript-client/dist/lib/model/all-knowledge-base-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllKnowledgeBasePrivilege = void 0;
/**
 * The privilege for a Knowledge Base which could be an inherited privilege coming from object up in hierarchy.
**/
var AllKnowledgeBasePrivilege;
(function (AllKnowledgeBasePrivilege) {
    AllKnowledgeBasePrivilege["Write"] = "WRITE";
    AllKnowledgeBasePrivilege["Admin"] = "ADMIN";
    AllKnowledgeBasePrivilege["Select"] = "SELECT";
    AllKnowledgeBasePrivilege["Manage"] = "MANAGE";
})(AllKnowledgeBasePrivilege || (exports.AllKnowledgeBasePrivilege = AllKnowledgeBasePrivilege = {}));
(function (AllKnowledgeBasePrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    AllKnowledgeBasePrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AllKnowledgeBasePrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(AllKnowledgeBasePrivilege || (exports.AllKnowledgeBasePrivilege = AllKnowledgeBasePrivilege = {}));
