// aidp-typescript-client/dist/lib/model/all-privilege-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllPrivilegeType = void 0;
/**
 * List of all privileges in the AI Data Platform Workbench.
**/
var AllPrivilegeType;
(function (AllPrivilegeType) {
    AllPrivilegeType["User"] = "USER";
    AllPrivilegeType["Administrator"] = "ADMINISTRATOR";
    AllPrivilegeType["Read"] = "READ";
    AllPrivilegeType["Use"] = "USE";
    AllPrivilegeType["Manage"] = "MANAGE";
    AllPrivilegeType["Admin"] = "ADMIN";
    AllPrivilegeType["CreateCatalog"] = "CREATE_CATALOG";
    AllPrivilegeType["Select"] = "SELECT";
    AllPrivilegeType["Insert"] = "INSERT";
    AllPrivilegeType["Update"] = "UPDATE";
    AllPrivilegeType["Delete"] = "DELETE";
    AllPrivilegeType["Alter"] = "ALTER";
    AllPrivilegeType["CreateSchema"] = "CREATE_SCHEMA";
    AllPrivilegeType["Write"] = "WRITE";
    AllPrivilegeType["CreateView"] = "CREATE_VIEW";
    AllPrivilegeType["CreateVolume"] = "CREATE_VOLUME";
    AllPrivilegeType["CreateTable"] = "CREATE_TABLE";
    AllPrivilegeType["CreateKnowledgeBase"] = "CREATE_KNOWLEDGE_BASE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AllPrivilegeType["UnknownValue"] = "UNKNOWN_VALUE";
})(AllPrivilegeType || (exports.AllPrivilegeType = AllPrivilegeType = {}));
(function (AllPrivilegeType) {
    function getJsonObj(obj) {
        return obj;
    }
    AllPrivilegeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AllPrivilegeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AllPrivilegeType || (exports.AllPrivilegeType = AllPrivilegeType = {}));
