// aidp-typescript-client/dist/lib/model/all-resource-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllResourceType = void 0;
/**
 * List of sub-resources that are RBAC managed by AI Data Platform Workbench.
**/
var AllResourceType;
(function (AllResourceType) {
    AllResourceType["Workspace"] = "WORKSPACE";
    AllResourceType["Workflow"] = "WORKFLOW";
    AllResourceType["Folder"] = "FOLDER";
    AllResourceType["File"] = "FILE";
    AllResourceType["Cluster"] = "CLUSTER";
    AllResourceType["MasterCatalog"] = "MASTER_CATALOG";
    AllResourceType["Catalog"] = "CATALOG";
    AllResourceType["Schema"] = "SCHEMA";
    AllResourceType["Table"] = "TABLE";
    AllResourceType["Share"] = "SHARE";
    AllResourceType["Recipient"] = "RECIPIENT";
    AllResourceType["Volume"] = "VOLUME";
    AllResourceType["View"] = "VIEW";
    AllResourceType["KnowledgeBase"] = "KNOWLEDGE_BASE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AllResourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(AllResourceType || (exports.AllResourceType = AllResourceType = {}));
(function (AllResourceType) {
    function getJsonObj(obj) {
        return obj;
    }
    AllResourceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AllResourceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AllResourceType || (exports.AllResourceType = AllResourceType = {}));
