// aidp-typescript-client/dist/lib/model/all-volume-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllVolumePrivilege = void 0;
/**
 * The privilege for a volume. Can be an inherited privilege coming from object up in hierarchy.
**/
var AllVolumePrivilege;
(function (AllVolumePrivilege) {
    AllVolumePrivilege["Read"] = "READ";
    AllVolumePrivilege["Write"] = "WRITE";
    AllVolumePrivilege["Admin"] = "ADMIN";
    AllVolumePrivilege["Select"] = "SELECT";
    AllVolumePrivilege["Manage"] = "MANAGE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AllVolumePrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(AllVolumePrivilege || (exports.AllVolumePrivilege = AllVolumePrivilege = {}));
(function (AllVolumePrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    AllVolumePrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AllVolumePrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(AllVolumePrivilege || (exports.AllVolumePrivilege = AllVolumePrivilege = {}));
