// aidp-typescript-client/dist/lib/model/async-operation-action-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AsyncOperationActionType = void 0;
/**
 * The action type of an async operation.
**/
var AsyncOperationActionType;
(function (AsyncOperationActionType) {
    AsyncOperationActionType["Unknown"] = "UNKNOWN";
    AsyncOperationActionType["CreateCatalog"] = "CREATE_CATALOG";
    AsyncOperationActionType["UpdateCatalog"] = "UPDATE_CATALOG";
    AsyncOperationActionType["DeleteCatalog"] = "DELETE_CATALOG";
    AsyncOperationActionType["TestConnectionCatalog"] = "TEST_CONNECTION_CATALOG";
    AsyncOperationActionType["CreateUserSchemaInCatalog"] = "CREATE_USER_SCHEMA_IN_CATALOG";
    AsyncOperationActionType["CreateSchema"] = "CREATE_SCHEMA";
    AsyncOperationActionType["DeleteSchema"] = "DELETE_SCHEMA";
    AsyncOperationActionType["CreateTable"] = "CREATE_TABLE";
    AsyncOperationActionType["UpdateTable"] = "UPDATE_TABLE";
    AsyncOperationActionType["DeleteTable"] = "DELETE_TABLE";
    AsyncOperationActionType["CreateVolume"] = "CREATE_VOLUME";
    AsyncOperationActionType["DeleteVolume"] = "DELETE_VOLUME";
    AsyncOperationActionType["CopyVolumeFile"] = "COPY_VOLUME_FILE";
    AsyncOperationActionType["MoveVolumeFile"] = "MOVE_VOLUME_FILE";
    AsyncOperationActionType["DeleteVolumeFile"] = "DELETE_VOLUME_FILE";
    AsyncOperationActionType["DeleteVolumeFolder"] = "DELETE_VOLUME_FOLDER";
    AsyncOperationActionType["CreateWorkspace"] = "CREATE_WORKSPACE";
    AsyncOperationActionType["UpdateWorkspace"] = "UPDATE_WORKSPACE";
    AsyncOperationActionType["DeleteWorkspace"] = "DELETE_WORKSPACE";
    AsyncOperationActionType["CreateCluster"] = "CREATE_CLUSTER";
    AsyncOperationActionType["UpdateCluster"] = "UPDATE_CLUSTER";
    AsyncOperationActionType["StartCluster"] = "START_CLUSTER";
    AsyncOperationActionType["StopCluster"] = "STOP_CLUSTER";
    AsyncOperationActionType["RestartCluster"] = "RESTART_CLUSTER";
    AsyncOperationActionType["PatchClusterLibraries"] = "PATCH_CLUSTER_LIBRARIES";
    AsyncOperationActionType["DeleteCluster"] = "DELETE_CLUSTER";
    AsyncOperationActionType["ManageExtractedEntities"] = "MANAGE_EXTRACTED_ENTITIES";
    AsyncOperationActionType["RefreshEntity"] = "REFRESH_ENTITY";
    AsyncOperationActionType["DownloadClusterLog"] = "DOWNLOAD_CLUSTER_LOG";
    AsyncOperationActionType["MigrateExternalCatalog"] = "MIGRATE_EXTERNAL_CATALOG";
    AsyncOperationActionType["UpdateKnowledgeBase"] = "UPDATE_KNOWLEDGE_BASE";
    AsyncOperationActionType["DeleteKnowledgeBase"] = "DELETE_KNOWLEDGE_BASE";
    AsyncOperationActionType["CreateKnowledgeBase"] = "CREATE_KNOWLEDGE_BASE";
    AsyncOperationActionType["DeleteKnowledgeBaseJob"] = "DELETE_KNOWLEDGE_BASE_JOB";
    AsyncOperationActionType["CreateKnowledgeBaseJob"] = "CREATE_KNOWLEDGE_BASE_JOB";
    AsyncOperationActionType["CreateKnowledgeBaseJobRun"] = "CREATE_KNOWLEDGE_BASE_JOB_RUN";
    AsyncOperationActionType["DeployAgentFlow"] = "DEPLOY_AGENT_FLOW";
    AsyncOperationActionType["CreateGitFolder"] = "CREATE_GIT_FOLDER";
    AsyncOperationActionType["UpdateGitRepository"] = "UPDATE_GIT_REPOSITORY";
    AsyncOperationActionType["CreateBundle"] = "CREATE_BUNDLE";
    AsyncOperationActionType["BundleDeploy"] = "BUNDLE_DEPLOY";
    AsyncOperationActionType["BundlePurge"] = "BUNDLE_PURGE";
    AsyncOperationActionType["BundleSync"] = "BUNDLE_SYNC";
    AsyncOperationActionType["MarkAsBundle"] = "MARK_AS_BUNDLE";
    AsyncOperationActionType["GitCommitPush"] = "GIT_COMMIT_PUSH";
    AsyncOperationActionType["GitCreateBranch"] = "GIT_CREATE_BRANCH";
    AsyncOperationActionType["GitCheckoutBranch"] = "GIT_CHECKOUT_BRANCH";
    AsyncOperationActionType["GitOperationPull"] = "GIT_OPERATION_PULL";
    AsyncOperationActionType["GitOperationMerge"] = "GIT_OPERATION_MERGE";
    AsyncOperationActionType["GitOperationRebase"] = "GIT_OPERATION_REBASE";
    AsyncOperationActionType["GitOperationReset"] = "GIT_OPERATION_RESET";
    AsyncOperationActionType["GitOperationResetState"] = "GIT_OPERATION_RESET_STATE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AsyncOperationActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(AsyncOperationActionType || (exports.AsyncOperationActionType = AsyncOperationActionType = {}));
(function (AsyncOperationActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    AsyncOperationActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AsyncOperationActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AsyncOperationActionType || (exports.AsyncOperationActionType = AsyncOperationActionType = {}));
