// aidp-typescript-client/dist/lib/model/async-operation-resource-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AsyncOperationResourceType = void 0;
/**
 * The resource type of an async operation.
**/
var AsyncOperationResourceType;
(function (AsyncOperationResourceType) {
    AsyncOperationResourceType["Unknown"] = "UNKNOWN";
    AsyncOperationResourceType["Catalog"] = "CATALOG";
    AsyncOperationResourceType["Schema"] = "SCHEMA";
    AsyncOperationResourceType["Table"] = "TABLE";
    AsyncOperationResourceType["View"] = "VIEW";
    AsyncOperationResourceType["Volume"] = "VOLUME";
    AsyncOperationResourceType["VolumeFile"] = "VOLUME_FILE";
    AsyncOperationResourceType["Workspace"] = "WORKSPACE";
    AsyncOperationResourceType["WorkspaceObject"] = "WORKSPACE_OBJECT";
    AsyncOperationResourceType["Cluster"] = "CLUSTER";
    AsyncOperationResourceType["AiCompute"] = "AI_COMPUTE";
    AsyncOperationResourceType["KnowledgeBase"] = "KNOWLEDGE_BASE";
    AsyncOperationResourceType["KnowledgeBaseJob"] = "KNOWLEDGE_BASE_JOB";
    AsyncOperationResourceType["KnowledgeBaseJobRun"] = "KNOWLEDGE_BASE_JOB_RUN";
    AsyncOperationResourceType["AgentFlow"] = "AGENT_FLOW";
    AsyncOperationResourceType["GitOperation"] = "GIT_OPERATION";
    AsyncOperationResourceType["BundleOperation"] = "BUNDLE_OPERATION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AsyncOperationResourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(AsyncOperationResourceType || (exports.AsyncOperationResourceType = AsyncOperationResourceType = {}));
(function (AsyncOperationResourceType) {
    function getJsonObj(obj) {
        return obj;
    }
    AsyncOperationResourceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AsyncOperationResourceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AsyncOperationResourceType || (exports.AsyncOperationResourceType = AsyncOperationResourceType = {}));
