// aidp-typescript-client/dist/lib/model/async-operation-status.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AsyncOperationStatus = void 0;
/**
 * The status of an async operation.
**/
var AsyncOperationStatus;
(function (AsyncOperationStatus) {
    AsyncOperationStatus["InProgress"] = "IN_PROGRESS";
    AsyncOperationStatus["Succeeded"] = "SUCCEEDED";
    AsyncOperationStatus["Failed"] = "FAILED";
    AsyncOperationStatus["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AsyncOperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(AsyncOperationStatus || (exports.AsyncOperationStatus = AsyncOperationStatus = {}));
(function (AsyncOperationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    AsyncOperationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AsyncOperationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(AsyncOperationStatus || (exports.AsyncOperationStatus = AsyncOperationStatus = {}));
