// aidp-typescript-client/dist/lib/model/atp-external-catalog-connection-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AtpExternalCatalogConnectionPropertyKey = void 0;
/**
 * ATP external catalog connection property keys.
**/
var AtpExternalCatalogConnectionPropertyKey;
(function (AtpExternalCatalogConnectionPropertyKey) {
    AtpExternalCatalogConnectionPropertyKey["AtpWalletContentBase64"] = "ATP_WALLET_CONTENT_BASE64";
    AtpExternalCatalogConnectionPropertyKey["AtpWalletPassword"] = "ATP_WALLET_PASSWORD";
    AtpExternalCatalogConnectionPropertyKey["AtpUsername"] = "ATP_USERNAME";
    AtpExternalCatalogConnectionPropertyKey["AtpPassword"] = "ATP_PASSWORD";
    AtpExternalCatalogConnectionPropertyKey["AtpTnsAlias"] = "ATP_TNS_ALIAS";
    AtpExternalCatalogConnectionPropertyKey["AtpStagingTenancyOcid"] = "ATP_STAGING_TENANCY_OCID";
    AtpExternalCatalogConnectionPropertyKey["AtpStagingRegion"] = "ATP_STAGING_REGION";
    AtpExternalCatalogConnectionPropertyKey["AtpStagingNamespace"] = "ATP_STAGING_NAMESPACE";
    AtpExternalCatalogConnectionPropertyKey["AtpStagingBucket"] = "ATP_STAGING_BUCKET";
    AtpExternalCatalogConnectionPropertyKey["AtpStagingFilePrefix"] = "ATP_STAGING_FILE_PREFIX";
})(AtpExternalCatalogConnectionPropertyKey || (exports.AtpExternalCatalogConnectionPropertyKey = AtpExternalCatalogConnectionPropertyKey = {}));
(function (AtpExternalCatalogConnectionPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    AtpExternalCatalogConnectionPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AtpExternalCatalogConnectionPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(AtpExternalCatalogConnectionPropertyKey || (exports.AtpExternalCatalogConnectionPropertyKey = AtpExternalCatalogConnectionPropertyKey = {}));
