// aidp-typescript-client/dist/lib/model/attached-session.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachedSession = void 0;
var AttachedSession;
(function (AttachedSession) {
    let Type;
    (function (Type) {
        Type["Notebook"] = "NOTEBOOK";
        Type["File"] = "FILE";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = AttachedSession.Type || (AttachedSession.Type = {}));
    let State;
    (function (State) {
        State["Active"] = "ACTIVE";
        State["Idle"] = "IDLE";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        State["UnknownValue"] = "UNKNOWN_VALUE";
    })(State = AttachedSession.State || (AttachedSession.State = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttachedSession.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttachedSession.getDeserializedJsonObj = getDeserializedJsonObj;
})(AttachedSession || (exports.AttachedSession = AttachedSession = {}));
