// aidp-typescript-client/dist/lib/model/bundle-deployed-resource.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.BundleDeployedResource = void 0;
var BundleDeployedResource;
(function (BundleDeployedResource) {
    let Type;
    (function (Type) {
        Type["Job"] = "JOB";
        Type["Agentflow"] = "AGENTFLOW";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = BundleDeployedResource.Type || (BundleDeployedResource.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BundleDeployedResource.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BundleDeployedResource.getDeserializedJsonObj = getDeserializedJsonObj;
})(BundleDeployedResource || (exports.BundleDeployedResource = BundleDeployedResource = {}));
