// aidp-typescript-client/dist/lib/model/catalog-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogLifecycleState = void 0;
/**
 * The current status of the catalog.
**/
var CatalogLifecycleState;
(function (CatalogLifecycleState) {
    CatalogLifecycleState["Active"] = "ACTIVE";
    CatalogLifecycleState["Creating"] = "CREATING";
    CatalogLifecycleState["Deleting"] = "DELETING";
    CatalogLifecycleState["Updating"] = "UPDATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CatalogLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(CatalogLifecycleState || (exports.CatalogLifecycleState = CatalogLifecycleState = {}));
(function (CatalogLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    CatalogLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CatalogLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(CatalogLifecycleState || (exports.CatalogLifecycleState = CatalogLifecycleState = {}));
