// aidp-typescript-client/dist/lib/model/catalog-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogPermissionSummary = void 0;
var CatalogPermissionSummary;
(function (CatalogPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Select"] = "SELECT";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["CreateSchema"] = "CREATE_SCHEMA";
        GranteePermissions["Admin"] = "ADMIN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = CatalogPermissionSummary.GranteePermissions || (CatalogPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CatalogPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CatalogPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(CatalogPermissionSummary || (exports.CatalogPermissionSummary = CatalogPermissionSummary = {}));
