// aidp-typescript-client/dist/lib/model/catalog-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogPrivilege = void 0;
/**
 * Catalog privilege
**/
var CatalogPrivilege;
(function (CatalogPrivilege) {
    CatalogPrivilege["Select"] = "SELECT";
    CatalogPrivilege["Manage"] = "MANAGE";
    CatalogPrivilege["CreateSchema"] = "CREATE_SCHEMA";
    CatalogPrivilege["Admin"] = "ADMIN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CatalogPrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(CatalogPrivilege || (exports.CatalogPrivilege = CatalogPrivilege = {}));
(function (CatalogPrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    CatalogPrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CatalogPrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(CatalogPrivilege || (exports.CatalogPrivilege = CatalogPrivilege = {}));
