// aidp-typescript-client/dist/lib/model/cluster-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterPrivilege = void 0;
/**
 * Privilege for a cluster.
**/
var ClusterPrivilege;
(function (ClusterPrivilege) {
    ClusterPrivilege["Read"] = "READ";
    ClusterPrivilege["Use"] = "USE";
    ClusterPrivilege["Admin"] = "ADMIN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ClusterPrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(ClusterPrivilege || (exports.ClusterPrivilege = ClusterPrivilege = {}));
(function (ClusterPrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    ClusterPrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ClusterPrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(ClusterPrivilege || (exports.ClusterPrivilege = ClusterPrivilege = {}));
