// aidp-typescript-client/dist/lib/model/cluster-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterPermissionSummary = void 0;
var ClusterPermissionSummary;
(function (ClusterPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Use"] = "USE";
        GranteePermissions["Admin"] = "ADMIN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = ClusterPermissionSummary.GranteePermissions || (ClusterPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ClusterPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ClusterPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ClusterPermissionSummary || (exports.ClusterPermissionSummary = ClusterPermissionSummary = {}));
