// aidp-typescript-client/dist/lib/model/cluster-library-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterLibrarySummary = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var ClusterLibrarySummary;
(function (ClusterLibrarySummary) {
    let Status;
    (function (Status) {
        Status["Pending"] = "PENDING";
        Status["Resolving"] = "RESOLVING";
        Status["Installing"] = "INSTALLING";
        Status["Installed"] = "INSTALLED";
        Status["Failed"] = "FAILED";
        Status["InstallOnRestart"] = "INSTALL_ON_RESTART";
        Status["UninstallOnRestart"] = "UNINSTALL_ON_RESTART";
        Status["Skipped"] = "SKIPPED";
        Status["Deleted"] = "DELETED";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = ClusterLibrarySummary.Status || (ClusterLibrarySummary.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "WORKSPACE_FILE":
                    return model.WorkspaceFileClusterLibrarySummary.getJsonObj(jsonObj, true);
                case "VOLUME_FILE":
                    return model.VolumeFileClusterLibrarySummary.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    ClusterLibrarySummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "WORKSPACE_FILE":
                    return model.WorkspaceFileClusterLibrarySummary.getDeserializedJsonObj(jsonObj, true);
                case "VOLUME_FILE":
                    return model.VolumeFileClusterLibrarySummary.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    ClusterLibrarySummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ClusterLibrarySummary || (exports.ClusterLibrarySummary = ClusterLibrarySummary = {}));
