// aidp-typescript-client/dist/lib/model/conflict-resolve-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConflictResolveDetails = void 0;
var ConflictResolveDetails;
(function (ConflictResolveDetails) {
    let Action;
    (function (Action) {
        Action["Local"] = "LOCAL";
        Action["Remote"] = "REMOTE";
        Action["MarkResolved"] = "MARK_RESOLVED";
    })(Action = ConflictResolveDetails.Action || (ConflictResolveDetails.Action = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ConflictResolveDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ConflictResolveDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConflictResolveDetails || (exports.ConflictResolveDetails = ConflictResolveDetails = {}));
