// aidp-typescript-client/dist/lib/model/continuous.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Continuous = void 0;
var Continuous;
(function (Continuous) {
    let PauseStatus;
    (function (PauseStatus) {
        PauseStatus["Paused"] = "PAUSED";
        PauseStatus["Unpaused"] = "UNPAUSED";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        PauseStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(PauseStatus = Continuous.PauseStatus || (Continuous.PauseStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Continuous.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Continuous.getDeserializedJsonObj = getDeserializedJsonObj;
})(Continuous || (exports.Continuous = Continuous = {}));
