// aidp-typescript-client/dist/lib/model/content.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Content = void 0;
var Content;
(function (Content) {
    let Type;
    (function (Type) {
        Type["Notebook"] = "notebook";
        Type["File"] = "file";
        Type["Directory"] = "directory";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = Content.Type || (Content.Type = {}));
    let Format;
    (function (Format) {
        Format["Null"] = "NULL";
        Format["Json"] = "json";
        Format["Text"] = "text";
        Format["Base64"] = "base64";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Format["UnknownValue"] = "UNKNOWN_VALUE";
    })(Format = Content.Format || (Content.Format = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'last_modified': obj.lastModified,
            'hash_algorithm': obj.hashAlgorithm,
        });
        delete jsonObj.lastModified;
        delete jsonObj.hashAlgorithm;
        return jsonObj;
    }
    Content.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'lastModified': obj["last_modified"],
            'hashAlgorithm': obj["hash_algorithm"],
        });
        delete jsonObj["last_modified"];
        delete jsonObj["hash_algorithm"];
        return jsonObj;
    }
    Content.getDeserializedJsonObj = getDeserializedJsonObj;
})(Content || (exports.Content = Content = {}));
