// aidp-typescript-client/dist/lib/model/content-moderation-category.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentModerationCategory = void 0;
/**
 * Content moderation categories
**/
var ContentModerationCategory;
(function (ContentModerationCategory) {
    ContentModerationCategory["HateSpeech"] = "HATE_SPEECH";
    ContentModerationCategory["Harassment"] = "HARASSMENT";
    ContentModerationCategory["Violence"] = "VIOLENCE";
    ContentModerationCategory["Sexual"] = "SEXUAL";
    ContentModerationCategory["Derogatory"] = "DEROGATORY";
    ContentModerationCategory["Toxic"] = "TOXIC";
})(ContentModerationCategory || (exports.ContentModerationCategory = ContentModerationCategory = {}));
(function (ContentModerationCategory) {
    function getJsonObj(obj) {
        return obj;
    }
    ContentModerationCategory.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ContentModerationCategory.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContentModerationCategory || (exports.ContentModerationCategory = ContentModerationCategory = {}));
