// aidp-typescript-client/dist/lib/model/crawler-last-refresh-status.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CrawlerLastRefreshStatus = void 0;
/**
 * The status for an entity refresh.
**/
var CrawlerLastRefreshStatus;
(function (CrawlerLastRefreshStatus) {
    CrawlerLastRefreshStatus["Failed"] = "FAILED";
    CrawlerLastRefreshStatus["Success"] = "SUCCESS";
    CrawlerLastRefreshStatus["InProgress"] = "IN_PROGRESS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CrawlerLastRefreshStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(CrawlerLastRefreshStatus || (exports.CrawlerLastRefreshStatus = CrawlerLastRefreshStatus = {}));
(function (CrawlerLastRefreshStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    CrawlerLastRefreshStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CrawlerLastRefreshStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(CrawlerLastRefreshStatus || (exports.CrawlerLastRefreshStatus = CrawlerLastRefreshStatus = {}));
