// aidp-typescript-client/dist/lib/model/credential-v2-life-cycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CredentialV2LifeCycleState = void 0;
/**
 * The current lifecycle state of the credential object.
**/
var CredentialV2LifeCycleState;
(function (CredentialV2LifeCycleState) {
    CredentialV2LifeCycleState["Active"] = "ACTIVE";
    CredentialV2LifeCycleState["Creating"] = "CREATING";
    CredentialV2LifeCycleState["Updating"] = "UPDATING";
    CredentialV2LifeCycleState["Deleting"] = "DELETING";
})(CredentialV2LifeCycleState || (exports.CredentialV2LifeCycleState = CredentialV2LifeCycleState = {}));
(function (CredentialV2LifeCycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    CredentialV2LifeCycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CredentialV2LifeCycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(CredentialV2LifeCycleState || (exports.CredentialV2LifeCycleState = CredentialV2LifeCycleState = {}));
