// aidp-typescript-client/dist/lib/model/credential-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CredentialLifecycleState = void 0;
/**
 * Lifecycle state values exposed via the public credential API.
**/
var CredentialLifecycleState;
(function (CredentialLifecycleState) {
    CredentialLifecycleState["Active"] = "ACTIVE";
    CredentialLifecycleState["Creating"] = "CREATING";
    CredentialLifecycleState["Updating"] = "UPDATING";
    CredentialLifecycleState["Deleting"] = "DELETING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CredentialLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(CredentialLifecycleState || (exports.CredentialLifecycleState = CredentialLifecycleState = {}));
(function (CredentialLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    CredentialLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CredentialLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(CredentialLifecycleState || (exports.CredentialLifecycleState = CredentialLifecycleState = {}));
