// aidp-typescript-client/dist/lib/model/credential-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CredentialType = void 0;
/**
 * The type of credential supported by the credential store.
**/
var CredentialType;
(function (CredentialType) {
    CredentialType["SecretToken"] = "SECRET_TOKEN";
    CredentialType["VaultReference"] = "VAULT_REFERENCE";
    CredentialType["ServiceAccount"] = "SERVICE_ACCOUNT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CredentialType["UnknownValue"] = "UNKNOWN_VALUE";
})(CredentialType || (exports.CredentialType = CredentialType = {}));
(function (CredentialType) {
    function getJsonObj(obj) {
        return obj;
    }
    CredentialType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CredentialType.getDeserializedJsonObj = getDeserializedJsonObj;
})(CredentialType || (exports.CredentialType = CredentialType = {}));
