// aidp-typescript-client/dist/lib/model/data-format.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataFormat = void 0;
/**
 * The table data format type.
**/
var DataFormat;
(function (DataFormat) {
    DataFormat["Avro"] = "AVRO";
    DataFormat["Orc"] = "ORC";
    DataFormat["Parquet"] = "PARQUET";
    DataFormat["Textfile"] = "TEXTFILE";
    DataFormat["Json"] = "JSON";
    DataFormat["Csv"] = "CSV";
    DataFormat["Delta"] = "DELTA";
    DataFormat["Iceberg"] = "ICEBERG";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DataFormat["UnknownValue"] = "UNKNOWN_VALUE";
})(DataFormat || (exports.DataFormat = DataFormat = {}));
(function (DataFormat) {
    function getJsonObj(obj) {
        return obj;
    }
    DataFormat.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DataFormat.getDeserializedJsonObj = getDeserializedJsonObj;
})(DataFormat || (exports.DataFormat = DataFormat = {}));
