// aidp-typescript-client/dist/lib/model/deployment-job-run-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentJobRunState = void 0;
/**
 * Status of the run.
**/
var DeploymentJobRunState;
(function (DeploymentJobRunState) {
    DeploymentJobRunState["DeploymentJobRunStateUnspecified"] = "DEPLOYMENT_JOB_RUN_STATE_UNSPECIFIED";
    DeploymentJobRunState["NoValidDeploymentJobFound"] = "NO_VALID_DEPLOYMENT_JOB_FOUND";
    DeploymentJobRunState["Running"] = "RUNNING";
    DeploymentJobRunState["Succeeded"] = "SUCCEEDED";
    DeploymentJobRunState["Failed"] = "FAILED";
    DeploymentJobRunState["Pending"] = "PENDING";
    DeploymentJobRunState["Approval"] = "APPROVAL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DeploymentJobRunState["UnknownValue"] = "UNKNOWN_VALUE";
})(DeploymentJobRunState || (exports.DeploymentJobRunState = DeploymentJobRunState = {}));
(function (DeploymentJobRunState) {
    function getJsonObj(obj) {
        return obj;
    }
    DeploymentJobRunState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeploymentJobRunState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentJobRunState || (exports.DeploymentJobRunState = DeploymentJobRunState = {}));
