// aidp-typescript-client/dist/lib/model/deployment-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentLifecycleState = void 0;
/**
 * LifecycleState of an Agent Flow Session or Deployment.
**/
var DeploymentLifecycleState;
(function (DeploymentLifecycleState) {
    DeploymentLifecycleState["Creating"] = "CREATING";
    DeploymentLifecycleState["Active"] = "ACTIVE";
    DeploymentLifecycleState["Inactive"] = "INACTIVE";
    DeploymentLifecycleState["Failed"] = "FAILED";
    DeploymentLifecycleState["Deleted"] = "DELETED";
})(DeploymentLifecycleState || (exports.DeploymentLifecycleState = DeploymentLifecycleState = {}));
(function (DeploymentLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    DeploymentLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeploymentLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentLifecycleState || (exports.DeploymentLifecycleState = DeploymentLifecycleState = {}));
