// aidp-typescript-client/dist/lib/model/deployment-job-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentJobState = void 0;
/**
 * Status of the job.
**/
var DeploymentJobState;
(function (DeploymentJobState) {
    DeploymentJobState["DeploymentJobConnectionStateUnspecified"] = "DEPLOYMENT_JOB_CONNECTION_STATE_UNSPECIFIED";
    DeploymentJobState["NotSetUp"] = "NOT_SET_UP";
    DeploymentJobState["Connected"] = "CONNECTED";
    DeploymentJobState["NotFound"] = "NOT_FOUND";
    DeploymentJobState["RequiredParametersChanged"] = "REQUIRED_PARAMETERS_CHANGED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DeploymentJobState["UnknownValue"] = "UNKNOWN_VALUE";
})(DeploymentJobState || (exports.DeploymentJobState = DeploymentJobState = {}));
(function (DeploymentJobState) {
    function getJsonObj(obj) {
        return obj;
    }
    DeploymentJobState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeploymentJobState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentJobState || (exports.DeploymentJobState = DeploymentJobState = {}));
