// aidp-typescript-client/dist/lib/model/exadata-external-catalog-connection-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExadataExternalCatalogConnectionPropertyKey = void 0;
/**
 * Exadata external catalog connection property keys.
**/
var ExadataExternalCatalogConnectionPropertyKey;
(function (ExadataExternalCatalogConnectionPropertyKey) {
    ExadataExternalCatalogConnectionPropertyKey["ExadataWalletContentBase64"] = "EXADATA_WALLET_CONTENT_BASE64";
    ExadataExternalCatalogConnectionPropertyKey["ExadataWalletPassword"] = "EXADATA_WALLET_PASSWORD";
    ExadataExternalCatalogConnectionPropertyKey["ExadataUsername"] = "EXADATA_USERNAME";
    ExadataExternalCatalogConnectionPropertyKey["ExadataPassword"] = "EXADATA_PASSWORD";
    ExadataExternalCatalogConnectionPropertyKey["ExadataHost"] = "EXADATA_HOST";
    ExadataExternalCatalogConnectionPropertyKey["ExadataPort"] = "EXADATA_PORT";
    ExadataExternalCatalogConnectionPropertyKey["ExadataDatabaseName"] = "EXADATA_DATABASE_NAME";
    ExadataExternalCatalogConnectionPropertyKey["ExadataDatabaseSid"] = "EXADATA_DATABASE_SID";
    ExadataExternalCatalogConnectionPropertyKey["ExadataSslEnabled"] = "EXADATA_SSL_ENABLED";
    ExadataExternalCatalogConnectionPropertyKey["ExadataStagingTenancyOcid"] = "EXADATA_STAGING_TENANCY_OCID";
    ExadataExternalCatalogConnectionPropertyKey["ExadataStagingRegion"] = "EXADATA_STAGING_REGION";
    ExadataExternalCatalogConnectionPropertyKey["ExadataStagingNamespace"] = "EXADATA_STAGING_NAMESPACE";
    ExadataExternalCatalogConnectionPropertyKey["ExadataStagingBucket"] = "EXADATA_STAGING_BUCKET";
    ExadataExternalCatalogConnectionPropertyKey["ExadataStagingFilePrefix"] = "EXADATA_STAGING_FILE_PREFIX";
    ExadataExternalCatalogConnectionPropertyKey["WorkspaceKey"] = "WORKSPACE_KEY";
})(ExadataExternalCatalogConnectionPropertyKey || (exports.ExadataExternalCatalogConnectionPropertyKey = ExadataExternalCatalogConnectionPropertyKey = {}));
(function (ExadataExternalCatalogConnectionPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    ExadataExternalCatalogConnectionPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ExadataExternalCatalogConnectionPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExadataExternalCatalogConnectionPropertyKey || (exports.ExadataExternalCatalogConnectionPropertyKey = ExadataExternalCatalogConnectionPropertyKey = {}));
