// aidp-typescript-client/dist/lib/model/experiment-run-status.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExperimentRunStatus = void 0;
/**
 * Status of the run.
**/
var ExperimentRunStatus;
(function (ExperimentRunStatus) {
    ExperimentRunStatus["Running"] = "RUNNING";
    ExperimentRunStatus["Scheduled"] = "SCHEDULED";
    ExperimentRunStatus["Finished"] = "FINISHED";
    ExperimentRunStatus["Failed"] = "FAILED";
    ExperimentRunStatus["Killed"] = "KILLED";
    ExperimentRunStatus["InternalError"] = "INTERNAL_ERROR";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ExperimentRunStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(ExperimentRunStatus || (exports.ExperimentRunStatus = ExperimentRunStatus = {}));
(function (ExperimentRunStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    ExperimentRunStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ExperimentRunStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExperimentRunStatus || (exports.ExperimentRunStatus = ExperimentRunStatus = {}));
