// aidp-typescript-client/dist/lib/model/exported-task-run-output-contents.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportedTaskRunOutputContents = void 0;
var ExportedTaskRunOutputContents;
(function (ExportedTaskRunOutputContents) {
    let Format;
    (function (Format) {
        Format["Html"] = "HTML";
        Format["Ipynb"] = "IPYNB";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Format["UnknownValue"] = "UNKNOWN_VALUE";
    })(Format = ExportedTaskRunOutputContents.Format || (ExportedTaskRunOutputContents.Format = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExportedTaskRunOutputContents.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExportedTaskRunOutputContents.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExportedTaskRunOutputContents || (exports.ExportedTaskRunOutputContents = ExportedTaskRunOutputContents = {}));
