// aidp-typescript-client/dist/lib/model/exported-contents.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportedContents = void 0;
var ExportedContents;
(function (ExportedContents) {
    let Format;
    (function (Format) {
        Format["Html"] = "html";
        Format["Ipynb"] = "ipynb";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Format["UnknownValue"] = "UNKNOWN_VALUE";
    })(Format = ExportedContents.Format || (ExportedContents.Format = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExportedContents.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExportedContents.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExportedContents || (exports.ExportedContents = ExportedContents = {}));
