// aidp-typescript-client/dist/lib/model/external-table-location-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalTableLocationType = void 0;
/**
 * The location type of an external table.
**/
var ExternalTableLocationType;
(function (ExternalTableLocationType) {
    ExternalTableLocationType["ObjectStorage"] = "OBJECT_STORAGE";
    ExternalTableLocationType["Mount"] = "MOUNT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ExternalTableLocationType["UnknownValue"] = "UNKNOWN_VALUE";
})(ExternalTableLocationType || (exports.ExternalTableLocationType = ExternalTableLocationType = {}));
(function (ExternalTableLocationType) {
    function getJsonObj(obj) {
        return obj;
    }
    ExternalTableLocationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ExternalTableLocationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExternalTableLocationType || (exports.ExternalTableLocationType = ExternalTableLocationType = {}));
