// aidp-typescript-client/dist/lib/model/external-catalog-source-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalCatalogSourceType = void 0;
/**
 * The source type of an external catalog.
**/
var ExternalCatalogSourceType;
(function (ExternalCatalogSourceType) {
    ExternalCatalogSourceType["Adw"] = "ADW";
    ExternalCatalogSourceType["Alh"] = "ALH";
    ExternalCatalogSourceType["Kafka"] = "KAFKA";
    ExternalCatalogSourceType["Atp"] = "ATP";
    ExternalCatalogSourceType["Oracle"] = "ORACLE";
    ExternalCatalogSourceType["Exadata"] = "EXADATA";
    ExternalCatalogSourceType["OracleAnalytics"] = "ORACLE_ANALYTICS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ExternalCatalogSourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(ExternalCatalogSourceType || (exports.ExternalCatalogSourceType = ExternalCatalogSourceType = {}));
(function (ExternalCatalogSourceType) {
    function getJsonObj(obj) {
        return obj;
    }
    ExternalCatalogSourceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ExternalCatalogSourceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExternalCatalogSourceType || (exports.ExternalCatalogSourceType = ExternalCatalogSourceType = {}));
