// aidp-typescript-client/dist/lib/model/fetch-mcp-objects-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FetchMcpObjectsDetails = void 0;
const model = __importStar(require("../model"));
var FetchMcpObjectsDetails;
(function (FetchMcpObjectsDetails) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = FetchMcpObjectsDetails.SortOrder || (FetchMcpObjectsDetails.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
    })(SortBy = FetchMcpObjectsDetails.SortBy || (FetchMcpObjectsDetails.SortBy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'mcpTool': obj.mcpTool ?
                model.McpTool.getJsonObj(obj.mcpTool) : undefined,
        });
        return jsonObj;
    }
    FetchMcpObjectsDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'mcpTool': obj.mcpTool ?
                model.McpTool.getDeserializedJsonObj(obj.mcpTool) : undefined,
        });
        return jsonObj;
    }
    FetchMcpObjectsDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(FetchMcpObjectsDetails || (exports.FetchMcpObjectsDetails = FetchMcpObjectsDetails = {}));
