// aidp-typescript-client/dist/lib/model/field-info.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldInfo = void 0;
var FieldInfo;
(function (FieldInfo) {
    let FieldType;
    (function (FieldType) {
        FieldType["String"] = "STRING";
        FieldType["Number"] = "NUMBER";
        FieldType["Boolean"] = "BOOLEAN";
        FieldType["Array"] = "ARRAY";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        FieldType["UnknownValue"] = "UNKNOWN_VALUE";
    })(FieldType = FieldInfo.FieldType || (FieldInfo.FieldType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FieldInfo.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FieldInfo.getDeserializedJsonObj = getDeserializedJsonObj;
})(FieldInfo || (exports.FieldInfo = FieldInfo = {}));
