// aidp-typescript-client/dist/lib/model/git-config.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.GitConfig = void 0;
var GitConfig;
(function (GitConfig) {
    let Provider;
    (function (Provider) {
        Provider["Github"] = "GITHUB";
        Provider["Bitbucket"] = "BITBUCKET";
        Provider["Gitlab"] = "GITLAB";
        Provider["OciDevops"] = "OCI_DEVOPS";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Provider["UnknownValue"] = "UNKNOWN_VALUE";
    })(Provider = GitConfig.Provider || (GitConfig.Provider = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitConfig.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitConfig.getDeserializedJsonObj = getDeserializedJsonObj;
})(GitConfig || (exports.GitConfig = GitConfig = {}));
