// aidp-typescript-client/dist/lib/model/git-diff-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.GitDiffSummary = void 0;
var GitDiffSummary;
(function (GitDiffSummary) {
    let ChangeType;
    (function (ChangeType) {
        ChangeType["Added"] = "ADDED";
        ChangeType["Modified"] = "MODIFIED";
        ChangeType["Deleted"] = "DELETED";
        ChangeType["Renamed"] = "RENAMED";
        ChangeType["Copied"] = "COPIED";
        ChangeType["TypeChanged"] = "TYPE_CHANGED";
        ChangeType["Unmerged"] = "UNMERGED";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        ChangeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ChangeType = GitDiffSummary.ChangeType || (GitDiffSummary.ChangeType = {}));
    let ConflictType;
    (function (ConflictType) {
        ConflictType["BothModified"] = "BOTH_MODIFIED";
        ConflictType["BothAdded"] = "BOTH_ADDED";
        ConflictType["BothDeleted"] = "BOTH_DELETED";
        ConflictType["AddedByUs"] = "ADDED_BY_US";
        ConflictType["AddedByThem"] = "ADDED_BY_THEM";
        ConflictType["DeletedByUs"] = "DELETED_BY_US";
        ConflictType["DeletedByThem"] = "DELETED_BY_THEM";
        ConflictType["Unknown"] = "UNKNOWN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        ConflictType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ConflictType = GitDiffSummary.ConflictType || (GitDiffSummary.ConflictType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitDiffSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitDiffSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(GitDiffSummary || (exports.GitDiffSummary = GitDiffSummary = {}));
