// aidp-typescript-client/dist/lib/model/git-diff-detail.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.GitDiffDetail = void 0;
var GitDiffDetail;
(function (GitDiffDetail) {
    let TruncatedReason;
    (function (TruncatedReason) {
        TruncatedReason["MaxPatchBytes"] = "MAX_PATCH_BYTES";
        TruncatedReason["Binary"] = "BINARY";
        TruncatedReason["TooLarge"] = "TOO_LARGE";
        TruncatedReason["Unknown"] = "UNKNOWN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        TruncatedReason["UnknownValue"] = "UNKNOWN_VALUE";
    })(TruncatedReason = GitDiffDetail.TruncatedReason || (GitDiffDetail.TruncatedReason = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitDiffDetail.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitDiffDetail.getDeserializedJsonObj = getDeserializedJsonObj;
})(GitDiffDetail || (exports.GitDiffDetail = GitDiffDetail = {}));
