// aidp-typescript-client/dist/lib/model/git-operation-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.GitOperationState = void 0;
var GitOperationState;
(function (GitOperationState) {
    let State;
    (function (State) {
        State["Clean"] = "CLEAN";
        State["MergeInProgress"] = "MERGE_IN_PROGRESS";
        State["RebaseInProgress"] = "REBASE_IN_PROGRESS";
        State["PullInProgress"] = "PULL_IN_PROGRESS";
        State["DetachedHead"] = "DETACHED_HEAD";
        State["Unknown"] = "UNKNOWN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        State["UnknownValue"] = "UNKNOWN_VALUE";
    })(State = GitOperationState.State || (GitOperationState.State = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitOperationState.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GitOperationState.getDeserializedJsonObj = getDeserializedJsonObj;
})(GitOperationState || (exports.GitOperationState = GitOperationState = {}));
