// aidp-typescript-client/dist/lib/model/git-account-provider-name.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.GitAccountProviderName = void 0;
/**
 * The name of the Git provider.
**/
var GitAccountProviderName;
(function (GitAccountProviderName) {
    GitAccountProviderName["Github"] = "GITHUB";
    GitAccountProviderName["Bitbucket"] = "BITBUCKET";
    GitAccountProviderName["Gitlab"] = "GITLAB";
    GitAccountProviderName["OciDevops"] = "OCI_DEVOPS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    GitAccountProviderName["UnknownValue"] = "UNKNOWN_VALUE";
})(GitAccountProviderName || (exports.GitAccountProviderName = GitAccountProviderName = {}));
(function (GitAccountProviderName) {
    function getJsonObj(obj) {
        return obj;
    }
    GitAccountProviderName.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    GitAccountProviderName.getDeserializedJsonObj = getDeserializedJsonObj;
})(GitAccountProviderName || (exports.GitAccountProviderName = GitAccountProviderName = {}));
