// aidp-typescript-client/dist/lib/model/identity-domain-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityDomainSummary = void 0;
var IdentityDomainSummary;
(function (IdentityDomainSummary) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Inactive"] = "INACTIVE";
    })(LifecycleState = IdentityDomainSummary.LifecycleState || (IdentityDomainSummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    IdentityDomainSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    IdentityDomainSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(IdentityDomainSummary || (exports.IdentityDomainSummary = IdentityDomainSummary = {}));
