// aidp-typescript-client/dist/lib/model/job-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobPermissionSummary = void 0;
var JobPermissionSummary;
(function (JobPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Use"] = "USE";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["Admin"] = "ADMIN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = JobPermissionSummary.GranteePermissions || (JobPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    JobPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    JobPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(JobPermissionSummary || (exports.JobPermissionSummary = JobPermissionSummary = {}));
