// aidp-typescript-client/dist/lib/model/kafka-external-catalog-connection-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KafkaExternalCatalogConnectionPropertyKey = void 0;
/**
 * Kafka external catalog connection property keys.
**/
var KafkaExternalCatalogConnectionPropertyKey;
(function (KafkaExternalCatalogConnectionPropertyKey) {
    KafkaExternalCatalogConnectionPropertyKey["BootstrapServers"] = "BOOTSTRAP_SERVERS";
    KafkaExternalCatalogConnectionPropertyKey["Username"] = "USERNAME";
    KafkaExternalCatalogConnectionPropertyKey["Password"] = "PASSWORD";
    KafkaExternalCatalogConnectionPropertyKey["EnableSsl"] = "ENABLE_SSL";
    KafkaExternalCatalogConnectionPropertyKey["SaslMode"] = "SASL_MODE";
})(KafkaExternalCatalogConnectionPropertyKey || (exports.KafkaExternalCatalogConnectionPropertyKey = KafkaExternalCatalogConnectionPropertyKey = {}));
(function (KafkaExternalCatalogConnectionPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    KafkaExternalCatalogConnectionPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KafkaExternalCatalogConnectionPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(KafkaExternalCatalogConnectionPropertyKey || (exports.KafkaExternalCatalogConnectionPropertyKey = KafkaExternalCatalogConnectionPropertyKey = {}));
