// aidp-typescript-client/dist/lib/model/kb-vhnsw-index-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KbVHnswIndexDetails = void 0;
var KbVHnswIndexDetails;
(function (KbVHnswIndexDetails) {
    let Distance;
    (function (Distance) {
        Distance["Cosine"] = "COSINE";
        Distance["Euclidean"] = "EUCLIDEAN";
        Distance["L2Squared"] = "L2_SQUARED";
        Distance["Dot"] = "DOT";
        Distance["Manhattan"] = "MANHATTAN";
        Distance["Hamming"] = "HAMMING";
        Distance["Jaccard"] = "JACCARD";
    })(Distance = KbVHnswIndexDetails.Distance || (KbVHnswIndexDetails.Distance = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KbVHnswIndexDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KbVHnswIndexDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(KbVHnswIndexDetails || (exports.KbVHnswIndexDetails = KbVHnswIndexDetails = {}));
