// aidp-typescript-client/dist/lib/model/kb-vivf-index-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KbVIvfIndexDetails = void 0;
var KbVIvfIndexDetails;
(function (KbVIvfIndexDetails) {
    let Distance;
    (function (Distance) {
        Distance["Cosine"] = "COSINE";
        Distance["Euclidean"] = "EUCLIDEAN";
        Distance["L2Squared"] = "L2_SQUARED";
        Distance["Dot"] = "DOT";
        Distance["Manhattan"] = "MANHATTAN";
        Distance["Hamming"] = "HAMMING";
        Distance["Jaccard"] = "JACCARD";
    })(Distance = KbVIvfIndexDetails.Distance || (KbVIvfIndexDetails.Distance = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KbVIvfIndexDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KbVIvfIndexDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(KbVIvfIndexDetails || (exports.KbVIvfIndexDetails = KbVIvfIndexDetails = {}));
