// aidp-typescript-client/dist/lib/model/kernel.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Kernel = void 0;
var Kernel;
(function (Kernel) {
    let ExecutionState;
    (function (ExecutionState) {
        ExecutionState["Unknown"] = "unknown";
        ExecutionState["Starting"] = "starting";
        ExecutionState["Idle"] = "idle";
        ExecutionState["Busy"] = "busy";
        ExecutionState["Terminating"] = "terminating";
        ExecutionState["Restarting"] = "restarting";
        ExecutionState["Autorestarting"] = "autorestarting";
        ExecutionState["Dead"] = "dead";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        ExecutionState["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExecutionState = Kernel.ExecutionState || (Kernel.ExecutionState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'last_activity': obj.lastActivity,
            'execution_state': obj.executionState,
        });
        delete jsonObj.lastActivity;
        delete jsonObj.executionState;
        return jsonObj;
    }
    Kernel.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'lastActivity': obj["last_activity"],
            'executionState': obj["execution_state"],
        });
        delete jsonObj["last_activity"];
        delete jsonObj["execution_state"];
        return jsonObj;
    }
    Kernel.getDeserializedJsonObj = getDeserializedJsonObj;
})(Kernel || (exports.Kernel = Kernel = {}));
