// aidp-typescript-client/dist/lib/model/knowledge-base-update-operation-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBaseUpdateOperationType = void 0;
/**
 * Update supported in KnowledgeBase/source
**/
var KnowledgeBaseUpdateOperationType;
(function (KnowledgeBaseUpdateOperationType) {
    KnowledgeBaseUpdateOperationType["MetadataUpdate"] = "METADATA_UPDATE";
    KnowledgeBaseUpdateOperationType["SourcesUpdate"] = "SOURCES_UPDATE";
    KnowledgeBaseUpdateOperationType["AddIndex"] = "ADD_INDEX";
    KnowledgeBaseUpdateOperationType["DropIndex"] = "DROP_INDEX";
})(KnowledgeBaseUpdateOperationType || (exports.KnowledgeBaseUpdateOperationType = KnowledgeBaseUpdateOperationType = {}));
(function (KnowledgeBaseUpdateOperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseUpdateOperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseUpdateOperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(KnowledgeBaseUpdateOperationType || (exports.KnowledgeBaseUpdateOperationType = KnowledgeBaseUpdateOperationType = {}));
