// aidp-typescript-client/dist/lib/model/knowledge-base-embedding-model-source-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBaseEmbeddingModelSourceType = void 0;
/**
 * Type of Sources supported where Embedding is generated
**/
var KnowledgeBaseEmbeddingModelSourceType;
(function (KnowledgeBaseEmbeddingModelSourceType) {
    KnowledgeBaseEmbeddingModelSourceType["Adw23Ai"] = "ADW_23_AI";
    KnowledgeBaseEmbeddingModelSourceType["Adw26Ai"] = "ADW_26_AI";
    KnowledgeBaseEmbeddingModelSourceType["GenAi"] = "GEN_AI";
})(KnowledgeBaseEmbeddingModelSourceType || (exports.KnowledgeBaseEmbeddingModelSourceType = KnowledgeBaseEmbeddingModelSourceType = {}));
(function (KnowledgeBaseEmbeddingModelSourceType) {
    function getJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseEmbeddingModelSourceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseEmbeddingModelSourceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(KnowledgeBaseEmbeddingModelSourceType || (exports.KnowledgeBaseEmbeddingModelSourceType = KnowledgeBaseEmbeddingModelSourceType = {}));
