// aidp-typescript-client/dist/lib/model/knowledge-base-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBasePermissionSummary = void 0;
var KnowledgeBasePermissionSummary;
(function (KnowledgeBasePermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Write"] = "WRITE";
        GranteePermissions["Admin"] = "ADMIN";
        GranteePermissions["Select"] = "SELECT";
        GranteePermissions["Manage"] = "MANAGE";
    })(GranteePermissions = KnowledgeBasePermissionSummary.GranteePermissions || (KnowledgeBasePermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KnowledgeBasePermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KnowledgeBasePermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(KnowledgeBasePermissionSummary || (exports.KnowledgeBasePermissionSummary = KnowledgeBasePermissionSummary = {}));
