// aidp-typescript-client/dist/lib/model/knowledge-base-job-run-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBaseJobRunLifecycleState = void 0;
/**
 * Lifecycle state for KnowledgeBase Job Run
**/
var KnowledgeBaseJobRunLifecycleState;
(function (KnowledgeBaseJobRunLifecycleState) {
    KnowledgeBaseJobRunLifecycleState["Accepted"] = "ACCEPTED";
    KnowledgeBaseJobRunLifecycleState["Canceling"] = "CANCELING";
    KnowledgeBaseJobRunLifecycleState["Canceled"] = "CANCELED";
    KnowledgeBaseJobRunLifecycleState["Failed"] = "FAILED";
    KnowledgeBaseJobRunLifecycleState["Succeeded"] = "SUCCEEDED";
    KnowledgeBaseJobRunLifecycleState["InProgress"] = "IN_PROGRESS";
})(KnowledgeBaseJobRunLifecycleState || (exports.KnowledgeBaseJobRunLifecycleState = KnowledgeBaseJobRunLifecycleState = {}));
(function (KnowledgeBaseJobRunLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseJobRunLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseJobRunLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(KnowledgeBaseJobRunLifecycleState || (exports.KnowledgeBaseJobRunLifecycleState = KnowledgeBaseJobRunLifecycleState = {}));
