// aidp-typescript-client/dist/lib/model/knowledge-base-job-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBaseJobLifecycleState = void 0;
/**
 * Lifecycle state for KB JOB
**/
var KnowledgeBaseJobLifecycleState;
(function (KnowledgeBaseJobLifecycleState) {
    KnowledgeBaseJobLifecycleState["Active"] = "ACTIVE";
    KnowledgeBaseJobLifecycleState["Inactive"] = "INACTIVE";
    KnowledgeBaseJobLifecycleState["Creating"] = "CREATING";
    KnowledgeBaseJobLifecycleState["Deleting"] = "DELETING";
})(KnowledgeBaseJobLifecycleState || (exports.KnowledgeBaseJobLifecycleState = KnowledgeBaseJobLifecycleState = {}));
(function (KnowledgeBaseJobLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseJobLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseJobLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(KnowledgeBaseJobLifecycleState || (exports.KnowledgeBaseJobLifecycleState = KnowledgeBaseJobLifecycleState = {}));
