// aidp-typescript-client/dist/lib/model/knowledge-base-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBaseLifecycleState = void 0;
/**
 * Lifecycle state for KnowledgeBase
**/
var KnowledgeBaseLifecycleState;
(function (KnowledgeBaseLifecycleState) {
    KnowledgeBaseLifecycleState["Creating"] = "CREATING";
    KnowledgeBaseLifecycleState["Active"] = "ACTIVE";
    KnowledgeBaseLifecycleState["Updating"] = "UPDATING";
    KnowledgeBaseLifecycleState["Deleting"] = "DELETING";
    KnowledgeBaseLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    KnowledgeBaseLifecycleState["Deleted"] = "DELETED";
    KnowledgeBaseLifecycleState["Failed"] = "FAILED";
})(KnowledgeBaseLifecycleState || (exports.KnowledgeBaseLifecycleState = KnowledgeBaseLifecycleState = {}));
(function (KnowledgeBaseLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KnowledgeBaseLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(KnowledgeBaseLifecycleState || (exports.KnowledgeBaseLifecycleState = KnowledgeBaseLifecycleState = {}));
