// aidp-typescript-client/dist/lib/model/lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.LifecycleState = void 0;
var LifecycleState;
(function (LifecycleState) {
    let Status;
    (function (Status) {
        Status["Pending"] = "PENDING";
        Status["Queued"] = "QUEUED";
        Status["Running"] = "RUNNING";
        Status["Skipped"] = "SKIPPED";
        Status["InternalError"] = "INTERNAL_ERROR";
        Status["Blocked"] = "BLOCKED";
        Status["Success"] = "SUCCESS";
        Status["Failed"] = "FAILED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
        Status["UpstreamCanceled"] = "UPSTREAM_CANCELED";
        Status["UpstreamFailed"] = "UPSTREAM_FAILED";
        Status["Excluded"] = "EXCLUDED";
        Status["TimedOut"] = "TIMED_OUT";
        Status["PausedMaintenance"] = "PAUSED_MAINTENANCE";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = LifecycleState.Status || (LifecycleState.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LifecycleState.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(LifecycleState || (exports.LifecycleState = LifecycleState = {}));
