// aidp-typescript-client/dist/lib/model/list-experiments-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListExperimentsDetails = void 0;
var ListExperimentsDetails;
(function (ListExperimentsDetails) {
    let ViewType;
    (function (ViewType) {
        ViewType["ActiveOnly"] = "ACTIVE_ONLY";
        ViewType["DeletedOnly"] = "DELETED_ONLY";
        ViewType["All"] = "ALL";
    })(ViewType = ListExperimentsDetails.ViewType || (ListExperimentsDetails.ViewType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'max_results': obj.maxResults,
            'page_token': obj.pageToken,
            'order_by': obj.orderBy,
            'view_type': obj.viewType,
        });
        delete jsonObj.maxResults;
        delete jsonObj.pageToken;
        delete jsonObj.orderBy;
        delete jsonObj.viewType;
        return jsonObj;
    }
    ListExperimentsDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'maxResults': obj["max_results"],
            'pageToken': obj["page_token"],
            'orderBy': obj["order_by"],
            'viewType': obj["view_type"],
        });
        delete jsonObj["max_results"];
        delete jsonObj["page_token"];
        delete jsonObj["order_by"];
        delete jsonObj["view_type"];
        return jsonObj;
    }
    ListExperimentsDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ListExperimentsDetails || (exports.ListExperimentsDetails = ListExperimentsDetails = {}));
