// aidp-typescript-client/dist/lib/model/list-role-permission-scope-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRolePermissionScopeType = void 0;
/**
 * The permission scope for a role based on how it was obtained.
**/
var ListRolePermissionScopeType;
(function (ListRolePermissionScopeType) {
    ListRolePermissionScopeType["Direct"] = "DIRECT";
    ListRolePermissionScopeType["Inherited"] = "INHERITED";
    ListRolePermissionScopeType["All"] = "ALL";
})(ListRolePermissionScopeType || (exports.ListRolePermissionScopeType = ListRolePermissionScopeType = {}));
(function (ListRolePermissionScopeType) {
    function getJsonObj(obj) {
        return obj;
    }
    ListRolePermissionScopeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ListRolePermissionScopeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ListRolePermissionScopeType || (exports.ListRolePermissionScopeType = ListRolePermissionScopeType = {}));
