// aidp-typescript-client/dist/lib/model/list-experiment-runs-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListExperimentRunsDetails = void 0;
var ListExperimentRunsDetails;
(function (ListExperimentRunsDetails) {
    let RunViewType;
    (function (RunViewType) {
        RunViewType["ActiveOnly"] = "ACTIVE_ONLY";
        RunViewType["DeletedOnly"] = "DELETED_ONLY";
        RunViewType["All"] = "ALL";
    })(RunViewType = ListExperimentRunsDetails.RunViewType || (ListExperimentRunsDetails.RunViewType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'experiment_ids': obj.experimentIds,
            'max_results': obj.maxResults,
            'page_token': obj.pageToken,
            'order_by': obj.orderBy,
            'run_view_type': obj.runViewType,
        });
        delete jsonObj.experimentIds;
        delete jsonObj.maxResults;
        delete jsonObj.pageToken;
        delete jsonObj.orderBy;
        delete jsonObj.runViewType;
        return jsonObj;
    }
    ListExperimentRunsDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'experimentIds': obj["experiment_ids"],
            'maxResults': obj["max_results"],
            'pageToken': obj["page_token"],
            'orderBy': obj["order_by"],
            'runViewType': obj["run_view_type"],
        });
        delete jsonObj["experiment_ids"];
        delete jsonObj["max_results"];
        delete jsonObj["page_token"];
        delete jsonObj["order_by"];
        delete jsonObj["run_view_type"];
        return jsonObj;
    }
    ListExperimentRunsDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ListExperimentRunsDetails || (exports.ListExperimentRunsDetails = ListExperimentRunsDetails = {}));
