// aidp-typescript-client/dist/lib/model/list-filter-enum.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFilterEnum = void 0;
/**
 * Accepted values of list filters
**/
var ListFilterEnum;
(function (ListFilterEnum) {
    ListFilterEnum["Type"] = "TYPE";
    ListFilterEnum["Owner"] = "OWNER";
    ListFilterEnum["CreatedBy"] = "CREATED_BY";
    ListFilterEnum["Catalog"] = "CATALOG";
    ListFilterEnum["Schema"] = "SCHEMA";
    ListFilterEnum["Workspace"] = "WORKSPACE";
    ListFilterEnum["Job"] = "JOB";
    ListFilterEnum["Cluster"] = "CLUSTER";
})(ListFilterEnum || (exports.ListFilterEnum = ListFilterEnum = {}));
(function (ListFilterEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    ListFilterEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ListFilterEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(ListFilterEnum || (exports.ListFilterEnum = ListFilterEnum = {}));
