// aidp-typescript-client/dist/lib/model/master-catalog-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MasterCatalogLifecycleState = void 0;
/**
 * The state of the Master Catalog
**/
var MasterCatalogLifecycleState;
(function (MasterCatalogLifecycleState) {
    MasterCatalogLifecycleState["Active"] = "ACTIVE";
    MasterCatalogLifecycleState["Creating"] = "CREATING";
    MasterCatalogLifecycleState["Deleting"] = "DELETING";
})(MasterCatalogLifecycleState || (exports.MasterCatalogLifecycleState = MasterCatalogLifecycleState = {}));
(function (MasterCatalogLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    MasterCatalogLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MasterCatalogLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(MasterCatalogLifecycleState || (exports.MasterCatalogLifecycleState = MasterCatalogLifecycleState = {}));
