// aidp-typescript-client/dist/lib/model/master-catalog-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MasterCatalogPermissionSummary = void 0;
var MasterCatalogPermissionSummary;
(function (MasterCatalogPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["CreateCatalog"] = "CREATE_CATALOG";
        GranteePermissions["Admin"] = "ADMIN";
        GranteePermissions["CreateShare"] = "CREATE_SHARE";
        GranteePermissions["CreateRecipient"] = "CREATE_RECIPIENT";
        GranteePermissions["CreateCredential"] = "CREATE_CREDENTIAL";
    })(GranteePermissions = MasterCatalogPermissionSummary.GranteePermissions || (MasterCatalogPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MasterCatalogPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MasterCatalogPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(MasterCatalogPermissionSummary || (exports.MasterCatalogPermissionSummary = MasterCatalogPermissionSummary = {}));
