// aidp-typescript-client/dist/lib/model/master-catalog-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MasterCatalogPrivilege = void 0;
/**
 * The privilege for a Master Catalog. MASTER CATALOG ADMIN: This permission will allow the user to: - Grant/Revoke permissions - View all catalogs in the Master Catalog - Create Catalogs - DELETE Catalog CREATE CATALOG: This permission will enable users to add create a catalog (internal or external)
**/
var MasterCatalogPrivilege;
(function (MasterCatalogPrivilege) {
    MasterCatalogPrivilege["CreateCatalog"] = "CREATE_CATALOG";
    MasterCatalogPrivilege["Admin"] = "ADMIN";
    MasterCatalogPrivilege["CreateShare"] = "CREATE_SHARE";
    MasterCatalogPrivilege["CreateRecipient"] = "CREATE_RECIPIENT";
    MasterCatalogPrivilege["CreateCredential"] = "CREATE_CREDENTIAL";
})(MasterCatalogPrivilege || (exports.MasterCatalogPrivilege = MasterCatalogPrivilege = {}));
(function (MasterCatalogPrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    MasterCatalogPrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MasterCatalogPrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(MasterCatalogPrivilege || (exports.MasterCatalogPrivilege = MasterCatalogPrivilege = {}));
