// aidp-typescript-client/dist/lib/model/model-version-status.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelVersionStatus = void 0;
/**
 * Status of the run.
**/
var ModelVersionStatus;
(function (ModelVersionStatus) {
    ModelVersionStatus["PendingRegistration"] = "PENDING_REGISTRATION";
    ModelVersionStatus["FailedRegistration"] = "FAILED_REGISTRATION";
    ModelVersionStatus["Ready"] = "READY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ModelVersionStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(ModelVersionStatus || (exports.ModelVersionStatus = ModelVersionStatus = {}));
(function (ModelVersionStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    ModelVersionStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ModelVersionStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(ModelVersionStatus || (exports.ModelVersionStatus = ModelVersionStatus = {}));
