// aidp-typescript-client/dist/lib/model/node-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeType = void 0;
/**
 * Node type.
**/
var NodeType;
(function (NodeType) {
    NodeType["StartNode"] = "START_NODE";
    NodeType["Guardrails"] = "GUARDRAILS";
    NodeType["Agent"] = "AGENT";
    NodeType["SupervisorAgent"] = "SUPERVISOR_AGENT";
    NodeType["NestedAgentFlow"] = "NESTED_AGENT_FLOW";
    NodeType["ExternalAgent"] = "EXTERNAL_AGENT";
    NodeType["HumanInTheLoop"] = "HUMAN_IN_THE_LOOP";
    NodeType["SqlTool"] = "SQL_TOOL";
    NodeType["PromptTool"] = "PROMPT_TOOL";
    NodeType["RagTool"] = "RAG_TOOL";
    NodeType["McpTool"] = "MCP_TOOL";
    NodeType["HttpTool"] = "HTTP_TOOL";
    NodeType["CustomTool"] = "CUSTOM_TOOL";
})(NodeType || (exports.NodeType = NodeType = {}));
(function (NodeType) {
    function getJsonObj(obj) {
        return obj;
    }
    NodeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NodeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(NodeType || (exports.NodeType = NodeType = {}));
