// aidp-typescript-client/dist/lib/model/node-io.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeIo = void 0;
const model = __importStar(require("../model"));
var NodeIo;
(function (NodeIo) {
    let DataType;
    (function (DataType) {
        DataType["Flow"] = "FLOW";
        DataType["Str"] = "STR";
        DataType["Int"] = "INT";
        DataType["Bool"] = "BOOL";
        DataType["Float"] = "FLOAT";
        DataType["List"] = "LIST";
        DataType["Dict"] = "DICT";
        DataType["Any"] = "ANY";
        DataType["Tools"] = "TOOLS";
        DataType["Llms"] = "LLMS";
        DataType["Agents"] = "AGENTS";
        DataType["Planner"] = "PLANNER";
        DataType["Flows"] = "FLOWS";
    })(DataType = NodeIo.DataType || (NodeIo.DataType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'color': obj.color ?
                model.NodeIoColor.getJsonObj(obj.color) : undefined,
        });
        return jsonObj;
    }
    NodeIo.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'color': obj.color ?
                model.NodeIoColor.getDeserializedJsonObj(obj.color) : undefined,
        });
        return jsonObj;
    }
    NodeIo.getDeserializedJsonObj = getDeserializedJsonObj;
})(NodeIo || (exports.NodeIo = NodeIo = {}));
