// aidp-typescript-client/dist/lib/model/notebook-task.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotebookTask = void 0;
const model = __importStar(require("../model"));
var NotebookTask;
(function (NotebookTask) {
    let Source;
    (function (Source) {
        Source["Workspace"] = "WORKSPACE";
        Source["GitProvider"] = "GIT_PROVIDER";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Source["UnknownValue"] = "UNKNOWN_VALUE";
    })(Source = NotebookTask.Source || (NotebookTask.Source = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, isParentJsonObj ? obj : model.Task.getJsonObj(obj)), {
            'gitConfig': obj.gitConfig ?
                model.GitConfig.getJsonObj(obj.gitConfig) : undefined,
            'cluster': obj.cluster ?
                model.JobCluster.getJsonObj(obj.cluster) : undefined,
            'parameters': obj.parameters ?
                obj.parameters.map((item) => { return model.Parameter.getJsonObj(item); })
                : undefined,
        });
        return jsonObj;
    }
    NotebookTask.getJsonObj = getJsonObj;
    NotebookTask.type = 'NOTEBOOK_TASK';
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, isParentJsonObj ? obj : model.Task.getDeserializedJsonObj(obj)), {
            'gitConfig': obj.gitConfig ?
                model.GitConfig.getDeserializedJsonObj(obj.gitConfig) : undefined,
            'cluster': obj.cluster ?
                model.JobCluster.getDeserializedJsonObj(obj.cluster) : undefined,
            'parameters': obj.parameters ?
                obj.parameters.map((item) => { return model.Parameter.getDeserializedJsonObj(item); })
                : undefined,
        });
        return jsonObj;
    }
    NotebookTask.getDeserializedJsonObj = getDeserializedJsonObj;
})(NotebookTask || (exports.NotebookTask = NotebookTask = {}));
