// aidp-typescript-client/dist/lib/model/notify-cluster-event-handler-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotifyClusterEventHandlerDetails = void 0;
const model = __importStar(require("../model"));
var NotifyClusterEventHandlerDetails;
(function (NotifyClusterEventHandlerDetails) {
    let Phase;
    (function (Phase) {
        Phase["Started"] = "STARTED";
        Phase["Completed"] = "COMPLETED";
    })(Phase = NotifyClusterEventHandlerDetails.Phase || (NotifyClusterEventHandlerDetails.Phase = {}));
    let State;
    (function (State) {
        State["Accepted"] = "ACCEPTED";
        State["Creating"] = "CREATING";
        State["Active"] = "ACTIVE";
        State["Deleting"] = "DELETING";
        State["Deleted"] = "DELETED";
        State["Failed"] = "FAILED";
        State["Stopping"] = "STOPPING";
        State["Stopped"] = "STOPPED";
        State["Updating"] = "UPDATING";
        State["Restarting"] = "RESTARTING";
        State["Starting"] = "STARTING";
        State["NetworkConfigurationAttachInProgress"] = "NETWORK_CONFIGURATION_ATTACH_IN_PROGRESS";
        State["NetworkConfigurationAttachSuccessful"] = "NETWORK_CONFIGURATION_ATTACH_SUCCESSFUL";
        State["NetworkConfigurationAttachFailed"] = "NETWORK_CONFIGURATION_ATTACH_FAILED";
        State["NetworkConfigurationDetachInProgress"] = "NETWORK_CONFIGURATION_DETACH_IN_PROGRESS";
        State["NetworkConfigurationDetachSuccessful"] = "NETWORK_CONFIGURATION_DETACH_SUCCESSFUL";
        State["NetworkConfigurationDetachFailed"] = "NETWORK_CONFIGURATION_DETACH_FAILED";
    })(State = NotifyClusterEventHandlerDetails.State || (NotifyClusterEventHandlerDetails.State = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'clusterEvent': obj.clusterEvent ?
                model.ClusterEvent.getJsonObj(obj.clusterEvent) : undefined,
        });
        return jsonObj;
    }
    NotifyClusterEventHandlerDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'clusterEvent': obj.clusterEvent ?
                model.ClusterEvent.getDeserializedJsonObj(obj.clusterEvent) : undefined,
        });
        return jsonObj;
    }
    NotifyClusterEventHandlerDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(NotifyClusterEventHandlerDetails || (exports.NotifyClusterEventHandlerDetails = NotifyClusterEventHandlerDetails = {}));
