// aidp-typescript-client/dist/lib/model/oac-external-catalog-connection-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.OacExternalCatalogConnectionPropertyKey = void 0;
/**
 * Oracle Analytics external catalog connection property keys
**/
var OacExternalCatalogConnectionPropertyKey;
(function (OacExternalCatalogConnectionPropertyKey) {
    OacExternalCatalogConnectionPropertyKey["OacEndpointUrl"] = "OAC_ENDPOINT_URL";
    OacExternalCatalogConnectionPropertyKey["OacIdcsEndpointUrl"] = "OAC_IDCS_ENDPOINT_URL";
    OacExternalCatalogConnectionPropertyKey["OacIdcsClientId"] = "OAC_IDCS_CLIENT_ID";
    OacExternalCatalogConnectionPropertyKey["OacIdcsClientScope"] = "OAC_IDCS_CLIENT_SCOPE";
    OacExternalCatalogConnectionPropertyKey["OacIdcsCertificate"] = "OAC_IDCS_CERTIFICATE";
    OacExternalCatalogConnectionPropertyKey["OacIdcsPrivateKey"] = "OAC_IDCS_PRIVATE_KEY";
})(OacExternalCatalogConnectionPropertyKey || (exports.OacExternalCatalogConnectionPropertyKey = OacExternalCatalogConnectionPropertyKey = {}));
(function (OacExternalCatalogConnectionPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    OacExternalCatalogConnectionPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OacExternalCatalogConnectionPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(OacExternalCatalogConnectionPropertyKey || (exports.OacExternalCatalogConnectionPropertyKey = OacExternalCatalogConnectionPropertyKey = {}));
