// aidp-typescript-client/dist/lib/model/object-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectType = void 0;
/**
 * Type of Object
**/
var ObjectType;
(function (ObjectType) {
    ObjectType["DataLake"] = "DATA_LAKE";
    ObjectType["Role"] = "ROLE";
    ObjectType["Catalog"] = "CATALOG";
    ObjectType["Schema"] = "SCHEMA";
    ObjectType["Table"] = "TABLE";
    ObjectType["Volume"] = "VOLUME";
    ObjectType["View"] = "VIEW";
    ObjectType["Workspace"] = "WORKSPACE";
    ObjectType["WorkspaceFile"] = "WORKSPACE_FILE";
    ObjectType["WorkspaceFolder"] = "WORKSPACE_FOLDER";
    ObjectType["Compute"] = "COMPUTE";
    ObjectType["Folder"] = "FOLDER";
    ObjectType["File"] = "FILE";
    ObjectType["AutoPopulate"] = "AUTO_POPULATE";
    ObjectType["VolumeFile"] = "VOLUME_FILE";
    ObjectType["VolumeFolder"] = "VOLUME_FOLDER";
    ObjectType["KnowledgeBase"] = "KNOWLEDGE_BASE";
    ObjectType["KnowledgeBaseJob"] = "KNOWLEDGE_BASE_JOB";
    ObjectType["KnowledgeBaseJobRun"] = "KNOWLEDGE_BASE_JOB_RUN";
    ObjectType["KnowledgeBaseOwnership"] = "KNOWLEDGE_BASE_OWNERSHIP";
    ObjectType["VectorStoreCredentials"] = "VECTOR_STORE_CREDENTIALS";
    ObjectType["AuditLog"] = "AUDIT_LOG";
    ObjectType["Workflow"] = "WORKFLOW";
    ObjectType["Credential"] = "CREDENTIAL";
    ObjectType["AgentFlow"] = "AGENT_FLOW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ObjectType["UnknownValue"] = "UNKNOWN_VALUE";
})(ObjectType || (exports.ObjectType = ObjectType = {}));
(function (ObjectType) {
    function getJsonObj(obj) {
        return obj;
    }
    ObjectType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ObjectType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ObjectType || (exports.ObjectType = ObjectType = {}));
