// aidp-typescript-client/dist/lib/model/operation.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Operation = void 0;
/**
 * Type of operation
**/
var Operation;
(function (Operation) {
    Operation["Create"] = "CREATE";
    Operation["Update"] = "UPDATE";
    Operation["Delete"] = "DELETE";
    Operation["View"] = "VIEW";
    Operation["Grant"] = "GRANT";
    Operation["Revoke"] = "REVOKE";
    Operation["Attach"] = "ATTACH";
    Operation["Detach"] = "DETACH";
    Operation["ViewLogs"] = "VIEW_LOGS";
    Operation["Rename"] = "RENAME";
    Operation["Terminate"] = "TERMINATE";
    Operation["Move"] = "MOVE";
    Operation["Execute"] = "EXECUTE";
    Operation["ManageAccess"] = "MANAGE_ACCESS";
    Operation["Query"] = "QUERY";
    Operation["Manage"] = "MANAGE";
    Operation["Read"] = "READ";
    Operation["Write"] = "WRITE";
    Operation["Start"] = "START";
    Operation["Stop"] = "STOP";
    Operation["Copy"] = "COPY";
    Operation["Deploy"] = "DEPLOY";
    Operation["Undeploy"] = "UNDEPLOY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    Operation["UnknownValue"] = "UNKNOWN_VALUE";
})(Operation || (exports.Operation = Operation = {}));
(function (Operation) {
    function getJsonObj(obj) {
        return obj;
    }
    Operation.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    Operation.getDeserializedJsonObj = getDeserializedJsonObj;
})(Operation || (exports.Operation = Operation = {}));
