// aidp-typescript-client/dist/lib/model/oracle-external-catalog-connection-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.OracleExternalCatalogConnectionPropertyKey = void 0;
/**
 * Oracle Database external catalog connection property keys.
**/
var OracleExternalCatalogConnectionPropertyKey;
(function (OracleExternalCatalogConnectionPropertyKey) {
    OracleExternalCatalogConnectionPropertyKey["OracleWalletContentBase64"] = "ORACLE_WALLET_CONTENT_BASE64";
    OracleExternalCatalogConnectionPropertyKey["OracleWalletPassword"] = "ORACLE_WALLET_PASSWORD";
    OracleExternalCatalogConnectionPropertyKey["OracleUsername"] = "ORACLE_USERNAME";
    OracleExternalCatalogConnectionPropertyKey["OraclePassword"] = "ORACLE_PASSWORD";
    OracleExternalCatalogConnectionPropertyKey["OracleHost"] = "ORACLE_HOST";
    OracleExternalCatalogConnectionPropertyKey["OraclePort"] = "ORACLE_PORT";
    OracleExternalCatalogConnectionPropertyKey["OracleDatabaseName"] = "ORACLE_DATABASE_NAME";
    OracleExternalCatalogConnectionPropertyKey["OracleDatabaseSid"] = "ORACLE_DATABASE_SID";
    OracleExternalCatalogConnectionPropertyKey["OracleSslEnabled"] = "ORACLE_SSL_ENABLED";
    OracleExternalCatalogConnectionPropertyKey["OracleStagingTenancyOcid"] = "ORACLE_STAGING_TENANCY_OCID";
    OracleExternalCatalogConnectionPropertyKey["OracleStagingRegion"] = "ORACLE_STAGING_REGION";
    OracleExternalCatalogConnectionPropertyKey["OracleStagingNamespace"] = "ORACLE_STAGING_NAMESPACE";
    OracleExternalCatalogConnectionPropertyKey["OracleStagingBucket"] = "ORACLE_STAGING_BUCKET";
    OracleExternalCatalogConnectionPropertyKey["OracleStagingFilePrefix"] = "ORACLE_STAGING_FILE_PREFIX";
})(OracleExternalCatalogConnectionPropertyKey || (exports.OracleExternalCatalogConnectionPropertyKey = OracleExternalCatalogConnectionPropertyKey = {}));
(function (OracleExternalCatalogConnectionPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    OracleExternalCatalogConnectionPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OracleExternalCatalogConnectionPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(OracleExternalCatalogConnectionPropertyKey || (exports.OracleExternalCatalogConnectionPropertyKey = OracleExternalCatalogConnectionPropertyKey = {}));
