// aidp-typescript-client/dist/lib/model/par-access-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParAccessType = void 0;
/**
 * The operation that can be performed on this resource.
**/
var ParAccessType;
(function (ParAccessType) {
    ParAccessType["ObjectRead"] = "OBJECT_READ";
    ParAccessType["ObjectWrite"] = "OBJECT_WRITE";
    ParAccessType["ObjectReadWrite"] = "OBJECT_READ_WRITE";
    ParAccessType["AnyObjectWrite"] = "ANY_OBJECT_WRITE";
    ParAccessType["AnyObjectRead"] = "ANY_OBJECT_READ";
    ParAccessType["AnyObjectReadWrite"] = "ANY_OBJECT_READ_WRITE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ParAccessType["UnknownValue"] = "UNKNOWN_VALUE";
})(ParAccessType || (exports.ParAccessType = ParAccessType = {}));
(function (ParAccessType) {
    function getJsonObj(obj) {
        return obj;
    }
    ParAccessType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ParAccessType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ParAccessType || (exports.ParAccessType = ParAccessType = {}));
