// aidp-typescript-client/dist/lib/model/permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PermissionSummary = void 0;
var PermissionSummary;
(function (PermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Use"] = "USE";
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["Admin"] = "ADMIN";
    })(GranteePermissions = PermissionSummary.GranteePermissions || (PermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(PermissionSummary || (exports.PermissionSummary = PermissionSummary = {}));
