// aidp-typescript-client/dist/lib/model/policy-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyType = void 0;
/**
 * Type of safety guardrail policy
**/
var PolicyType;
(function (PolicyType) {
    PolicyType["ContentModeration"] = "CONTENT_MODERATION";
    PolicyType["PromptAttacksPrevention"] = "PROMPT_ATTACKS_PREVENTION";
    PolicyType["PiiDetection"] = "PII_DETECTION";
    PolicyType["DeniedTopics"] = "DENIED_TOPICS";
    PolicyType["WordFilters"] = "WORD_FILTERS";
    PolicyType["ContextualGrounding"] = "CONTEXTUAL_GROUNDING";
    PolicyType["CustomPolicy"] = "CUSTOM_POLICY";
})(PolicyType || (exports.PolicyType = PolicyType = {}));
(function (PolicyType) {
    function getJsonObj(obj) {
        return obj;
    }
    PolicyType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PolicyType.getDeserializedJsonObj = getDeserializedJsonObj;
})(PolicyType || (exports.PolicyType = PolicyType = {}));
