// aidp-typescript-client/dist/lib/model/recent-resource-item.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecentResourceItem = void 0;
var RecentResourceItem;
(function (RecentResourceItem) {
    let ResourceType;
    (function (ResourceType) {
        ResourceType["Notebook"] = "NOTEBOOK";
        ResourceType["File"] = "FILE";
        ResourceType["Catalog"] = "CATALOG";
        ResourceType["Schema"] = "SCHEMA";
        ResourceType["Table"] = "TABLE";
        ResourceType["Volume"] = "VOLUME";
        ResourceType["Job"] = "JOB";
        ResourceType["JobRun"] = "JOB_RUN";
        ResourceType["LakeFlow"] = "LAKE_FLOW";
        ResourceType["AgentLakeFlow"] = "AGENT_LAKE_FLOW";
        ResourceType["Cluster"] = "CLUSTER";
        ResourceType["AiCompute"] = "AI_COMPUTE";
        ResourceType["Folder"] = "FOLDER";
        ResourceType["VolumeDir"] = "VOLUME_DIR";
        ResourceType["VolumeFile"] = "VOLUME_FILE";
        ResourceType["View"] = "VIEW";
        ResourceType["Share"] = "SHARE";
        ResourceType["Recipient"] = "RECIPIENT";
        ResourceType["Extractor"] = "EXTRACTOR";
        ResourceType["AgentFlow"] = "AGENT_FLOW";
    })(ResourceType = RecentResourceItem.ResourceType || (RecentResourceItem.ResourceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RecentResourceItem.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RecentResourceItem.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecentResourceItem || (exports.RecentResourceItem = RecentResourceItem = {}));
