// aidp-typescript-client/dist/lib/model/recipient-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecipientLifecycleState = void 0;
/**
 * The state of the Recipient
**/
var RecipientLifecycleState;
(function (RecipientLifecycleState) {
    RecipientLifecycleState["Active"] = "ACTIVE";
    RecipientLifecycleState["Inactive"] = "INACTIVE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RecipientLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(RecipientLifecycleState || (exports.RecipientLifecycleState = RecipientLifecycleState = {}));
(function (RecipientLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    RecipientLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RecipientLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecipientLifecycleState || (exports.RecipientLifecycleState = RecipientLifecycleState = {}));
