// aidp-typescript-client/dist/lib/model/recipient-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecipientPrivilege = void 0;
/**
 * The privilege for a recipient
**/
var RecipientPrivilege;
(function (RecipientPrivilege) {
    RecipientPrivilege["Admin"] = "ADMIN";
    RecipientPrivilege["Use"] = "USE";
    RecipientPrivilege["Read"] = "READ";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RecipientPrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(RecipientPrivilege || (exports.RecipientPrivilege = RecipientPrivilege = {}));
(function (RecipientPrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    RecipientPrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RecipientPrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecipientPrivilege || (exports.RecipientPrivilege = RecipientPrivilege = {}));
