// aidp-typescript-client/dist/lib/model/recipient-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecipientPermissionSummary = void 0;
var RecipientPermissionSummary;
(function (RecipientPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Admin"] = "ADMIN";
        GranteePermissions["Use"] = "USE";
        GranteePermissions["Read"] = "READ";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = RecipientPermissionSummary.GranteePermissions || (RecipientPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RecipientPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RecipientPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecipientPermissionSummary || (exports.RecipientPermissionSummary = RecipientPermissionSummary = {}));
