// aidp-typescript-client/dist/lib/model/recipient-property-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecipientPropertyKey = void 0;
/**
 * Recipient property keys.
**/
var RecipientPropertyKey;
(function (RecipientPropertyKey) {
    RecipientPropertyKey["ActivationLinkEmail"] = "ACTIVATION_LINK_EMAIL";
    RecipientPropertyKey["DeltaShareEndpoint"] = "DELTA_SHARE_ENDPOINT";
    RecipientPropertyKey["ActivationLink"] = "ACTIVATION_LINK";
    RecipientPropertyKey["ActivationTokenKey"] = "ACTIVATION_TOKEN_KEY";
    RecipientPropertyKey["ActivationTokenExpiryTime"] = "ACTIVATION_TOKEN_EXPIRY_TIME";
    RecipientPropertyKey["BearerTokenKey"] = "BEARER_TOKEN_KEY";
    RecipientPropertyKey["BearerTokenExpiryTime"] = "BEARER_TOKEN_EXPIRY_TIME";
})(RecipientPropertyKey || (exports.RecipientPropertyKey = RecipientPropertyKey = {}));
(function (RecipientPropertyKey) {
    function getJsonObj(obj) {
        return obj;
    }
    RecipientPropertyKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RecipientPropertyKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecipientPropertyKey || (exports.RecipientPropertyKey = RecipientPropertyKey = {}));
