// aidp-typescript-client/dist/lib/model/repair-history.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepairHistory = void 0;
const model = __importStar(require("../model"));
var RepairHistory;
(function (RepairHistory) {
    let Type;
    (function (Type) {
        Type["Original"] = "ORIGINAL";
        Type["Repair"] = "REPAIR";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = RepairHistory.Type || (RepairHistory.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'state': obj.state ?
                model.State.getJsonObj(obj.state) : undefined,
            'parameters': obj.parameters ?
                obj.parameters.map((item) => { return model.Parameter.getJsonObj(item); })
                : undefined,
            'lifecycleStates': obj.lifecycleStates ?
                obj.lifecycleStates.map((item) => { return model.LifecycleState.getJsonObj(item); })
                : undefined,
        });
        return jsonObj;
    }
    RepairHistory.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'state': obj.state ?
                model.State.getDeserializedJsonObj(obj.state) : undefined,
            'parameters': obj.parameters ?
                obj.parameters.map((item) => { return model.Parameter.getDeserializedJsonObj(item); })
                : undefined,
            'lifecycleStates': obj.lifecycleStates ?
                obj.lifecycleStates.map((item) => { return model.LifecycleState.getDeserializedJsonObj(item); })
                : undefined,
        });
        return jsonObj;
    }
    RepairHistory.getDeserializedJsonObj = getDeserializedJsonObj;
})(RepairHistory || (exports.RepairHistory = RepairHistory = {}));
