// aidp-typescript-client/dist/lib/model/reset-git-folder-state-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResetGitFolderStateDetails = void 0;
var ResetGitFolderStateDetails;
(function (ResetGitFolderStateDetails) {
    let Action;
    (function (Action) {
        Action["AbortMerge"] = "ABORT_MERGE";
        Action["AbortRebase"] = "ABORT_REBASE";
        Action["AbortPull"] = "ABORT_PULL";
        Action["Discard"] = "DISCARD";
    })(Action = ResetGitFolderStateDetails.Action || (ResetGitFolderStateDetails.Action = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ResetGitFolderStateDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ResetGitFolderStateDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ResetGitFolderStateDetails || (exports.ResetGitFolderStateDetails = ResetGitFolderStateDetails = {}));
