// aidp-typescript-client/dist/lib/model/role-assignee-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleAssigneeType = void 0;
/**
 * The entity type to which a role can be assigned to/revoked from. It is also used for role Admin type.
**/
var RoleAssigneeType;
(function (RoleAssigneeType) {
    RoleAssigneeType["User"] = "USER";
    RoleAssigneeType["Role"] = "ROLE";
    RoleAssigneeType["Group"] = "GROUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RoleAssigneeType["UnknownValue"] = "UNKNOWN_VALUE";
})(RoleAssigneeType || (exports.RoleAssigneeType = RoleAssigneeType = {}));
(function (RoleAssigneeType) {
    function getJsonObj(obj) {
        return obj;
    }
    RoleAssigneeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RoleAssigneeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(RoleAssigneeType || (exports.RoleAssigneeType = RoleAssigneeType = {}));
