// aidp-typescript-client/dist/lib/model/run-output-data.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RunOutputData = void 0;
var RunOutputData;
(function (RunOutputData) {
    let Type;
    (function (Type) {
        Type["TextPlain"] = "TEXT_PLAIN";
        Type["Notebook"] = "NOTEBOOK";
        Type["TextHtml"] = "TEXT_HTML";
        Type["ImagePng"] = "IMAGE_PNG";
        Type["ImageSvg"] = "IMAGE_SVG";
        Type["ImageJpeg"] = "IMAGE_JPEG";
        Type["ApplicationPdf"] = "APPLICATION_PDF";
        Type["ApplicationJson"] = "APPLICATION_JSON";
        Type["FilePath"] = "FILE_PATH";
        Type["NotebookPath"] = "NOTEBOOK_PATH";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = RunOutputData.Type || (RunOutputData.Type = {}));
    let Charset;
    (function (Charset) {
        Charset["Utf8"] = "UTF_8";
        Charset["Utf16"] = "UTF_16";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Charset["UnknownValue"] = "UNKNOWN_VALUE";
    })(Charset = RunOutputData.Charset || (RunOutputData.Charset = {}));
    let Compression;
    (function (Compression) {
        Compression["Gzip"] = "GZIP";
        Compression["Deflate"] = "DEFLATE";
        Compression["Compress"] = "COMPRESS";
        Compression["Bzip2"] = "BZIP2";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Compression["UnknownValue"] = "UNKNOWN_VALUE";
    })(Compression = RunOutputData.Compression || (RunOutputData.Compression = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RunOutputData.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RunOutputData.getDeserializedJsonObj = getDeserializedJsonObj;
})(RunOutputData || (exports.RunOutputData = RunOutputData = {}));
