// aidp-typescript-client/dist/lib/model/schema-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaPermissionSummary = void 0;
var SchemaPermissionSummary;
(function (SchemaPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Select"] = "SELECT";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["Write"] = "WRITE";
        GranteePermissions["CreateView"] = "CREATE_VIEW";
        GranteePermissions["CreateVolume"] = "CREATE_VOLUME";
        GranteePermissions["CreateTable"] = "CREATE_TABLE";
        GranteePermissions["CreateKnowledgeBase"] = "CREATE_KNOWLEDGE_BASE";
        GranteePermissions["Admin"] = "ADMIN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = SchemaPermissionSummary.GranteePermissions || (SchemaPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SchemaPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SchemaPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaPermissionSummary || (exports.SchemaPermissionSummary = SchemaPermissionSummary = {}));
