// aidp-typescript-client/dist/lib/model/schema-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaPrivilege = void 0;
/**
 * The privilege for a schema. This privilege could be inherited from the object higher up in hierarchy.
**/
var SchemaPrivilege;
(function (SchemaPrivilege) {
    SchemaPrivilege["Select"] = "SELECT";
    SchemaPrivilege["Manage"] = "MANAGE";
    SchemaPrivilege["Write"] = "WRITE";
    SchemaPrivilege["CreateView"] = "CREATE_VIEW";
    SchemaPrivilege["CreateVolume"] = "CREATE_VOLUME";
    SchemaPrivilege["CreateTable"] = "CREATE_TABLE";
    SchemaPrivilege["CreateKnowledgeBase"] = "CREATE_KNOWLEDGE_BASE";
    SchemaPrivilege["Admin"] = "ADMIN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SchemaPrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(SchemaPrivilege || (exports.SchemaPrivilege = SchemaPrivilege = {}));
(function (SchemaPrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    SchemaPrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SchemaPrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaPrivilege || (exports.SchemaPrivilege = SchemaPrivilege = {}));
