// aidp-typescript-client/dist/lib/model/schema-catalog-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaCatalogType = void 0;
/**
 * The type of catalog for the schema.
**/
var SchemaCatalogType;
(function (SchemaCatalogType) {
    SchemaCatalogType["Adw"] = "ADW";
    SchemaCatalogType["Alh"] = "ALH";
    SchemaCatalogType["Standard"] = "STANDARD";
    SchemaCatalogType["KafkaTopic"] = "KAFKA_TOPIC";
    SchemaCatalogType["Atp"] = "ATP";
    SchemaCatalogType["Oracle"] = "ORACLE";
    SchemaCatalogType["Exadata"] = "EXADATA";
    SchemaCatalogType["OracleAnalytics"] = "ORACLE_ANALYTICS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SchemaCatalogType["UnknownValue"] = "UNKNOWN_VALUE";
})(SchemaCatalogType || (exports.SchemaCatalogType = SchemaCatalogType = {}));
(function (SchemaCatalogType) {
    function getJsonObj(obj) {
        return obj;
    }
    SchemaCatalogType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SchemaCatalogType.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaCatalogType || (exports.SchemaCatalogType = SchemaCatalogType = {}));
