// aidp-typescript-client/dist/lib/model/schedule.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schedule = void 0;
var Schedule;
(function (Schedule) {
    let PauseStatus;
    (function (PauseStatus) {
        PauseStatus["Paused"] = "PAUSED";
        PauseStatus["Unpaused"] = "UNPAUSED";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        PauseStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(PauseStatus = Schedule.PauseStatus || (Schedule.PauseStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Schedule.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Schedule.getDeserializedJsonObj = getDeserializedJsonObj;
})(Schedule || (exports.Schedule = Schedule = {}));
