// aidp-typescript-client/dist/lib/model/schema-details-key.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaDetailsKey = void 0;
/**
 * Schema details key. This can be referred to when the related catalog is an external (Non-ADW) catalog.
**/
var SchemaDetailsKey;
(function (SchemaDetailsKey) {
    SchemaDetailsKey["Partitions"] = "PARTITIONS";
    SchemaDetailsKey["ReplicationFactor"] = "REPLICATION_FACTOR";
    SchemaDetailsKey["BootstrapServers"] = "BOOTSTRAP_SERVERS";
    SchemaDetailsKey["PartitionsDetails"] = "PARTITIONS_DETAILS";
})(SchemaDetailsKey || (exports.SchemaDetailsKey = SchemaDetailsKey = {}));
(function (SchemaDetailsKey) {
    function getJsonObj(obj) {
        return obj;
    }
    SchemaDetailsKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SchemaDetailsKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaDetailsKey || (exports.SchemaDetailsKey = SchemaDetailsKey = {}));
