// aidp-typescript-client/dist/lib/model/schema-privilege-for-manage.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaPrivilegeForManage = void 0;
/**
 * The privilege for a schema.
**/
var SchemaPrivilegeForManage;
(function (SchemaPrivilegeForManage) {
    SchemaPrivilegeForManage["Select"] = "SELECT";
    SchemaPrivilegeForManage["Write"] = "WRITE";
    SchemaPrivilegeForManage["CreateView"] = "CREATE_VIEW";
    SchemaPrivilegeForManage["CreateVolume"] = "CREATE_VOLUME";
    SchemaPrivilegeForManage["CreateTable"] = "CREATE_TABLE";
    SchemaPrivilegeForManage["Admin"] = "ADMIN";
    SchemaPrivilegeForManage["CreateKnowledgeBase"] = "CREATE_KNOWLEDGE_BASE";
})(SchemaPrivilegeForManage || (exports.SchemaPrivilegeForManage = SchemaPrivilegeForManage = {}));
(function (SchemaPrivilegeForManage) {
    function getJsonObj(obj) {
        return obj;
    }
    SchemaPrivilegeForManage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SchemaPrivilegeForManage.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaPrivilegeForManage || (exports.SchemaPrivilegeForManage = SchemaPrivilegeForManage = {}));
