// aidp-typescript-client/dist/lib/model/schema-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaLifecycleState = void 0;
/**
 * The current state of the schema.
**/
var SchemaLifecycleState;
(function (SchemaLifecycleState) {
    SchemaLifecycleState["Active"] = "ACTIVE";
    SchemaLifecycleState["Creating"] = "CREATING";
    SchemaLifecycleState["Deleting"] = "DELETING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SchemaLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SchemaLifecycleState || (exports.SchemaLifecycleState = SchemaLifecycleState = {}));
(function (SchemaLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SchemaLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SchemaLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaLifecycleState || (exports.SchemaLifecycleState = SchemaLifecycleState = {}));
