// aidp-typescript-client/dist/lib/model/search-audit-logs-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchAuditLogsDetails = void 0;
var SearchAuditLogsDetails;
(function (SearchAuditLogsDetails) {
    let SortBy;
    (function (SortBy) {
        SortBy["StartedBy"] = "startedBy";
        SortBy["Status"] = "status";
        SortBy["Operation"] = "operation";
        SortBy["ObjectType"] = "objectType";
        SortBy["ObjectName"] = "objectName";
        SortBy["Time"] = "time";
    })(SortBy = SearchAuditLogsDetails.SortBy || (SearchAuditLogsDetails.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = SearchAuditLogsDetails.SortOrder || (SearchAuditLogsDetails.SortOrder = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchAuditLogsDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchAuditLogsDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(SearchAuditLogsDetails || (exports.SearchAuditLogsDetails = SearchAuditLogsDetails = {}));
