// aidp-typescript-client/dist/lib/model/session.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Session = void 0;
const model = __importStar(require("../model"));
var Session;
(function (Session) {
    let Type;
    (function (Type) {
        Type["Notebook"] = "notebook";
        Type["File"] = "file";
        Type["Agentflow"] = "agentflow";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = Session.Type || (Session.Type = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Failed"] = "FAILED";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Session.LifecycleState || (Session.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'cluster_id': obj.clusterId,
            'kernel': obj.kernel ?
                model.Kernel.getJsonObj(obj.kernel) : undefined,
        });
        delete jsonObj.clusterId;
        return jsonObj;
    }
    Session.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'clusterId': obj["cluster_id"],
            'kernel': obj.kernel ?
                model.Kernel.getDeserializedJsonObj(obj.kernel) : undefined,
        });
        delete jsonObj["cluster_id"];
        return jsonObj;
    }
    Session.getDeserializedJsonObj = getDeserializedJsonObj;
})(Session || (exports.Session = Session = {}));
