// aidp-typescript-client/dist/lib/model/share-data-asset-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShareDataAssetType = void 0;
/**
 * The asset type of an operation.
**/
var ShareDataAssetType;
(function (ShareDataAssetType) {
    ShareDataAssetType["Schema"] = "SCHEMA";
    ShareDataAssetType["Table"] = "TABLE";
    ShareDataAssetType["View"] = "VIEW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ShareDataAssetType["UnknownValue"] = "UNKNOWN_VALUE";
})(ShareDataAssetType || (exports.ShareDataAssetType = ShareDataAssetType = {}));
(function (ShareDataAssetType) {
    function getJsonObj(obj) {
        return obj;
    }
    ShareDataAssetType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ShareDataAssetType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ShareDataAssetType || (exports.ShareDataAssetType = ShareDataAssetType = {}));
