// aidp-typescript-client/dist/lib/model/share-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharePermissionSummary = void 0;
var SharePermissionSummary;
(function (SharePermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Admin"] = "ADMIN";
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Use"] = "USE";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = SharePermissionSummary.GranteePermissions || (SharePermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SharePermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SharePermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SharePermissionSummary || (exports.SharePermissionSummary = SharePermissionSummary = {}));
