// aidp-typescript-client/dist/lib/model/share-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharePrivilege = void 0;
/**
 * The privilege for a share, which can be an inherited privilege coming from the object higher in the hierarchy.
**/
var SharePrivilege;
(function (SharePrivilege) {
    SharePrivilege["Admin"] = "ADMIN";
    SharePrivilege["Read"] = "READ";
    SharePrivilege["Use"] = "USE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SharePrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(SharePrivilege || (exports.SharePrivilege = SharePrivilege = {}));
(function (SharePrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    SharePrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SharePrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(SharePrivilege || (exports.SharePrivilege = SharePrivilege = {}));
