// aidp-typescript-client/dist/lib/model/state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.State = void 0;
var State;
(function (State) {
    let Status;
    (function (Status) {
        Status["Pending"] = "PENDING";
        Status["Queued"] = "QUEUED";
        Status["Running"] = "RUNNING";
        Status["Skipped"] = "SKIPPED";
        Status["InternalError"] = "INTERNAL_ERROR";
        Status["Blocked"] = "BLOCKED";
        Status["Success"] = "SUCCESS";
        Status["Failed"] = "FAILED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
        Status["UpstreamCanceled"] = "UPSTREAM_CANCELED";
        Status["UpstreamFailed"] = "UPSTREAM_FAILED";
        Status["Excluded"] = "EXCLUDED";
        Status["TimedOut"] = "TIMED_OUT";
        Status["PausedMaintenance"] = "PAUSED_MAINTENANCE";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = State.Status || (State.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    State.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    State.getDeserializedJsonObj = getDeserializedJsonObj;
})(State || (exports.State = State = {}));
