// aidp-typescript-client/dist/lib/model/summarize-metrics-data-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeMetricsDataDetails = void 0;
var SummarizeMetricsDataDetails;
(function (SummarizeMetricsDataDetails) {
    let AggregationType;
    (function (AggregationType) {
        AggregationType["Max"] = "MAX";
        AggregationType["Mean"] = "MEAN";
        AggregationType["Min"] = "MIN";
        AggregationType["Sum"] = "SUM";
        AggregationType["Rate"] = "RATE";
        AggregationType["P50"] = "P50";
        AggregationType["P80"] = "P80";
        AggregationType["P90"] = "P90";
        AggregationType["P95"] = "P95";
        AggregationType["P99"] = "P99";
        AggregationType["P999"] = "P99_9";
    })(AggregationType = SummarizeMetricsDataDetails.AggregationType || (SummarizeMetricsDataDetails.AggregationType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SummarizeMetricsDataDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SummarizeMetricsDataDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(SummarizeMetricsDataDetails || (exports.SummarizeMetricsDataDetails = SummarizeMetricsDataDetails = {}));
