// aidp-typescript-client/dist/lib/model/table-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TablePermissionSummary = void 0;
var TablePermissionSummary;
(function (TablePermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Select"] = "SELECT";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["Write"] = "WRITE";
        GranteePermissions["Insert"] = "INSERT";
        GranteePermissions["Update"] = "UPDATE";
        GranteePermissions["Delete"] = "DELETE";
        GranteePermissions["Alter"] = "ALTER";
        GranteePermissions["Admin"] = "ADMIN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = TablePermissionSummary.GranteePermissions || (TablePermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    TablePermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    TablePermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(TablePermissionSummary || (exports.TablePermissionSummary = TablePermissionSummary = {}));
