// aidp-typescript-client/dist/lib/model/table-privilege-for-manage.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TablePrivilegeForManage = void 0;
/**
 * The privilege for a table.
**/
var TablePrivilegeForManage;
(function (TablePrivilegeForManage) {
    TablePrivilegeForManage["Select"] = "SELECT";
    TablePrivilegeForManage["Insert"] = "INSERT";
    TablePrivilegeForManage["Update"] = "UPDATE";
    TablePrivilegeForManage["Delete"] = "DELETE";
    TablePrivilegeForManage["Alter"] = "ALTER";
    TablePrivilegeForManage["Admin"] = "ADMIN";
})(TablePrivilegeForManage || (exports.TablePrivilegeForManage = TablePrivilegeForManage = {}));
(function (TablePrivilegeForManage) {
    function getJsonObj(obj) {
        return obj;
    }
    TablePrivilegeForManage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TablePrivilegeForManage.getDeserializedJsonObj = getDeserializedJsonObj;
})(TablePrivilegeForManage || (exports.TablePrivilegeForManage = TablePrivilegeForManage = {}));
