// aidp-typescript-client/dist/lib/model/table-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TablePrivilege = void 0;
/**
 * The privilege for a table. Can be inherited privilege coming from object higher up in hierarchy.
**/
var TablePrivilege;
(function (TablePrivilege) {
    TablePrivilege["Select"] = "SELECT";
    TablePrivilege["Manage"] = "MANAGE";
    TablePrivilege["Write"] = "WRITE";
    TablePrivilege["Insert"] = "INSERT";
    TablePrivilege["Update"] = "UPDATE";
    TablePrivilege["Delete"] = "DELETE";
    TablePrivilege["Alter"] = "ALTER";
    TablePrivilege["Admin"] = "ADMIN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TablePrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(TablePrivilege || (exports.TablePrivilege = TablePrivilege = {}));
(function (TablePrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    TablePrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TablePrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(TablePrivilege || (exports.TablePrivilege = TablePrivilege = {}));
