// aidp-typescript-client/dist/lib/model/table-lifecycle-state.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableLifecycleState = void 0;
/**
 * The current state of the table.
**/
var TableLifecycleState;
(function (TableLifecycleState) {
    TableLifecycleState["Active"] = "ACTIVE";
    TableLifecycleState["Creating"] = "CREATING";
    TableLifecycleState["Deleting"] = "DELETING";
    TableLifecycleState["Updating"] = "UPDATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TableLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(TableLifecycleState || (exports.TableLifecycleState = TableLifecycleState = {}));
(function (TableLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    TableLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TableLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(TableLifecycleState || (exports.TableLifecycleState = TableLifecycleState = {}));
