// aidp-typescript-client/dist/lib/model/table-catalog-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableCatalogType = void 0;
/**
 * The type of catalog for the table.
**/
var TableCatalogType;
(function (TableCatalogType) {
    TableCatalogType["Adw"] = "ADW";
    TableCatalogType["Alh"] = "ALH";
    TableCatalogType["Standard"] = "STANDARD";
    TableCatalogType["Atp"] = "ATP";
    TableCatalogType["Oracle"] = "ORACLE";
    TableCatalogType["Exadata"] = "EXADATA";
    TableCatalogType["OracleAnalytics"] = "ORACLE_ANALYTICS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TableCatalogType["UnknownValue"] = "UNKNOWN_VALUE";
})(TableCatalogType || (exports.TableCatalogType = TableCatalogType = {}));
(function (TableCatalogType) {
    function getJsonObj(obj) {
        return obj;
    }
    TableCatalogType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TableCatalogType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TableCatalogType || (exports.TableCatalogType = TableCatalogType = {}));
