// aidp-typescript-client/dist/lib/model/task.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Task = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var Task;
(function (Task) {
    let RunIf;
    (function (RunIf) {
        RunIf["AllSuccess"] = "ALL_SUCCESS";
        RunIf["AllDone"] = "ALL_DONE";
        RunIf["NoneFailed"] = "NONE_FAILED";
        RunIf["AtLeastOneSuccess"] = "AT_LEAST_ONE_SUCCESS";
        RunIf["AllFailed"] = "ALL_FAILED";
        RunIf["AtLeastOneFailed"] = "AT_LEAST_ONE_FAILED";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        RunIf["UnknownValue"] = "UNKNOWN_VALUE";
    })(RunIf = Task.RunIf || (Task.RunIf = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'dependsOn': obj.dependsOn ?
                obj.dependsOn.map((item) => { return model.DependsOn.getJsonObj(item); })
                : undefined,
        });
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "IF_ELSE_TASK":
                    return model.IfElseTask.getJsonObj(jsonObj, true);
                case "JOB_TASK":
                    return model.JobTask.getJsonObj(jsonObj, true);
                case "JAR_TASK":
                    return model.JarTask.getJsonObj(jsonObj, true);
                case "PYTHON_TASK":
                    return model.PythonTask.getJsonObj(jsonObj, true);
                case "NOTEBOOK_TASK":
                    return model.NotebookTask.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    Task.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'dependsOn': obj.dependsOn ?
                obj.dependsOn.map((item) => { return model.DependsOn.getDeserializedJsonObj(item); })
                : undefined,
        });
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "IF_ELSE_TASK":
                    return model.IfElseTask.getDeserializedJsonObj(jsonObj, true);
                case "JOB_TASK":
                    return model.JobTask.getDeserializedJsonObj(jsonObj, true);
                case "JAR_TASK":
                    return model.JarTask.getDeserializedJsonObj(jsonObj, true);
                case "PYTHON_TASK":
                    return model.PythonTask.getDeserializedJsonObj(jsonObj, true);
                case "NOTEBOOK_TASK":
                    return model.NotebookTask.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    Task.getDeserializedJsonObj = getDeserializedJsonObj;
})(Task || (exports.Task = Task = {}));
