// aidp-typescript-client/dist/lib/model/tool-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolPermissionSummary = void 0;
var ToolPermissionSummary;
(function (ToolPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["Admin"] = "ADMIN";
    })(GranteePermissions = ToolPermissionSummary.GranteePermissions || (ToolPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ToolPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ToolPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ToolPermissionSummary || (exports.ToolPermissionSummary = ToolPermissionSummary = {}));
