// aidp-typescript-client/dist/lib/model/update-cluster-async-operation-status-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateClusterAsyncOperationStatusDetails = void 0;
var UpdateClusterAsyncOperationStatusDetails;
(function (UpdateClusterAsyncOperationStatusDetails) {
    let State;
    (function (State) {
        State["Accepted"] = "ACCEPTED";
        State["Creating"] = "CREATING";
        State["Active"] = "ACTIVE";
        State["Deleting"] = "DELETING";
        State["Deleted"] = "DELETED";
        State["Failed"] = "FAILED";
        State["Stopping"] = "STOPPING";
        State["Stopped"] = "STOPPED";
        State["Updating"] = "UPDATING";
        State["Restarting"] = "RESTARTING";
        State["Starting"] = "STARTING";
        State["NetworkConfigurationAttachInProgress"] = "NETWORK_CONFIGURATION_ATTACH_IN_PROGRESS";
        State["NetworkConfigurationAttachSuccessful"] = "NETWORK_CONFIGURATION_ATTACH_SUCCESSFUL";
        State["NetworkConfigurationAttachFailed"] = "NETWORK_CONFIGURATION_ATTACH_FAILED";
        State["NetworkConfigurationDetachInProgress"] = "NETWORK_CONFIGURATION_DETACH_IN_PROGRESS";
        State["NetworkConfigurationDetachSuccessful"] = "NETWORK_CONFIGURATION_DETACH_SUCCESSFUL";
        State["NetworkConfigurationDetachFailed"] = "NETWORK_CONFIGURATION_DETACH_FAILED";
    })(State = UpdateClusterAsyncOperationStatusDetails.State || (UpdateClusterAsyncOperationStatusDetails.State = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateClusterAsyncOperationStatusDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateClusterAsyncOperationStatusDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateClusterAsyncOperationStatusDetails || (exports.UpdateClusterAsyncOperationStatusDetails = UpdateClusterAsyncOperationStatusDetails = {}));
