// aidp-typescript-client/dist/lib/model/update-file-metadata-action-type.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateFileMetadataActionType = void 0;
/**
 * Metadata update action.
**/
var UpdateFileMetadataActionType;
(function (UpdateFileMetadataActionType) {
    UpdateFileMetadataActionType["Update"] = "UPDATE";
    UpdateFileMetadataActionType["Append"] = "APPEND";
    UpdateFileMetadataActionType["Replace"] = "REPLACE";
    UpdateFileMetadataActionType["Reset"] = "RESET";
})(UpdateFileMetadataActionType || (exports.UpdateFileMetadataActionType = UpdateFileMetadataActionType = {}));
(function (UpdateFileMetadataActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    UpdateFileMetadataActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    UpdateFileMetadataActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateFileMetadataActionType || (exports.UpdateFileMetadataActionType = UpdateFileMetadataActionType = {}));
