// aidp-typescript-client/dist/lib/model/update-content-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateContentDetails = void 0;
var UpdateContentDetails;
(function (UpdateContentDetails) {
    let Type;
    (function (Type) {
        Type["Notebook"] = "notebook";
        Type["File"] = "file";
        Type["Directory"] = "directory";
    })(Type = UpdateContentDetails.Type || (UpdateContentDetails.Type = {}));
    let Format;
    (function (Format) {
        Format["Null"] = "NULL";
        Format["Json"] = "json";
        Format["Text"] = "text";
        Format["Base64"] = "base64";
    })(Format = UpdateContentDetails.Format || (UpdateContentDetails.Format = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateContentDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateContentDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateContentDetails || (exports.UpdateContentDetails = UpdateContentDetails = {}));
