// aidp-typescript-client/dist/lib/model/update-view-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateViewDetails = void 0;
const model = __importStar(require("../model"));
var UpdateViewDetails;
(function (UpdateViewDetails) {
    let UpdateMode;
    (function (UpdateMode) {
        UpdateMode["RenameView"] = "RENAME_VIEW";
        UpdateMode["AddProperties"] = "ADD_PROPERTIES";
        UpdateMode["DropProperties"] = "DROP_PROPERTIES";
        UpdateMode["UpdateViewDescription"] = "UPDATE_VIEW_DESCRIPTION";
        UpdateMode["UpdateViewQuery"] = "UPDATE_VIEW_QUERY";
        UpdateMode["UpdateColumnDescription"] = "UPDATE_COLUMN_DESCRIPTION";
    })(UpdateMode = UpdateViewDetails.UpdateMode || (UpdateViewDetails.UpdateMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'viewUpdateDetails': obj.viewUpdateDetails ?
                model.ViewUpdateDetails.getJsonObj(obj.viewUpdateDetails) : undefined,
        });
        return jsonObj;
    }
    UpdateViewDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            'viewUpdateDetails': obj.viewUpdateDetails ?
                model.ViewUpdateDetails.getDeserializedJsonObj(obj.viewUpdateDetails) : undefined,
        });
        return jsonObj;
    }
    UpdateViewDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateViewDetails || (exports.UpdateViewDetails = UpdateViewDetails = {}));
