// aidp-typescript-client/dist/lib/model/update-workspace-async-operation-status-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateWorkspaceAsyncOperationStatusDetails = void 0;
var UpdateWorkspaceAsyncOperationStatusDetails;
(function (UpdateWorkspaceAsyncOperationStatusDetails) {
    let State;
    (function (State) {
        State["NetworkConfigurationCreated"] = "NETWORK_CONFIGURATION_CREATED";
        State["NetworkConfigurationUpdated"] = "NETWORK_CONFIGURATION_UPDATED";
        State["NetworkConfigurationDeleted"] = "NETWORK_CONFIGURATION_DELETED";
        State["NetworkConfigurationFailed"] = "NETWORK_CONFIGURATION_FAILED";
    })(State = UpdateWorkspaceAsyncOperationStatusDetails.State || (UpdateWorkspaceAsyncOperationStatusDetails.State = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateWorkspaceAsyncOperationStatusDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateWorkspaceAsyncOperationStatusDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateWorkspaceAsyncOperationStatusDetails || (exports.UpdateWorkspaceAsyncOperationStatusDetails = UpdateWorkspaceAsyncOperationStatusDetails = {}));
