// aidp-typescript-client/dist/lib/model/upload-file-with-par-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.UploadFileWithParDetails = void 0;
var UploadFileWithParDetails;
(function (UploadFileWithParDetails) {
    let Action;
    (function (Action) {
        Action["Create"] = "CREATE";
        Action["Refresh"] = "REFRESH";
        Action["Update"] = "UPDATE";
    })(Action = UploadFileWithParDetails.Action || (UploadFileWithParDetails.Action = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UploadFileWithParDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UploadFileWithParDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UploadFileWithParDetails || (exports.UploadFileWithParDetails = UploadFileWithParDetails = {}));
