// aidp-typescript-client/dist/lib/model/view-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewPrivilege = void 0;
/**
 * The privilege for a View which could be an inherited privilege coming from object up in hierarchy.
**/
var ViewPrivilege;
(function (ViewPrivilege) {
    ViewPrivilege["Select"] = "SELECT";
    ViewPrivilege["Manage"] = "MANAGE";
    ViewPrivilege["Write"] = "WRITE";
    ViewPrivilege["Insert"] = "INSERT";
    ViewPrivilege["Update"] = "UPDATE";
    ViewPrivilege["Delete"] = "DELETE";
    ViewPrivilege["Alter"] = "ALTER";
    ViewPrivilege["Admin"] = "ADMIN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ViewPrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(ViewPrivilege || (exports.ViewPrivilege = ViewPrivilege = {}));
(function (ViewPrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    ViewPrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ViewPrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(ViewPrivilege || (exports.ViewPrivilege = ViewPrivilege = {}));
