// aidp-typescript-client/dist/lib/model/volume-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeSummary = void 0;
var VolumeSummary;
(function (VolumeSummary) {
    let VolumeType;
    (function (VolumeType) {
        VolumeType["Managed"] = "MANAGED";
        VolumeType["External"] = "EXTERNAL";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        VolumeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(VolumeType = VolumeSummary.VolumeType || (VolumeSummary.VolumeType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(VolumeSummary || (exports.VolumeSummary = VolumeSummary = {}));
