// aidp-typescript-client/dist/lib/model/volume.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Volume = void 0;
var Volume;
(function (Volume) {
    let VolumeType;
    (function (VolumeType) {
        VolumeType["Managed"] = "MANAGED";
        VolumeType["External"] = "EXTERNAL";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        VolumeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(VolumeType = Volume.VolumeType || (Volume.VolumeType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleted"] = "DELETED";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Volume.LifecycleState || (Volume.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Volume.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Volume.getDeserializedJsonObj = getDeserializedJsonObj;
})(Volume || (exports.Volume = Volume = {}));
