// aidp-typescript-client/dist/lib/model/volume-file-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeFileSummary = void 0;
var VolumeFileSummary;
(function (VolumeFileSummary) {
    let Type;
    (function (Type) {
        Type["File"] = "FILE";
        Type["Folder"] = "FOLDER";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = VolumeFileSummary.Type || (VolumeFileSummary.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeFileSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeFileSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(VolumeFileSummary || (exports.VolumeFileSummary = VolumeFileSummary = {}));
