// aidp-typescript-client/dist/lib/model/volume-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumePermissionSummary = void 0;
var VolumePermissionSummary;
(function (VolumePermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Write"] = "WRITE";
        GranteePermissions["Admin"] = "ADMIN";
        GranteePermissions["Select"] = "SELECT";
        GranteePermissions["Manage"] = "MANAGE";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = VolumePermissionSummary.GranteePermissions || (VolumePermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumePermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumePermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(VolumePermissionSummary || (exports.VolumePermissionSummary = VolumePermissionSummary = {}));
