// aidp-typescript-client/dist/lib/model/workspace-object-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceObjectPermissionSummary = void 0;
var WorkspaceObjectPermissionSummary;
(function (WorkspaceObjectPermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["Read"] = "READ";
        GranteePermissions["Use"] = "USE";
        GranteePermissions["Manage"] = "MANAGE";
        GranteePermissions["Admin"] = "ADMIN";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = WorkspaceObjectPermissionSummary.GranteePermissions || (WorkspaceObjectPermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspaceObjectPermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspaceObjectPermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkspaceObjectPermissionSummary || (exports.WorkspaceObjectPermissionSummary = WorkspaceObjectPermissionSummary = {}));
