// aidp-typescript-client/dist/lib/model/workspace-object-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceObjectPrivilege = void 0;
/**
 * The privilege for a workspace object.
**/
var WorkspaceObjectPrivilege;
(function (WorkspaceObjectPrivilege) {
    WorkspaceObjectPrivilege["Read"] = "READ";
    WorkspaceObjectPrivilege["Use"] = "USE";
    WorkspaceObjectPrivilege["Manage"] = "MANAGE";
    WorkspaceObjectPrivilege["Admin"] = "ADMIN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkspaceObjectPrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkspaceObjectPrivilege || (exports.WorkspaceObjectPrivilege = WorkspaceObjectPrivilege = {}));
(function (WorkspaceObjectPrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkspaceObjectPrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkspaceObjectPrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkspaceObjectPrivilege || (exports.WorkspaceObjectPrivilege = WorkspaceObjectPrivilege = {}));
