// aidp-typescript-client/dist/lib/model/workspace-permission-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspacePermissionSummary = void 0;
var WorkspacePermissionSummary;
(function (WorkspacePermissionSummary) {
    let GranteePermissions;
    (function (GranteePermissions) {
        GranteePermissions["User"] = "USER";
        GranteePermissions["PrivilegedUser"] = "PRIVILEGED_USER";
        GranteePermissions["Administrator"] = "ADMINISTRATOR";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        GranteePermissions["UnknownValue"] = "UNKNOWN_VALUE";
    })(GranteePermissions = WorkspacePermissionSummary.GranteePermissions || (WorkspacePermissionSummary.GranteePermissions = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspacePermissionSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspacePermissionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkspacePermissionSummary || (exports.WorkspacePermissionSummary = WorkspacePermissionSummary = {}));
