// aidp-typescript-client/dist/lib/model/workspace-privilege.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspacePrivilege = void 0;
/**
 * The privilege for a workspace.
**/
var WorkspacePrivilege;
(function (WorkspacePrivilege) {
    WorkspacePrivilege["User"] = "USER";
    WorkspacePrivilege["PrivilegedUser"] = "PRIVILEGED_USER";
    WorkspacePrivilege["Administrator"] = "ADMINISTRATOR";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkspacePrivilege["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkspacePrivilege || (exports.WorkspacePrivilege = WorkspacePrivilege = {}));
(function (WorkspacePrivilege) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkspacePrivilege.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkspacePrivilege.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkspacePrivilege || (exports.WorkspacePrivilege = WorkspacePrivilege = {}));
