// aidp-typescript-client/dist/lib/model/workspace-object-details.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceObjectDetails = void 0;
var WorkspaceObjectDetails;
(function (WorkspaceObjectDetails) {
    let Type;
    (function (Type) {
        Type["Notebook"] = "NOTEBOOK";
        Type["Job"] = "JOB";
        Type["LakeFlow"] = "LAKE_FLOW";
        Type["AgentLakeFlow"] = "AGENT_LAKE_FLOW";
        Type["GeneratedArtifact"] = "GENERATED_ARTIFACT";
        Type["Library"] = "LIBRARY";
        Type["File"] = "FILE";
        Type["Folder"] = "FOLDER";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = WorkspaceObjectDetails.Type || (WorkspaceObjectDetails.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspaceObjectDetails.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspaceObjectDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkspaceObjectDetails || (exports.WorkspaceObjectDetails = WorkspaceObjectDetails = {}));
