// aidp-typescript-client/dist/lib/model/workspace-summary.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceSummary = void 0;
var WorkspaceSummary;
(function (WorkspaceSummary) {
    let Type;
    (function (Type) {
        Type["Default"] = "DEFAULT";
        Type["User"] = "USER";
        /**
        * This value is used if a service returns a value for this enum that is not recognized by this
        * version of the SDK.
        */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = WorkspaceSummary.Type || (WorkspaceSummary.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspaceSummary.getJsonObj = getJsonObj;
    ;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkspaceSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkspaceSummary || (exports.WorkspaceSummary = WorkspaceSummary = {}));
