// aidp-typescript-client/dist/lib/request/index.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkspaceObjectsRequest = exports.ListWorkspaceObjectPermissionsRequest = exports.ListWorkspacesRequest = exports.ListWorkspacePermissionsRequest = exports.ListCreateWorkspacePermissionsRequest = exports.ListTaskRunsRequest = exports.ListRecentJobRunsRequest = exports.ListJobsRequest = exports.ListJobRunsRequest = exports.ListJobPermissionsRequest = exports.ListVolumesRequest = exports.ListVolumePermissionsRequest = exports.ListFilesRequest = exports.ListUserSettingsRequest = exports.ListViewsRequest = exports.ListViewPermissionsRequest = exports.ListTablesRequest = exports.ListTablePermissionsRequest = exports.ListSchemasRequest = exports.ListSchemaPermissionsRequest = exports.ListRolesRequest = exports.ListRolePermissionsRequest = exports.GetContentRequest = exports.ListGitDiffsRequest = exports.ListGitBranchesRequest = exports.ListSharesRequest = exports.ListShareRecipientsRequest = exports.ListSharePermissionsRequest = exports.ListShareDataAssetsRequest = exports.ListRecipientsRequest = exports.ListRecipientSharesRequest = exports.ListRecipientPermissionsRequest = exports.ListCredentialsRequest = exports.ListClustersRequest = exports.ListClusterPermissionsRequest = exports.ListClusterLibrariesRequest = exports.ListCatalogsRequest = exports.ListCatalogPermissionsRequest = exports.ListAsyncOperationsRequest = void 0;
const ListAsyncOperationsRequest = __importStar(require("./list-async-operations-request"));
exports.ListAsyncOperationsRequest = ListAsyncOperationsRequest.ListAsyncOperationsRequest;
const ListCatalogPermissionsRequest = __importStar(require("./list-catalog-permissions-request"));
exports.ListCatalogPermissionsRequest = ListCatalogPermissionsRequest.ListCatalogPermissionsRequest;
const ListCatalogsRequest = __importStar(require("./list-catalogs-request"));
exports.ListCatalogsRequest = ListCatalogsRequest.ListCatalogsRequest;
const ListClusterLibrariesRequest = __importStar(require("./list-cluster-libraries-request"));
exports.ListClusterLibrariesRequest = ListClusterLibrariesRequest.ListClusterLibrariesRequest;
const ListClusterPermissionsRequest = __importStar(require("./list-cluster-permissions-request"));
exports.ListClusterPermissionsRequest = ListClusterPermissionsRequest.ListClusterPermissionsRequest;
const ListClustersRequest = __importStar(require("./list-clusters-request"));
exports.ListClustersRequest = ListClustersRequest.ListClustersRequest;
const ListCredentialsRequest = __importStar(require("./list-credentials-request"));
exports.ListCredentialsRequest = ListCredentialsRequest.ListCredentialsRequest;
const ListRecipientPermissionsRequest = __importStar(require("./list-recipient-permissions-request"));
exports.ListRecipientPermissionsRequest = ListRecipientPermissionsRequest.ListRecipientPermissionsRequest;
const ListRecipientSharesRequest = __importStar(require("./list-recipient-shares-request"));
exports.ListRecipientSharesRequest = ListRecipientSharesRequest.ListRecipientSharesRequest;
const ListRecipientsRequest = __importStar(require("./list-recipients-request"));
exports.ListRecipientsRequest = ListRecipientsRequest.ListRecipientsRequest;
const ListShareDataAssetsRequest = __importStar(require("./list-share-data-assets-request"));
exports.ListShareDataAssetsRequest = ListShareDataAssetsRequest.ListShareDataAssetsRequest;
const ListSharePermissionsRequest = __importStar(require("./list-share-permissions-request"));
exports.ListSharePermissionsRequest = ListSharePermissionsRequest.ListSharePermissionsRequest;
const ListShareRecipientsRequest = __importStar(require("./list-share-recipients-request"));
exports.ListShareRecipientsRequest = ListShareRecipientsRequest.ListShareRecipientsRequest;
const ListSharesRequest = __importStar(require("./list-shares-request"));
exports.ListSharesRequest = ListSharesRequest.ListSharesRequest;
const ListGitBranchesRequest = __importStar(require("./list-git-branches-request"));
exports.ListGitBranchesRequest = ListGitBranchesRequest.ListGitBranchesRequest;
const ListGitDiffsRequest = __importStar(require("./list-git-diffs-request"));
exports.ListGitDiffsRequest = ListGitDiffsRequest.ListGitDiffsRequest;
const GetContentRequest = __importStar(require("./get-content-request"));
exports.GetContentRequest = GetContentRequest.GetContentRequest;
const ListRolePermissionsRequest = __importStar(require("./list-role-permissions-request"));
exports.ListRolePermissionsRequest = ListRolePermissionsRequest.ListRolePermissionsRequest;
const ListRolesRequest = __importStar(require("./list-roles-request"));
exports.ListRolesRequest = ListRolesRequest.ListRolesRequest;
const ListSchemaPermissionsRequest = __importStar(require("./list-schema-permissions-request"));
exports.ListSchemaPermissionsRequest = ListSchemaPermissionsRequest.ListSchemaPermissionsRequest;
const ListSchemasRequest = __importStar(require("./list-schemas-request"));
exports.ListSchemasRequest = ListSchemasRequest.ListSchemasRequest;
const ListTablePermissionsRequest = __importStar(require("./list-table-permissions-request"));
exports.ListTablePermissionsRequest = ListTablePermissionsRequest.ListTablePermissionsRequest;
const ListTablesRequest = __importStar(require("./list-tables-request"));
exports.ListTablesRequest = ListTablesRequest.ListTablesRequest;
const ListViewPermissionsRequest = __importStar(require("./list-view-permissions-request"));
exports.ListViewPermissionsRequest = ListViewPermissionsRequest.ListViewPermissionsRequest;
const ListViewsRequest = __importStar(require("./list-views-request"));
exports.ListViewsRequest = ListViewsRequest.ListViewsRequest;
const ListUserSettingsRequest = __importStar(require("./list-user-settings-request"));
exports.ListUserSettingsRequest = ListUserSettingsRequest.ListUserSettingsRequest;
const ListFilesRequest = __importStar(require("./list-files-request"));
exports.ListFilesRequest = ListFilesRequest.ListFilesRequest;
const ListVolumePermissionsRequest = __importStar(require("./list-volume-permissions-request"));
exports.ListVolumePermissionsRequest = ListVolumePermissionsRequest.ListVolumePermissionsRequest;
const ListVolumesRequest = __importStar(require("./list-volumes-request"));
exports.ListVolumesRequest = ListVolumesRequest.ListVolumesRequest;
const ListJobPermissionsRequest = __importStar(require("./list-job-permissions-request"));
exports.ListJobPermissionsRequest = ListJobPermissionsRequest.ListJobPermissionsRequest;
const ListJobRunsRequest = __importStar(require("./list-job-runs-request"));
exports.ListJobRunsRequest = ListJobRunsRequest.ListJobRunsRequest;
const ListJobsRequest = __importStar(require("./list-jobs-request"));
exports.ListJobsRequest = ListJobsRequest.ListJobsRequest;
const ListRecentJobRunsRequest = __importStar(require("./list-recent-job-runs-request"));
exports.ListRecentJobRunsRequest = ListRecentJobRunsRequest.ListRecentJobRunsRequest;
const ListTaskRunsRequest = __importStar(require("./list-task-runs-request"));
exports.ListTaskRunsRequest = ListTaskRunsRequest.ListTaskRunsRequest;
const ListCreateWorkspacePermissionsRequest = __importStar(require("./list-create-workspace-permissions-request"));
exports.ListCreateWorkspacePermissionsRequest = ListCreateWorkspacePermissionsRequest.ListCreateWorkspacePermissionsRequest;
const ListWorkspacePermissionsRequest = __importStar(require("./list-workspace-permissions-request"));
exports.ListWorkspacePermissionsRequest = ListWorkspacePermissionsRequest.ListWorkspacePermissionsRequest;
const ListWorkspacesRequest = __importStar(require("./list-workspaces-request"));
exports.ListWorkspacesRequest = ListWorkspacesRequest.ListWorkspacesRequest;
const ListWorkspaceObjectPermissionsRequest = __importStar(require("./list-workspace-object-permissions-request"));
exports.ListWorkspaceObjectPermissionsRequest = ListWorkspaceObjectPermissionsRequest.ListWorkspaceObjectPermissionsRequest;
const ListWorkspaceObjectsRequest = __importStar(require("./list-workspace-objects-request"));
exports.ListWorkspaceObjectsRequest = ListWorkspaceObjectsRequest.ListWorkspaceObjectsRequest;
