// aidp-typescript-client/dist/lib/request/list-job-runs-request.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListJobRunsRequest = void 0;
var ListJobRunsRequest;
(function (ListJobRunsRequest) {
    let Status;
    (function (Status) {
        Status["Pending"] = "PENDING";
        Status["Running"] = "RUNNING";
        Status["Skipped"] = "SKIPPED";
        Status["InternalError"] = "INTERNAL_ERROR";
        Status["Blocked"] = "BLOCKED";
        Status["Success"] = "SUCCESS";
        Status["Failed"] = "FAILED";
        Status["Canceled"] = "CANCELED";
        Status["UpstreamCanceled"] = "UPSTREAM_CANCELED";
        Status["UpstreamFailed"] = "UPSTREAM_FAILED";
        Status["Excluded"] = "EXCLUDED";
    })(Status = ListJobRunsRequest.Status || (ListJobRunsRequest.Status = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
        SortBy["Status"] = "status";
    })(SortBy = ListJobRunsRequest.SortBy || (ListJobRunsRequest.SortBy = {}));
})(ListJobRunsRequest || (exports.ListJobRunsRequest = ListJobRunsRequest = {}));
