// aidp-typescript-client/dist/lib/request/list-git-diffs-request.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListGitDiffsRequest = void 0;
var ListGitDiffsRequest;
(function (ListGitDiffsRequest) {
    let CompareTo;
    (function (CompareTo) {
        CompareTo["Head"] = "HEAD";
        CompareTo["Upstream"] = "UPSTREAM";
        CompareTo["Index"] = "INDEX";
        CompareTo["Worktree"] = "WORKTREE";
        CompareTo["MergeBase"] = "MERGE_BASE";
    })(CompareTo = ListGitDiffsRequest.CompareTo || (ListGitDiffsRequest.CompareTo = {}));
    let Filter;
    (function (Filter) {
        Filter["All"] = "ALL";
        Filter["DiffOnly"] = "DIFF_ONLY";
        Filter["ConflictsOnly"] = "CONFLICTS_ONLY";
    })(Filter = ListGitDiffsRequest.Filter || (ListGitDiffsRequest.Filter = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListGitDiffsRequest.SortBy || (ListGitDiffsRequest.SortBy = {}));
})(ListGitDiffsRequest || (exports.ListGitDiffsRequest = ListGitDiffsRequest = {}));
