// aidp-typescript-client/dist/lib/request/list-task-runs-request.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTaskRunsRequest = void 0;
var ListTaskRunsRequest;
(function (ListTaskRunsRequest) {
    let Status;
    (function (Status) {
        Status["Pending"] = "PENDING";
        Status["Running"] = "RUNNING";
        Status["Skipped"] = "SKIPPED";
        Status["InternalError"] = "INTERNAL_ERROR";
        Status["Blocked"] = "BLOCKED";
        Status["Success"] = "SUCCESS";
        Status["Failed"] = "FAILED";
        Status["Canceled"] = "CANCELED";
        Status["UpstreamCanceled"] = "UPSTREAM_CANCELED";
        Status["UpstreamFailed"] = "UPSTREAM_FAILED";
        Status["Excluded"] = "EXCLUDED";
    })(Status = ListTaskRunsRequest.Status || (ListTaskRunsRequest.Status = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
        SortBy["Status"] = "status";
    })(SortBy = ListTaskRunsRequest.SortBy || (ListTaskRunsRequest.SortBy = {}));
})(ListTaskRunsRequest || (exports.ListTaskRunsRequest = ListTaskRunsRequest = {}));
