// aidp-typescript-client/dist/lib/request/list-catalogs-request.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListCatalogsRequest = void 0;
var ListCatalogsRequest;
(function (ListCatalogsRequest) {
    let CatalogState;
    (function (CatalogState) {
        CatalogState["Active"] = "ACTIVE";
        CatalogState["Creating"] = "CREATING";
        CatalogState["Deleting"] = "DELETING";
    })(CatalogState = ListCatalogsRequest.CatalogState || (ListCatalogsRequest.CatalogState = {}));
    let CatalogType;
    (function (CatalogType) {
        CatalogType["Internal"] = "INTERNAL";
        CatalogType["External"] = "EXTERNAL";
    })(CatalogType = ListCatalogsRequest.CatalogType || (ListCatalogsRequest.CatalogType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListCatalogsRequest.SortBy || (ListCatalogsRequest.SortBy = {}));
})(ListCatalogsRequest || (exports.ListCatalogsRequest = ListCatalogsRequest = {}));
