// aidp-cli/dist/args.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GLOBAL_BOOLEAN_OPTIONS = exports.GLOBAL_VALUE_OPTIONS = void 0;
exports.parseGlobalOptions = parseGlobalOptions;
exports.consumeLeadingGlobalOptions = consumeLeadingGlobalOptions;
exports.applyGlobalOption = applyGlobalOption;
exports.splitInlineOption = splitInlineOption;
exports.optionValue = optionValue;
const config_1 = require("./config");
const errors_1 = require("./errors");
exports.GLOBAL_VALUE_OPTIONS = {
    "--config-file": "configFile",
    "--profile": "profile",
    "-p": "profile",
    "--auth": "auth",
    "--region": "region",
    "--endpoint": "endpoint",
    "--environment-prefix": "environmentPrefix",
    "--environmentprefix": "environmentPrefix",
    "--environment-domain": "environmentDomain",
    "--environment-host": "environmentHost",
    "--timeout": "timeout",
    "--instance-id": "instanceId"
};
exports.GLOBAL_BOOLEAN_OPTIONS = {
    "--debug": "debug"
};
function parseGlobalOptions(argv) {
    let globals = (0, config_1.defaultGlobalOptions)();
    const remaining = [];
    let helpRequested = false;
    let versionRequested = false;
    let commandSeen = false;
    let index = 0;
    while (index < argv.length) {
        const token = argv[index];
        if (token === "-h" || token === "--help") {
            helpRequested = true;
            index += 1;
            continue;
        }
        if (token === "-v" || token === "--version") {
            versionRequested = true;
            index += 1;
            continue;
        }
        if (!commandSeen && token in exports.GLOBAL_BOOLEAN_OPTIONS) {
            globals = { ...globals, [exports.GLOBAL_BOOLEAN_OPTIONS[token]]: true };
            index += 1;
            continue;
        }
        const [optionName, inlineValue] = splitInlineOption(token);
        if (!commandSeen && optionName in exports.GLOBAL_VALUE_OPTIONS) {
            const [value, nextIndex] = optionValue(argv, index, optionName, inlineValue);
            globals = applyGlobalOption(globals, optionName, value);
            index = nextIndex;
            continue;
        }
        if (!commandSeen && token.startsWith("-")) {
            throw new errors_1.CliError(`Unknown option '${optionName}'.`);
        }
        remaining.push(token);
        if (!token.startsWith("-")) {
            commandSeen = true;
        }
        index += 1;
    }
    return { args: remaining, globals, helpRequested, versionRequested };
}
function consumeLeadingGlobalOptions(tokens, globals) {
    let values = { ...globals };
    let index = 0;
    while (index < tokens.length) {
        const token = tokens[index];
        if (token in exports.GLOBAL_BOOLEAN_OPTIONS) {
            values = { ...values, [exports.GLOBAL_BOOLEAN_OPTIONS[token]]: true };
            index += 1;
            continue;
        }
        const [optionName, inlineValue] = splitInlineOption(token);
        if (optionName in exports.GLOBAL_VALUE_OPTIONS) {
            const [value, nextIndex] = optionValue(tokens, index, optionName, inlineValue);
            values = applyGlobalOption(values, optionName, value);
            index = nextIndex;
            continue;
        }
        break;
    }
    return [values, tokens.slice(index)];
}
function applyGlobalOption(globals, optionName, value) {
    const key = exports.GLOBAL_VALUE_OPTIONS[optionName];
    if (!key) {
        throw new errors_1.CliError(`Unknown global option ${optionName}.`);
    }
    if (key === "auth") {
        (0, config_1.assertAuthMode)(value);
        return { ...globals, auth: value };
    }
    if (key === "timeout") {
        return { ...globals, timeout: parseTimeout(value) };
    }
    return { ...globals, [key]: value };
}
function splitInlineOption(token) {
    const separatorIndex = token.indexOf("=");
    if (separatorIndex < 0) {
        return [token, undefined];
    }
    return [token.slice(0, separatorIndex), token.slice(separatorIndex + 1)];
}
function optionValue(argv, optionIndex, optionName, inlineValue) {
    if (inlineValue !== undefined) {
        return [inlineValue, optionIndex + 1];
    }
    const value = argv[optionIndex + 1];
    if (value === undefined) {
        throw new errors_1.CliError(`${optionName} requires a value.`);
    }
    return [value, optionIndex + 2];
}
function parseTimeout(value) {
    const timeout = Number(value);
    if (!Number.isFinite(timeout) || timeout <= 0) {
        throw new errors_1.CliError("--timeout must be a positive number of seconds.");
    }
    return timeout;
}
