// aidp-cli/dist/cli.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.main = main;
exports.shouldSuppressSdkRetryWarning = shouldSuppressSdkRetryWarning;
exports.formatCliError = formatCliError;
exports.safeHeaders = safeHeaders;
exports.bodyDebugSummary = bodyDebugSummary;
const fs_1 = require("fs");
const path_1 = require("path");
const util_1 = require("util");
const args_1 = require("./args");
const config_1 = require("./config");
const discovery_1 = require("./discovery");
const errors_1 = require("./errors");
const help_1 = require("./help");
const commandArgs_1 = require("./commandArgs");
const manifest_1 = require("./manifest");
const names_1 = require("./names");
const output_1 = require("./output");
const sdk_1 = require("./sdk");
async function main(argv = process.argv.slice(2)) {
    try {
        const manifest = (0, manifest_1.loadCommandManifest)();
        const parsed = (0, args_1.parseGlobalOptions)(argv);
        if (parsed.versionRequested) {
            printVersion();
            return 0;
        }
        if (parsed.args.length === 0) {
            console.log((0, help_1.rootHelp)(manifest));
            return 0;
        }
        return await dispatch(manifest, parsed.args, parsed.globals, parsed.helpRequested);
    }
    catch (error) {
        if (isServiceError(error)) {
            (0, output_1.printErrorResponse)(serviceErrorPayload(error));
            return serviceErrorStatus(error);
        }
        const detail = formatCliError(error);
        console.error(`aidp: error: ${detail}`);
        return error instanceof errors_1.CliError ? error.exitCode : 1;
    }
}
async function dispatch(manifest, args, globals, helpRequested) {
    const command = args[0];
    const tail = args.slice(1);
    if (command === "help") {
        return handleHelp(manifest, tail);
    }
    if (command === "version") {
        printVersion();
        return 0;
    }
    if (command === "command-groups") {
        console.log(helpRequested ? (0, help_1.commandGroupsHelp)() : (0, help_1.commandGroupsOutput)(manifest));
        return 0;
    }
    if (command === "search") {
        if (helpRequested || tail.length === 0) {
            console.log((0, help_1.searchHelp)());
        }
        else {
            handleSearch(manifest, tail);
        }
        return 0;
    }
    if (command === "configure") {
        if (helpRequested || tail.length === 0) {
            console.log((0, help_1.configureHelp)());
        }
        else {
            handleConfigure(tail);
        }
        return 0;
    }
    const group = (0, discovery_1.findCommandGroup)(manifest, command);
    if (!group) {
        throw unknownCommandGroup(manifest, command);
    }
    const [updatedGlobals, remainingTail] = (0, args_1.consumeLeadingGlobalOptions)(tail, globals);
    if (remainingTail.length === 0) {
        console.log((0, help_1.groupHelp)(group));
        return 0;
    }
    const commandName = remainingTail[0];
    const commandDefinition = (0, discovery_1.findCommand)(group, commandName);
    if (!commandDefinition) {
        throw unknownCommandName(group, commandName);
    }
    if (helpRequested) {
        console.log((0, help_1.commandHelp)(group, commandDefinition));
        return 0;
    }
    return await handleCommand(group, commandDefinition, remainingTail.slice(1), updatedGlobals);
}
function handleHelp(manifest, args) {
    if (args.length === 0) {
        console.log((0, help_1.rootHelp)(manifest));
        return 0;
    }
    if (args[0] === "command-groups") {
        console.log((0, help_1.commandGroupsHelp)());
        return 0;
    }
    if (args[0] === "search") {
        console.log((0, help_1.searchHelp)());
        return 0;
    }
    if (args[0] === "configure") {
        console.log((0, help_1.configureHelp)());
        return 0;
    }
    if (args[0] === "version") {
        printVersion();
        return 0;
    }
    const group = (0, discovery_1.findCommandGroup)(manifest, args[0]);
    if (!group) {
        throw unknownCommandGroup(manifest, args[0]);
    }
    if (args.length === 1) {
        console.log((0, help_1.groupHelp)(group));
        return 0;
    }
    const command = (0, discovery_1.findCommand)(group, args[1]);
    if (!command) {
        throw unknownCommandName(group, args[1]);
    }
    console.log((0, help_1.commandHelp)(group, command));
    return 0;
}
async function handleCommand(group, command, tokens, globals) {
    var _a;
    let invocation;
    try {
        invocation = (0, commandArgs_1.parseCommandOptions)(group, command, tokens, globals);
    }
    catch (error) {
        if (error instanceof errors_1.CliError) {
            throw (0, commandArgs_1.withCommandUsageHint)(group, command, error);
        }
        throw error;
    }
    const authProvider = await (0, config_1.buildAuthenticationDetailsProvider)(invocation.globals);
    const endpoint = (0, config_1.resolveEndpoint)(invocation.globals, authProvider);
    const clientClass = (_a = group.clientClass) !== null && _a !== void 0 ? _a : (0, sdk_1.loadSdkClientClass)(group.clientClassName);
    const client = new clientClass({ authenticationDetailsProvider: authProvider });
    (0, config_1.configureClientEndpoint)(client, endpoint);
    const capture = installHttpCapture(client, group, command, invocation.request, invocation.globals.debug);
    const sdkMethod = client[command.sdkMethodName];
    if (typeof sdkMethod !== "function") {
        throw new errors_1.CliError(`${group.name} has no command-name '${command.name}'.`);
    }
    try {
        const response = await callSdkMethod(client, sdkMethod, invocation.request);
        (0, output_1.printResponse)(response, capture.captured);
    }
    catch (error) {
        if (error instanceof errors_1.CliError) {
            throw (0, commandArgs_1.withCommandUsageHint)(group, command, error);
        }
        throw error;
    }
    finally {
        closeClient(client);
    }
    return 0;
}
async function callSdkMethod(client, sdkMethod, request) {
    const originalWarn = console.warn;
    console.warn = (...args) => {
        const message = args.map(String).join(" ");
        if (shouldSuppressSdkRetryWarning(message)) {
            return;
        }
        originalWarn.apply(console, args);
    };
    try {
        return await sdkMethod.call(client, request);
    }
    finally {
        console.warn = originalWarn;
    }
}
const SUPPRESSED_SDK_RETRY_WARNING_PREFIXES = [
    "Request cannot be retried. Not Retrying. Exception occurred",
    "All retry attempts have exhausted.",
    "Request failed with Exception :"
];
function shouldSuppressSdkRetryWarning(message) {
    return SUPPRESSED_SDK_RETRY_WARNING_PREFIXES.some((prefix) => message.startsWith(prefix));
}
function isServiceError(error) {
    if (!error || typeof error !== "object") {
        return false;
    }
    const value = error;
    return typeof value.statusCode === "number" && "serviceCode" in value && "message" in value;
}
function serviceErrorPayload(error) {
    var _a, _b, _c;
    return {
        status: serviceErrorStatus(error),
        code: String((_a = error.serviceCode) !== null && _a !== void 0 ? _a : ""),
        message: String((_b = error.message) !== null && _b !== void 0 ? _b : ""),
        "opc-request-id": (_c = error.opcRequestId) !== null && _c !== void 0 ? _c : null
    };
}
function serviceErrorStatus(error) {
    const status = Number(error.statusCode);
    return Number.isFinite(status) && status > 0 ? status : 1;
}
function formatCliError(error) {
    const sdkDetail = formatSdkErrorObject(error);
    if (sdkDetail) {
        return sdkDetail;
    }
    if (error instanceof Error) {
        return error.message;
    }
    if (isRecord(error)) {
        return (0, util_1.inspect)(error, { breakLength: 120, depth: 4 });
    }
    return String(error);
}
function formatSdkErrorObject(error) {
    if (!isRecord(error)) {
        return "";
    }
    const message = stringField(error, "message");
    const code = stringField(error, "code") || stringField(error, "serviceCode");
    const endpoint = stringField(error, "requestEndpoint");
    const troubleshooting = stringField(error, "troubleshootingPage") || stringField(error, "troubleshootingTips");
    if (!message && !code && !endpoint && !troubleshooting) {
        return "";
    }
    const parts = [];
    if (message && code) {
        parts.push(`request failed (${code}): ${message}`);
    }
    else if (message) {
        parts.push(`request failed: ${message}`);
    }
    else if (code) {
        parts.push(`request failed (${code})`);
    }
    if (endpoint) {
        parts.push(`Request Endpoint: ${endpoint}`);
    }
    if (troubleshooting) {
        parts.push(troubleshooting);
    }
    return parts.join(" ");
}
function stringField(value, key) {
    const field = value[key];
    return typeof field === "string" && field.trim() ? field : "";
}
function isRecord(value) {
    return typeof value === "object" && value !== null && !Array.isArray(value);
}
function handleConfigure(args) {
    const action = args[0];
    if (action === "get") {
        console.log("AIDP CLI configuration:");
        console.log(`  instance-id: ${process.env.AIDP_INSTANCE_ID || process.env.INSTANCE_ID || "(not set)"}`);
        console.log(`  endpoint: ${process.env.AIDP_ENDPOINT || process.env.OCI_CLI_ENDPOINT || "(not set)"}`);
        return;
    }
    if (action === "set") {
        throw new errors_1.CliError("aidp configure set is no longer supported. Use AIDP_INSTANCE_ID or --instance-id.");
    }
    throw new errors_1.CliError(`Unknown configure command '${action}'. Run 'aidp configure --help'.`);
}
function installHttpCapture(client, group, command, requestValues, debug) {
    const state = {};
    const original = client._httpClient;
    if (!original || typeof original.send !== "function") {
        return state;
    }
    client._httpClient = {
        ...original,
        send: async (...args) => {
            const request = args[0];
            if (debug) {
                printDebugDetails(group, command, request, requestValues);
            }
            const response = await original.send.apply(original, args);
            state.captured = await captureResponse(response);
            return response;
        }
    };
    return state;
}
async function captureResponse(response) {
    const headers = {};
    response.headers.forEach((value, key) => {
        headers[key] = value;
    });
    let data = null;
    try {
        const text = await response.clone().text();
        if (text.trim()) {
            try {
                data = JSON.parse(text);
            }
            catch {
                data = text;
            }
        }
    }
    catch {
        data = null;
    }
    return {
        data,
        headers,
        status: response.status
    };
}
function printDebugDetails(group, command, request, requestValues) {
    const requestFields = requestFieldMaps(command, requestValues);
    console.error("AIDP CLI debug request:");
    console.error(`  command_group: ${group.name}`);
    console.error(`  command_name: ${command.name}`);
    console.error(`  method: ${request.method || command.httpMethod}`);
    console.error(`  endpoint: ${endpointFromUrl(request.uri || "")}`);
    console.error("  base_path: ");
    console.error(`  url: ${request.uri || ""}`);
    console.error(`  resource_path: ${command.path}`);
    console.error(`  path_params: ${(0, output_1.stringify)(requestFields.pathParams)}`);
    console.error(`  query_params: ${(0, output_1.stringify)(requestFields.queryParams)}`);
    console.error(`  headers: ${(0, output_1.stringify)(safeHeaders(headersToObject(request.headers)))}`);
    console.error(`  body: ${bodyDebugSummary(request.body)}`);
}
function requestFieldMaps(command, requestValues) {
    const pathParams = {};
    const queryParams = {};
    for (const field of command.fields) {
        if (field.in === "path") {
            pathParams[field.originalName] = requestValues[field.name];
        }
        else if (field.in === "query") {
            queryParams[field.originalName] = requestValues[field.name];
        }
    }
    return { pathParams, queryParams };
}
function endpointFromUrl(value) {
    try {
        const url = new URL(value);
        const versionMatch = url.pathname.match(/^(\/\d{8})(?:\/|$)/);
        return `${url.protocol}//${url.host}${versionMatch ? versionMatch[1] : ""}`;
    }
    catch {
        return "";
    }
}
function headersToObject(headers) {
    const result = {};
    headers === null || headers === void 0 ? void 0 : headers.forEach((value, key) => {
        result[key] = value;
    });
    return result;
}
function safeHeaders(headers) {
    const sensitive = new Set(["authorization", "x-content-sha256", "security-token"]);
    return Object.fromEntries(Object.entries(headers).map(([key, value]) => [
        key,
        sensitive.has(key.toLowerCase()) ? "<redacted>" : value
    ]));
}
function bodyDebugSummary(body) {
    if (body === undefined || body === null) {
        return "none";
    }
    if (typeof body === "string") {
        try {
            const parsed = JSON.parse(body);
            if (parsed && typeof parsed === "object" && !Array.isArray(parsed)) {
                return "json object";
            }
            if (Array.isArray(parsed)) {
                return "json array";
            }
        }
        catch {
            return "string";
        }
        return "string";
    }
    if (body && typeof body === "object" && !Array.isArray(body)) {
        return "json object";
    }
    return Array.isArray(body) ? "array" : typeof body;
}
function closeClient(client) {
    const close = client.close;
    if (typeof close === "function") {
        close.call(client);
    }
}
function handleSearch(manifest, args) {
    const query = args.join(" ").trim();
    if (!query) {
        console.log((0, help_1.searchHelp)());
        return;
    }
    const results = [];
    for (const group of manifest.commandGroups) {
        const groupScore = searchScore(query, [group.name, group.description], [], group.name);
        if (groupScore) {
            results.push([groupScore, group.name, "", group.description]);
        }
        for (const command of group.commands) {
            const fields = command.fields.map((field) => field.cliName).join(" ");
            const commandScore = searchScore(query, [group.name, command.name, ...command.aliases], [command.summary, command.description, fields], `${group.name} ${command.name}`);
            if (commandScore) {
                results.push([commandScore, group.name, command.name, (0, help_1.commandSummary)(command)]);
            }
        }
    }
    if (results.length === 0) {
        console.log(`No results for '${query}'.`);
        return;
    }
    const filtered = filterSearchResults(results).sort(searchResultSortKey);
    console.log(`Search results for '${query}':`);
    const rows = filtered.slice(0, 30).map(([_score, groupName, commandName, description]) => [
        `${groupName} ${commandName}`.trim(),
        description
    ]);
    console.log(formatTable(rows));
    if (filtered.length > 30) {
        console.log(`\nShowing 30 of ${filtered.length} results.`);
    }
}
function formatTable(rows) {
    if (rows.length === 0) {
        return "";
    }
    const width = Math.max(...rows.map(([name]) => name.length));
    return rows
        .map(([name, description]) => `  ${name}${" ".repeat(Math.max(2, width - name.length + 2))}${description}`)
        .join("\n");
}
function filterSearchResults(results) {
    const topScore = Math.max(...results.map(([score]) => score));
    if (topScore < 850) {
        return results;
    }
    const cutoff = Math.floor(topScore * 0.5);
    return results.filter(([score]) => score >= cutoff);
}
function searchResultSortKey(left, right) {
    return right[0] - left[0] || left[1].localeCompare(right[1]) || left[2].localeCompare(right[2]);
}
const SEARCH_ACTION_PREFIXES = [
    "checkout",
    "commit",
    "create",
    "delete",
    "deploy",
    "download",
    "export",
    "generate",
    "head",
    "list",
    "manage",
    "merge",
    "patch",
    "pull",
    "push",
    "rebase",
    "remove",
    "restart",
    "restore",
    "search",
    "start",
    "stop",
    "summarize",
    "update",
    "upload",
    "get"
];
function searchScore(query, primaryValues, secondaryValues, sortValue) {
    const queryTokens = searchTokens(query);
    if (queryTokens.length === 0) {
        return 0;
    }
    const primaryValue = primaryValues.filter(Boolean).join(" ");
    const secondaryValue = secondaryValues.filter(Boolean).join(" ");
    const primaryTokens = searchTokens(primaryValue);
    const secondaryTokens = searchTokens(secondaryValue);
    if (primaryTokens.length === 0 && secondaryTokens.length === 0) {
        return 0;
    }
    const compactQuery = compactSearchText(query);
    const compactPrimary = compactSearchText(primaryValue);
    const compactSecondary = compactSearchText(secondaryValue);
    const compactSort = compactSearchText(sortValue);
    const compactSortVariants = compactSortValueVariants(sortValue);
    const expandedPrimaryTokens = expandSearchTokens(primaryTokens);
    const expandedSecondaryTokens = expandSearchTokens(secondaryTokens);
    if (isCompactActionQuery(queryTokens, compactQuery)) {
        return Math.max(...compactSortVariants.map((variant) => compactIdentitySearchScore(compactQuery, variant)));
    }
    let score = 0;
    let matched = false;
    const sortIdentityScore = Math.max(...compactSortVariants.map((variant) => compactIdentitySearchScore(compactQuery, variant)));
    if (sortIdentityScore >= 850) {
        score += sortIdentityScore;
        matched = true;
    }
    else if (compactQuery === compactSort) {
        score += 1000;
        matched = true;
    }
    else if (compactSort.startsWith(compactQuery)) {
        score += 900;
        matched = true;
    }
    else if (compactSort.includes(compactQuery)) {
        score += 850;
        matched = true;
    }
    else if (compactPrimary.includes(compactQuery)) {
        score += 700;
        matched = true;
    }
    else if (compactSecondary.includes(compactQuery)) {
        score += 350;
        matched = true;
    }
    if (queryTokens.every((token) => queryTokenMatches(token, expandedPrimaryTokens))) {
        score += 500;
        matched = true;
        if (searchTokensInOrder(queryTokens, primaryTokens)) {
            score += 150;
        }
    }
    else if (queryTokens.length === 1 && queryTokenMatches(queryTokens[0], expandedSecondaryTokens)) {
        score += 150;
        matched = true;
    }
    for (const token of queryTokens) {
        if (expandedPrimaryTokens.has(token)) {
            score += 25;
        }
        else if (queryTokens.length === 1 && expandedSecondaryTokens.has(token)) {
            score += 10;
        }
    }
    const fuzzyRatio = Math.max(...compactSortVariants.map((variant) => sequenceRatio(compactQuery, variant)));
    if (fuzzyRatio >= 0.72 && compactLengthRatio(compactQuery, compactSort) >= 0.8) {
        score += Math.floor(fuzzyRatio * 250);
        matched = true;
    }
    return matched ? score : 0;
}
function isCompactActionQuery(queryTokens, compactQuery) {
    return (queryTokens.length === 1 &&
        SEARCH_ACTION_PREFIXES.some((action) => compactQuery.startsWith(action) && compactQuery.length > action.length + 2));
}
function compactIdentitySearchScore(compactQuery, compactSort) {
    for (const variant of compactSearchVariants(compactQuery)) {
        if (variant === compactSort) {
            return 1000;
        }
        if (compactSort.startsWith(variant)) {
            return 900;
        }
        if (compactSort.includes(variant)) {
            return 850;
        }
    }
    return 0;
}
function compactSearchVariants(value) {
    const variants = [value];
    if (value.length > 4 && value.endsWith("ies")) {
        variants.push(`${value.slice(0, -3)}y`);
    }
    if (value.length > 3 && value.endsWith("s")) {
        variants.push(value.slice(0, -1));
    }
    return Array.from(new Set(variants));
}
function compactSortValueVariants(value) {
    const tokens = searchTokens(value);
    const variants = [compactSearchText(value)];
    if (tokens.length >= 2 && SEARCH_ACTION_PREFIXES.includes(tokens[tokens.length - 1])) {
        variants.push([tokens[tokens.length - 1], ...tokens.slice(0, -1)].join(""));
    }
    return Array.from(new Set(variants));
}
function compactLengthRatio(left, right) {
    if (!left || !right) {
        return 0;
    }
    return Math.min(left.length, right.length) / Math.max(left.length, right.length);
}
function searchTokens(value) {
    var _a;
    return (_a = value.toLowerCase().match(/[a-z0-9]+/g)) !== null && _a !== void 0 ? _a : [];
}
function compactSearchText(value) {
    return searchTokens(value).join("");
}
function expandSearchTokens(tokens) {
    const expanded = new Set();
    for (const token of tokens) {
        for (const variant of searchTokenVariants(token)) {
            expanded.add(variant);
        }
    }
    return expanded;
}
function searchTokenVariants(token) {
    const variants = new Set([token]);
    if (token.length > 4 && token.endsWith("ies")) {
        variants.add(`${token.slice(0, -3)}y`);
    }
    if (token.length > 3 && token.endsWith("s")) {
        variants.add(token.slice(0, -1));
    }
    return variants;
}
function queryTokenMatches(queryToken, valueTokens) {
    for (const variant of searchTokenVariants(queryToken)) {
        if (valueTokens.has(variant)) {
            return true;
        }
    }
    if (queryToken.length < 5) {
        return false;
    }
    return Array.from(valueTokens).some((token) => sequenceRatio(queryToken, token) >= 0.82);
}
function searchTokensInOrder(queryTokens, valueTokens) {
    let valueIndex = 0;
    const expandedValues = valueTokens.map(searchTokenVariants);
    for (const queryToken of queryTokens) {
        const variants = searchTokenVariants(queryToken);
        let found = false;
        while (valueIndex < expandedValues.length) {
            if (intersects(variants, expandedValues[valueIndex])) {
                valueIndex += 1;
                found = true;
                break;
            }
            valueIndex += 1;
        }
        if (!found) {
            return false;
        }
    }
    return true;
}
function intersects(left, right) {
    for (const item of left) {
        if (right.has(item)) {
            return true;
        }
    }
    return false;
}
function sequenceRatio(left, right) {
    if (left === right) {
        return 1;
    }
    const distance = levenshtein(left, right);
    return 1 - distance / Math.max(left.length, right.length, 1);
}
function levenshtein(left, right) {
    const previous = Array.from({ length: right.length + 1 }, (_value, index) => index);
    for (let i = 0; i < left.length; i += 1) {
        let last = previous[0];
        previous[0] = i + 1;
        for (let j = 0; j < right.length; j += 1) {
            const old = previous[j + 1];
            previous[j + 1] =
                left[i] === right[j]
                    ? last
                    : Math.min(last + 1, previous[j] + 1, previous[j + 1] + 1);
            last = old;
        }
    }
    return previous[right.length];
}
function unknownCommandGroup(manifest, name) {
    const candidates = [
        "command-groups",
        "search",
        "configure",
        "help",
        "version",
        ...manifest.commandGroups.map((group) => group.name)
    ];
    return new errors_1.CliError((0, commandArgs_1.unknownMessage)("command group", name, candidates, "aidp command-groups"));
}
function unknownCommandName(group, name) {
    const candidates = group.commands.map((command) => command.name);
    let matches = (0, commandArgs_1.closeMatches)(name, candidates, 3);
    if (matches.length === 0) {
        const normalized = (0, names_1.normalizedLookupName)(name);
        matches = group.commands
            .filter((command) => [command.sdkMethodName, command.operationId, ...command.aliases].some((alias) => (0, names_1.normalizedLookupName)(alias) === normalized))
            .map((command) => command.name)
            .slice(0, 3);
    }
    return new errors_1.CliError((0, commandArgs_1.unknownMessage)("command name", name, candidates, `aidp ${group.name} --help`, matches));
}
function printVersion() {
    console.log(`aidp npm cli version-${packageVersion()}`);
}
function packageVersion() {
    var _a;
    try {
        const packageJson = JSON.parse((0, fs_1.readFileSync)((0, path_1.join)(__dirname, "..", "package.json"), "utf8"));
        return (_a = packageJson.version) !== null && _a !== void 0 ? _a : "unknown";
    }
    catch {
        return "unknown";
    }
}
