// aidp-cli/dist/commandArgs.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseCommandOptions = parseCommandOptions;
exports.commandArgumentFields = commandArgumentFields;
exports.commandOptionFields = commandOptionFields;
exports.commandRequiresInstanceId = commandRequiresInstanceId;
exports.commandUsesRawBody = commandUsesRawBody;
exports.argumentMetavar = argumentMetavar;
exports.withCommandUsageHint = withCommandUsageHint;
exports.unknownMessage = unknownMessage;
exports.closeMatches = closeMatches;
const args_1 = require("./args");
const config_1 = require("./config");
const errors_1 = require("./errors");
const bodySecurity_1 = require("./bodySecurity");
const json_1 = require("./json");
const names_1 = require("./names");
function parseCommandOptions(group, command, tokens, globals) {
    let parsedGlobals = { ...globals };
    const request = {};
    const positionalFields = commandArgumentFields(command);
    const optionFields = commandOptionMap(command);
    const optionNames = Array.from(optionFields.keys()).sort();
    let positionalIndex = 0;
    let noRequestId = false;
    let index = 0;
    while (index < tokens.length) {
        const token = tokens[index];
        if (!token.startsWith("-")) {
            const field = positionalFields[positionalIndex];
            if (!field) {
                throw new errors_1.CliError(`Unexpected argument '${token}'.`);
            }
            request[field.name] = parseFieldValue(field, token, argumentMetavar(field));
            positionalIndex += 1;
            index += 1;
            continue;
        }
        if (token === "-h" || token === "--help") {
            index += 1;
            continue;
        }
        if (token === "--no-request-id") {
            noRequestId = true;
            index += 1;
            continue;
        }
        if (token in args_1.GLOBAL_BOOLEAN_OPTIONS) {
            parsedGlobals = {
                ...parsedGlobals,
                [args_1.GLOBAL_BOOLEAN_OPTIONS[token]]: true
            };
            index += 1;
            continue;
        }
        const [optionName, inlineValue] = (0, args_1.splitInlineOption)(token);
        if (optionName in args_1.GLOBAL_VALUE_OPTIONS) {
            const [value, nextIndex] = (0, args_1.optionValue)(tokens, index, optionName, inlineValue);
            parsedGlobals = (0, args_1.applyGlobalOption)(parsedGlobals, optionName, value);
            index = nextIndex;
            continue;
        }
        if (optionName === "--body") {
            if (!command.bodyField) {
                throw new errors_1.CliError(`${command.name} does not accept --body.`);
            }
            const [value, nextIndex] = (0, args_1.optionValue)(tokens, index, optionName, inlineValue);
            const usesRawBody = commandUsesRawBody(command);
            const body = usesRawBody ? (0, json_1.readRawBodyArgument)(value) : (0, json_1.parseJsonObject)((0, json_1.readJsonArgument)(value), "--body");
            if (!usesRawBody && (0, bodySecurity_1.jsonBodyArgumentSource)(value) === "inline" && (0, bodySecurity_1.bodyContainsSensitiveField)(command, body)) {
                throw new errors_1.CliError("Inline --body JSON is blocked because this request body contains sensitive fields. " +
                    "Use --body @request.json, --body file:///path/request.json, or --body -.");
            }
            request[command.bodyField] = body;
            index = nextIndex;
            continue;
        }
        if (optionName === "--opc-request-id") {
            const [value, nextIndex] = (0, args_1.optionValue)(tokens, index, optionName, inlineValue);
            request.opcRequestId = value;
            index = nextIndex;
            continue;
        }
        const field = optionFields.get(optionName);
        if (!field) {
            throw new errors_1.CliError(unknownMessage("option", optionName, optionNames, `aidp ${group.name} ${command.name} -h`));
        }
        const [value, nextIndex] = (0, args_1.optionValue)(tokens, index, optionName, inlineValue);
        request[field.name] = parseFieldValue(field, value, optionName);
        index = nextIndex;
    }
    validatePositionalArguments(positionalFields, positionalIndex);
    applyInstanceId(command, request, parsedGlobals);
    validateRequiredFields(command, request, positionalFields);
    addRequestId(command, request, noRequestId);
    return { globals: parsedGlobals, request };
}
function commandArgumentFields(command) {
    return command.fields
        .filter((field) => {
        return field.required && field.in === "path" && field.name !== "aiDataPlatformId";
    })
        .map(normalizeField);
}
function commandOptionFields(command) {
    const argumentFieldNames = new Set(commandArgumentFields(command).map((field) => field.name));
    return command.fields
        .filter((field) => {
        if (field.name === "aiDataPlatformId" || field.name === "opcRequestId") {
            return false;
        }
        if (argumentFieldNames.has(field.name)) {
            return false;
        }
        return field.in !== "body";
    })
        .map(normalizeField);
}
function commandRequiresInstanceId(command) {
    return command.fields.some((field) => field.name === "aiDataPlatformId" && field.required);
}
function commandUsesRawBody(command) {
    return Boolean(command.bodyField &&
        !command.bodyModel &&
        command.bodyFields.length === 0 &&
        Object.keys(command.bodyModels).length === 0);
}
function argumentMetavar(field) {
    return field.cliName.replace(/-/g, "_").toUpperCase();
}
function withCommandUsageHint(group, command, error) {
    const help = `Use "aidp ${group.name} ${command.name} -h" for command help.`;
    if (error.message.includes(help)) {
        return error;
    }
    return new errors_1.CliError(`${error.message}\n${help}`, error.exitCode);
}
function unknownMessage(kind, name, candidates, helpCommand, matches = closeMatches(name, candidates, 3)) {
    const parts = [`Unknown ${kind} '${name}'.`];
    if (matches.length > 0) {
        parts.push(`Did you mean ${matches.map((match) => `'${match}'`).join(", ")}?`);
    }
    parts.push(`Run "${helpCommand}" for help.`);
    return parts.join(" ");
}
function closeMatches(value, candidates, limit = 3) {
    const normalizedValue = (0, names_1.normalizedLookupName)(value);
    const scored = Array.from(new Set(candidates))
        .map((candidate) => {
        const normalizedCandidate = (0, names_1.normalizedLookupName)(candidate);
        return {
            candidate,
            score: matchScore(normalizedValue, normalizedCandidate)
        };
    })
        .filter((item) => item.score >= 0.55)
        .sort((left, right) => right.score - left.score || left.candidate.localeCompare(right.candidate));
    return scored.slice(0, limit).map((item) => item.candidate);
}
function commandOptionMap(command) {
    const options = new Map();
    for (const field of commandOptionFields(command)) {
        options.set(`--${field.cliName}`, field);
    }
    return options;
}
function validatePositionalArguments(fields, parsedCount) {
    if (parsedCount >= fields.length) {
        return;
    }
    throw new errors_1.CliError(`Missing required argument ${argumentMetavar(fields[parsedCount])}.`);
}
function applyInstanceId(command, request, globals) {
    if (!commandRequiresInstanceId(command) || request.aiDataPlatformId !== undefined) {
        return;
    }
    if (!globals.instanceId) {
        throw new errors_1.CliError("Missing AIDP instance OCID. Set --instance-id or AIDP_INSTANCE_ID.");
    }
    request.aiDataPlatformId = globals.instanceId;
}
function validateRequiredFields(command, request, positionalFields) {
    const positionalFieldNames = new Set(positionalFields.map((field) => field.name));
    for (const field of command.fields) {
        if (!field.required || request[field.name] !== undefined) {
            continue;
        }
        if (field.name === "aiDataPlatformId" || positionalFieldNames.has(field.name)) {
            continue;
        }
        if (field.in === "body") {
            throw new errors_1.CliError("Missing required option --body.");
        }
        throw new errors_1.CliError(`Missing required option --${field.cliName}.`);
    }
}
function addRequestId(command, request, noRequestId) {
    if (noRequestId || request.opcRequestId !== undefined) {
        return;
    }
    if (command.fields.some((field) => field.name === "opcRequestId")) {
        request.opcRequestId = (0, config_1.buildRequestId)();
    }
}
function parseFieldValue(field, rawValue, label) {
    const enumValues = fieldEnumValues(field);
    if (enumValues.length > 0) {
        const matched = enumValues.find((value) => String(value) === rawValue);
        if (matched === undefined) {
            throw new errors_1.CliError(`${label} must be one of: ${enumValues.map(String).join(", ")}.`);
        }
        return matched;
    }
    if (field.type === "boolean" || field.type === "bool") {
        return parseBoolean(rawValue, label);
    }
    if (field.type === "number" || field.type === "float" || field.type === "integer" || field.type === "int") {
        return parseNumber(rawValue, label, field.type === "integer" || field.type === "int");
    }
    if (field.type === "array") {
        const value = (0, json_1.parseJsonValue)(rawValue);
        if (!Array.isArray(value)) {
            throw new errors_1.CliError(`${label} must be a JSON array.`);
        }
        return value;
    }
    if (field.type === "object") {
        return (0, json_1.parseJsonObject)((0, json_1.readJsonArgument)(rawValue), label);
    }
    return rawValue;
}
function parseBoolean(value, label) {
    const normalized = value.toLowerCase();
    if (normalized === "true") {
        return true;
    }
    if (normalized === "false") {
        return false;
    }
    throw new errors_1.CliError(`${label} must be true or false.`);
}
function parseNumber(value, label, integer) {
    const parsed = Number(value);
    if (!Number.isFinite(parsed) || (integer && !Number.isInteger(parsed))) {
        throw new errors_1.CliError(`${label} must be ${integer ? "an integer" : "a number"}.`);
    }
    return parsed;
}
function normalizeField(field) {
    return {
        ...field,
        enumValues: fieldEnumValues(field)
    };
}
function fieldEnumValues(field) {
    return Array.isArray(field.enumValues) ? field.enumValues : [];
}
function matchScore(value, candidate) {
    if (!value || !candidate) {
        return 0;
    }
    if (value === candidate) {
        return 1;
    }
    if (candidate.startsWith(value)) {
        return 0.95;
    }
    if (candidate.includes(value)) {
        return 0.85;
    }
    const distance = levenshtein(value, candidate);
    return 1 - distance / Math.max(value.length, candidate.length, 1);
}
function levenshtein(left, right) {
    const previous = Array.from({ length: right.length + 1 }, (_value, index) => index);
    for (let i = 0; i < left.length; i += 1) {
        let last = previous[0];
        previous[0] = i + 1;
        for (let j = 0; j < right.length; j += 1) {
            const old = previous[j + 1];
            previous[j + 1] =
                left[i] === right[j]
                    ? last
                    : Math.min(last + 1, previous[j] + 1, previous[j + 1] + 1);
            last = old;
        }
    }
    return previous[right.length];
}
