// aidp-cli/dist/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AUTH_CHOICES = exports.DEFAULT_ENVIRONMENT_DOMAIN = exports.DEFAULT_ENVIRONMENT_PREFIX = exports.DEFAULT_CONFIG_FILE = exports.DEFAULT_AUTH = exports.DEFAULT_PROFILE = void 0;
exports.defaultGlobalOptions = defaultGlobalOptions;
exports.assertAuthMode = assertAuthMode;
exports.expandHome = expandHome;
exports.configuredInstanceId = configuredInstanceId;
exports.parseProfile = parseProfile;
exports.resolveRegion = resolveRegion;
exports.buildAuthenticationDetailsProvider = buildAuthenticationDetailsProvider;
exports.resolveEndpoint = resolveEndpoint;
exports.configureClientEndpoint = configureClientEndpoint;
exports.buildRequestId = buildRequestId;
const fs_1 = require("fs");
const crypto_1 = require("crypto");
const common = require("oci-common");
const errors_1 = require("./errors");
exports.DEFAULT_PROFILE = "DEFAULT";
exports.DEFAULT_AUTH = "security_token";
exports.DEFAULT_CONFIG_FILE = "~/.oci/config";
exports.DEFAULT_ENVIRONMENT_PREFIX = "aidp";
exports.DEFAULT_ENVIRONMENT_DOMAIN = "oraclecloud.com";
exports.AUTH_CHOICES = [
    "api_key",
    "security_token",
    "instance_principal",
    "resource_principal"
];
const CONFIG_ENV_OVERRIDES = {
    OCI_CLI_USER: "user",
    OCI_CLI_FINGERPRINT: "fingerprint",
    OCI_CLI_KEY_FILE: "key_file",
    OCI_CLI_KEY_CONTENT: "key_content",
    OCI_CLI_TENANCY: "tenancy",
    OCI_CLI_PASSPHRASE: "pass_phrase",
    OCI_CLI_SECURITY_TOKEN_FILE: "security_token_file"
};
function defaultGlobalOptions() {
    const auth = (process.env.OCI_CLI_AUTH || exports.DEFAULT_AUTH);
    assertAuthMode(auth);
    return {
        auth,
        configFile: process.env.OCI_CLI_CONFIG_FILE || exports.DEFAULT_CONFIG_FILE,
        debug: false,
        endpoint: process.env.AIDP_ENDPOINT || process.env.OCI_CLI_ENDPOINT,
        environmentDomain: exports.DEFAULT_ENVIRONMENT_DOMAIN,
        environmentPrefix: exports.DEFAULT_ENVIRONMENT_PREFIX,
        instanceId: configuredInstanceId(),
        profile: process.env.OCI_CLI_PROFILE || exports.DEFAULT_PROFILE,
        region: process.env.OCI_CLI_REGION
    };
}
function assertAuthMode(value) {
    if (!exports.AUTH_CHOICES.includes(value)) {
        throw new errors_1.CliError(`argument --auth: invalid choice: '${value}' (choose from ${exports.AUTH_CHOICES.join(", ")})`);
    }
}
function expandHome(value) {
    if (value === "~") {
        return process.env.HOME || value;
    }
    if (value.startsWith("~/")) {
        return `${process.env.HOME || ""}/${value.slice(2)}`;
    }
    return value;
}
function configuredInstanceId() {
    return process.env.AIDP_INSTANCE_ID || process.env.INSTANCE_ID || undefined;
}
function parseProfile(configFilePath, profileName) {
    const rawConfig = (0, fs_1.readFileSync)(expandHome(configFilePath), "utf8");
    const lines = rawConfig.split(/\r?\n/);
    let currentProfile;
    const sections = new Map();
    for (const rawLine of lines) {
        const line = rawLine.trim();
        if (!line || line.startsWith("#") || line.startsWith(";")) {
            continue;
        }
        const sectionMatch = line.match(/^\[(.+)\]$/);
        if (sectionMatch) {
            currentProfile = sectionMatch[1].trim();
            if (!sections.has(currentProfile)) {
                sections.set(currentProfile, {});
            }
            continue;
        }
        if (!currentProfile) {
            continue;
        }
        const separatorIndex = line.indexOf("=");
        if (separatorIndex < 0) {
            continue;
        }
        const key = line.slice(0, separatorIndex).trim();
        const value = line.slice(separatorIndex + 1).trim();
        sections.get(currentProfile)[key] = value;
    }
    const profile = sections.get(profileName);
    if (!profile) {
        throw new errors_1.CliError(`OCI profile ${profileName} was not found in ${expandHome(configFilePath)}.`);
    }
    applyConfigEnvOverrides(profile);
    return profile;
}
function resolveRegion(options, authProvider) {
    if (options.region) {
        return options.region;
    }
    const providerRegion = getProviderRegion(authProvider);
    if (providerRegion) {
        return providerRegion;
    }
    if (options.auth !== "instance_principal" && options.auth !== "resource_principal") {
        return parseProfile(options.configFile, options.profile).region;
    }
    return undefined;
}
async function buildAuthenticationDetailsProvider(options) {
    const configFile = expandHome(options.configFile);
    if (options.auth === "api_key") {
        return new common.ConfigFileAuthenticationDetailsProvider(configFile, options.profile);
    }
    if (options.auth === "security_token") {
        return new common.SessionAuthDetailProvider(configFile, options.profile);
    }
    if (options.auth === "instance_principal") {
        const builder = new common.InstancePrincipalsAuthenticationDetailsProviderBuilder();
        return await builder.build();
    }
    return common.ResourcePrincipalAuthenticationDetailsProvider.builder();
}
function resolveEndpoint(options, authProvider) {
    if (options.endpoint) {
        return normalizeEndpoint(options.endpoint);
    }
    if (options.environmentHost) {
        return `https://${options.environmentHost
            .trim()
            .replace(/^https?:\/\//, "")
            .replace(/\/+$/, "")}`;
    }
    const region = resolveRegion(options, authProvider);
    if (!region) {
        throw new errors_1.CliError("Set --region, --endpoint, --environment-host, or region in the OCI config profile.");
    }
    if (/^https?:\/\//.test(region)) {
        throw new errors_1.CliError("Region must be an OCI region identifier. For a full service URL, use --endpoint, AIDP_ENDPOINT, or OCI_CLI_ENDPOINT.");
    }
    return `https://${options.environmentPrefix}.${region}.oci.${options.environmentDomain}`.replace(/\/+$/, "");
}
function configureClientEndpoint(client, endpoint) {
    client.endpoint = endpoint;
}
function buildRequestId() {
    return `aidp-cli-${(0, crypto_1.randomUUID)()}`;
}
function normalizeEndpoint(endpoint) {
    const stripped = endpoint.trim().replace(/\/+$/, "");
    if (!stripped) {
        throw new errors_1.CliError("--endpoint cannot be empty.");
    }
    if (stripped.startsWith("https://") || stripped.startsWith("http://")) {
        return stripped;
    }
    return `https://${stripped}`;
}
function applyConfigEnvOverrides(profile) {
    for (const [envName, configKey] of Object.entries(CONFIG_ENV_OVERRIDES)) {
        const value = process.env[envName];
        if (value) {
            profile[configKey] = value;
        }
    }
}
function getProviderRegion(authProvider) {
    var _a;
    if (!authProvider) {
        return undefined;
    }
    const provider = authProvider;
    const getRegion = provider.getRegion;
    if (typeof getRegion === "function") {
        return normalizeRegionValue(getRegion.call(authProvider));
    }
    return (_a = normalizeRegionValue(provider.region)) !== null && _a !== void 0 ? _a : normalizeRegionValue(provider.regionId);
}
function normalizeRegionValue(value) {
    if (typeof value === "string" && value.trim()) {
        return value.trim();
    }
    if (value && typeof value === "object") {
        const regionId = value.regionId;
        if (typeof regionId === "string" && regionId.trim()) {
            return regionId.trim();
        }
    }
    return undefined;
}
