// aidp-cli/dist/help.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UTILITY_COMMANDS = void 0;
exports.rootHelp = rootHelp;
exports.commandGroupsHelp = commandGroupsHelp;
exports.commandGroupsOutput = commandGroupsOutput;
exports.groupHelp = groupHelp;
exports.commandHelp = commandHelp;
exports.searchHelp = searchHelp;
exports.configureHelp = configureHelp;
exports.rootFlagRows = rootFlagRows;
exports.globalFlagRows = globalFlagRows;
exports.formatTable = formatTable;
exports.formatFlags = formatFlags;
exports.commandSummary = commandSummary;
const bodySecurity_1 = require("./bodySecurity");
const commandArgs_1 = require("./commandArgs");
const MAX_BODY_EXAMPLE_DEPTH = 12;
exports.UTILITY_COMMANDS = [
    ["command-groups", "List API command groups."],
    ["search", "Search command groups and command names."],
    ["configure", "Show AIDP CLI environment defaults."],
    ["help", "Help about any command."],
    ["version", "Show CLI version."]
];
function rootHelp(manifest) {
    return [
        "AIDP CLI",
        "",
        "Usage:",
        "  aidp <command-group> <command-name> [flags]",
        "",
        "API Command Groups:",
        ...formatTable(manifest.commandGroups.map((group) => [group.name, group.description])),
        "",
        "Utility Commands:",
        ...formatTable(exports.UTILITY_COMMANDS),
        "",
        "Flags:",
        ...formatFlags(rootFlagRows()),
        "",
        'Use "aidp <command-group> --help" for more information about a command group.'
    ].join("\n");
}
function commandGroupsHelp() {
    return [
        "List AIDP API command groups.",
        "",
        "Usage:",
        "  aidp command-groups [flags]",
        "",
        "Flags:",
        ...formatFlags([["-h, --help", "help for command-groups"]]),
        "",
        "Global Flags:",
        ...formatFlags(globalFlagRows())
    ].join("\n");
}
function commandGroupsOutput(manifest) {
    const lines = [
        "Command Groups:",
        ...formatTable(manifest.commandGroups.map((group) => [group.name, group.description])),
        "",
        'Use "aidp <command-group> --help" for command names and examples.'
    ];
    const example = runnableCommandGroupsExample(manifest);
    if (example) {
        lines.push("", "Example:", example);
    }
    return lines.join("\n");
}
function groupHelp(group) {
    const lines = [
        group.description,
        "",
        "Usage:",
        `  aidp ${group.name} [flags]`,
        `  aidp ${group.name} <command-name> [arguments] [flags]`,
        ""
    ];
    for (const [section, commands] of Object.entries(groupedCommands(group))) {
        lines.push(`${section}:`);
        lines.push(...formatTable(commands.map((command) => [command.name, commandSummary(command)])));
        lines.push("");
    }
    lines.push("Flags:", ...formatFlags([["-h, --help", `help for ${group.name}`]]), "", "Global Flags:", ...formatFlags(globalFlagRows()), "", `Use "aidp ${group.name} <command-name> --help" for more information about a command.`);
    return lines.join("\n");
}
function commandHelp(group, command) {
    const lines = [];
    const description = commandDescription(command);
    if (description) {
        lines.push(...wrapParagraph(description), "");
    }
    const argumentFields = (0, commandArgs_1.commandArgumentFields)(command);
    const usageArgs = argumentFields.map(commandArgs_1.argumentMetavar).join(" ");
    lines.push("Usage:", `  aidp ${group.name} ${command.name}${usageArgs ? ` ${usageArgs}` : ""} [flags]`, "");
    if (argumentFields.length > 0) {
        lines.push("Arguments:");
        for (const field of argumentFields) {
            lines.push(...formatArgument(field));
        }
        lines.push("");
    }
    lines.push("Flags:", ...formatFlags(commandFlagRows(command)), "", "Global Flags:", ...formatFlags(globalFlagRows()), "", "Examples:", ...formatExamples(group, command));
    const bodySample = exampleBody(command);
    if (bodySample !== undefined) {
        lines.push("", "Example JSON:", JSON.stringify(bodySample, null, 2));
    }
    const variants = rootBodyVariantExamples(command);
    if (variants.length > 0) {
        lines.push("", "Body variants:");
        for (const variant of variants) {
            lines.push("", `Example JSON - ${variant.label}:`, JSON.stringify(variant.sample, null, 2));
        }
    }
    const nestedVariants = nestedBodyVariantExamples(command);
    if (nestedVariants.length > 0) {
        lines.push("", "Nested body variants:");
        for (const variant of nestedVariants) {
            lines.push("", `Example JSON for ${variant.path} - ${variant.label}:`, JSON.stringify(variant.sample, null, 2));
        }
    }
    const requiredFields = bodyRequiredFields(command);
    if (requiredFields.length > 0) {
        lines.push("", "Required JSON fields:", ...requiredFields.map((field) => `  ${field}`));
    }
    const enumFields = bodyEnumFields(command);
    if (Object.keys(enumFields).length > 0) {
        lines.push("", "Allowed JSON values:");
        for (const [field, values] of Object.entries(enumFields).sort(([left], [right]) => left.localeCompare(right))) {
            lines.push(`  ${field}: ${values.map(String).join(", ")}`);
        }
    }
    return lines.join("\n");
}
function searchHelp() {
    return [
        "Search AIDP command groups, command names, descriptions, and flags.",
        "",
        "Usage:",
        "  aidp search QUERY [flags]",
        "",
        "Examples:",
        "  aidp search workspace",
        "  aidp search list workspaces",
        "  aidp search permission",
        "",
        "Flags:",
        ...formatFlags([["-h, --help", "help for search"]]),
        "",
        "Global Flags:",
        ...formatFlags(globalFlagRows())
    ].join("\n");
}
function configureHelp() {
    return [
        "Show AIDP CLI environment defaults.",
        "",
        "Usage:",
        "  aidp configure get",
        "",
        "Set defaults with AIDP_INSTANCE_ID and AIDP_ENDPOINT environment variables.",
        "",
        "Flags:",
        ...formatFlags([["-h, --help", "help for configure"]])
    ].join("\n");
}
function rootFlagRows() {
    return [
        ["--debug", "enable debug logging"],
        ["-h, --help", "help for aidp"],
        ["-p, --profile", "OCI config profile; the default profile: DEFAULT"],
        ["--auth", "OCI authentication mode; options: api_key, security_token, instance_principal, resource_principal; default: security_token"],
        ["--config-file", "OCI config file path"],
        ["--region", "OCI region"],
        ["--endpoint", "AIDP data plane endpoint override; default endpoint points to https://aidp.<region>.oci.oraclecloud.com"],
        ["--instance-id", "AIDP instance OCID"],
        ["--timeout", "connection/read timeout in seconds"],
        ["-v, --version", "version for aidp"]
    ];
}
function globalFlagRows() {
    return rootFlagRows().filter(([flag]) => flag !== "-h, --help" && flag !== "-v, --version");
}
function formatTable(rows, indent = 2, gap = 2) {
    if (rows.length === 0) {
        return [];
    }
    const width = Math.max(...rows.map(([name]) => name.length));
    return rows.map(([name, description]) => {
        const padding = " ".repeat(Math.max(gap, width - name.length + gap));
        return `${" ".repeat(indent)}${name}${padding}${description}`;
    });
}
function formatFlags(rows) {
    return formatTable(rows, 2, 2);
}
function commandSummary(command) {
    return firstSentence(command.summary || command.description);
}
function groupedCommands(group) {
    var _a;
    const sections = {};
    for (const command of group.commands) {
        const section = command.section || "Available Commands";
        sections[section] = (_a = sections[section]) !== null && _a !== void 0 ? _a : [];
        sections[section].push(command);
    }
    if (sections["Available Commands"]) {
        const available = sections["Available Commands"];
        delete sections["Available Commands"];
        return {
            "Available Commands": available,
            ...Object.fromEntries(Object.entries(sections).sort(([left], [right]) => left.localeCompare(right)))
        };
    }
    return Object.fromEntries(Object.entries(sections).sort(([left], [right]) => left.localeCompare(right)));
}
function commandDescription(command) {
    let description = command.description || command.summary;
    if (!description) {
        return "";
    }
    if (!/[.!?]$/.test(description)) {
        description += ".";
    }
    if (command.deprecated && !description.toLowerCase().startsWith("deprecated")) {
        description = `Deprecated. ${description}`;
    }
    return description;
}
function firstSentence(value) {
    const cleaned = String(value || "").trim().replace(/\s+/g, " ");
    if (!cleaned) {
        return "";
    }
    for (const delimiter of [". ", "! ", "? "]) {
        if (cleaned.includes(delimiter)) {
            return `${cleaned.split(delimiter, 1)[0].replace(/[.!?]+$/, "")}.`;
        }
    }
    return /[.!?]$/.test(cleaned) ? cleaned : `${cleaned}.`;
}
function commandFlagRows(command) {
    const rows = [["-h, --help", `help for ${command.name}`]];
    for (const field of (0, commandArgs_1.commandOptionFields)(command)) {
        rows.push([`--${field.cliName}`, fieldHelp(field)]);
    }
    if (command.bodyField) {
        rows.push(["--body", bodyFlagHelp(command)]);
    }
    rows.push(["--opc-request-id", "request ID; generated automatically when omitted"], ["--no-request-id", "do not add opc_request_id automatically"]);
    return rows;
}
function bodyFlagHelp(command) {
    if ((0, commandArgs_1.commandUsesRawBody)(command)) {
        return "raw body string, @path/to/file, file:///path/file, or - for stdin";
    }
    if ((0, bodySecurity_1.commandHasSensitiveBodyFields)(command)) {
        return "@path/to/file.json, file:///path/request.json, or - for stdin; inline JSON is blocked when it contains sensitive fields";
    }
    return "inline JSON string, @path/to/file.json, file:///path/request.json, or - for stdin";
}
function fieldHelp(field) {
    const parts = [];
    let description = field.description;
    if (field.enumValues.length > 0) {
        description = description.replace(/\s*Allowed values(?:\s+are)?:\s*[^.]+\.?$/i, "").trim();
    }
    if (description) {
        parts.push(`${description.replace(/\.+$/, "")}.`);
    }
    if (field.required) {
        parts.push("Required.");
    }
    if (field.enumValues.length > 0) {
        parts.push(`Allowed values: ${field.enumValues.map(String).join(", ")}.`);
    }
    return parts.join(" ");
}
function formatArgument(field) {
    const label = `  ${(0, commandArgs_1.argumentMetavar)(field)}:`;
    return wrapParagraph(field.description || "Required command argument.", label.length + 1).map((line, index) => (index === 0 ? `${label} ${line.trimStart()}` : line));
}
function wrapParagraph(value, indent = 0) {
    const width = 88;
    const words = value.trim().replace(/\s+/g, " ").split(" ").filter(Boolean);
    const lines = [];
    let line = " ".repeat(indent);
    for (const word of words) {
        if (line.trim() && `${line} ${word}`.length > width) {
            lines.push(line);
            line = `${" ".repeat(indent)}${word}`;
        }
        else {
            line = line.trim() ? `${line} ${word}` : `${line}${word}`;
        }
    }
    if (line.trim() || lines.length === 0) {
        lines.push(line);
    }
    return lines;
}
function formatExamples(group, command) {
    let base = `  aidp ${group.name} ${command.name}`;
    for (const field of (0, commandArgs_1.commandArgumentFields)(command)) {
        base += ` <${field.cliName.replace(/-/g, "_")}>`;
    }
    if ((0, commandArgs_1.commandRequiresInstanceId)(command)) {
        base += " --instance-id <ocid>";
    }
    if (command.bodyField) {
        base += (0, commandArgs_1.commandUsesRawBody)(command) ? " --body @body.txt" : " --body @request.json";
    }
    return [base];
}
function runnableCommandGroupsExample(manifest) {
    const candidates = [];
    for (const group of manifest.commandGroups) {
        for (const command of group.commands) {
            const argumentCount = (0, commandArgs_1.commandArgumentFields)(command).length;
            const bodyCount = command.bodyField ? 1 : 0;
            const actionRank = command.name.startsWith("list-") || command.name === "list" ? 0 : 1;
            candidates.push([[bodyCount, argumentCount, actionRank, group.name, command.name], group, command]);
        }
    }
    candidates.sort((left, right) => compareRank(left[0], right[0]));
    const candidate = candidates[0];
    return candidate ? formatExamples(candidate[1], candidate[2])[0] : "";
}
function compareRank(left, right) {
    for (let i = 0; i < left.length; i += 1) {
        const leftValue = left[i];
        const rightValue = right[i];
        if (typeof leftValue === "number" && typeof rightValue === "number") {
            if (leftValue !== rightValue) {
                return leftValue - rightValue;
            }
        }
        else if (String(leftValue) !== String(rightValue)) {
            return String(leftValue).localeCompare(String(rightValue));
        }
    }
    return 0;
}
function exampleBody(command) {
    if (!command.bodyField) {
        return undefined;
    }
    const model = rootBodyModel(command);
    const fields = model ? model.fields : command.bodyFields;
    return sampleObjectForFields(fields, command, new Set(), 0);
}
function rootBodyVariantExamples(command) {
    const rootModel = rootBodyModel(command);
    if (!rootModel || rootModel.variants.length === 0) {
        return [];
    }
    return rootModel.variants
        .filter((variant) => Boolean(command.bodyModels[variant.modelName]))
        .map((variant) => {
        return {
            label: `${variant.modelName} (${variant.discriminatorField}=${String(variant.discriminatorValue)})`,
            sample: variantBodySample(command, command.bodyModel, variant)
        };
    });
}
function nestedBodyVariantExamples(command) {
    const rootModel = rootBodyModel(command);
    if (!rootModel) {
        return [];
    }
    return collectNestedBodyVariantExamples(rootModel.fields, command, "", new Set([command.bodyModel]), 0);
}
function collectNestedBodyVariantExamples(fields, command, parentPath, seenModels, depth) {
    if (depth >= MAX_BODY_EXAMPLE_DEPTH) {
        return [];
    }
    const examples = [];
    for (const field of fields) {
        if (!field.modelName || !command.bodyModels[field.modelName]) {
            continue;
        }
        const path = parentPath ? `${parentPath}.${field.name}` : field.name;
        const modelPath = field.type === "array" ? `${path}[]` : path;
        const model = command.bodyModels[field.modelName];
        for (const variant of model.variants) {
            if (!command.bodyModels[variant.modelName]) {
                continue;
            }
            examples.push({
                path: modelPath,
                label: `${variant.modelName} (${variant.discriminatorField}=${String(variant.discriminatorValue)})`,
                sample: variantBodySample(command, field.modelName, variant)
            });
        }
        if (!seenModels.has(field.modelName)) {
            const nextSeen = new Set(seenModels);
            nextSeen.add(field.modelName);
            examples.push(...collectNestedBodyVariantExamples(model.fields, command, modelPath, nextSeen, depth + 1));
        }
    }
    return examples;
}
function variantBodySample(command, baseModelName, variant) {
    const baseModel = command.bodyModels[baseModelName];
    const variantModel = command.bodyModels[variant.modelName];
    const sample = {
        ...sampleObjectForFields(baseModel.fields, command, new Set(), 0),
        ...sampleObjectForFields(variantModel.fields, command, new Set([baseModelName]), 0)
    };
    sample[variant.discriminatorField] = variant.discriminatorValue;
    return sample;
}
function sampleObjectForFields(fields, command, seenModels, depth) {
    return Object.fromEntries(fields.map((field) => [field.name, sampleBodyValue(field, command, seenModels, depth)]));
}
function rootBodyFields(command) {
    var _a, _b;
    return (_b = (_a = rootBodyModel(command)) === null || _a === void 0 ? void 0 : _a.fields) !== null && _b !== void 0 ? _b : command.bodyFields;
}
function rootBodyModel(command) {
    return command.bodyModel ? command.bodyModels[command.bodyModel] : undefined;
}
function sampleBodyValue(field, command, seenModels, depth) {
    if (field.type === "array") {
        return [sampleArrayItem(field, command, seenModels, depth)];
    }
    if (field.modelName) {
        return sampleModelValue(field.modelName, command, seenModels, depth);
    }
    return sampleScalarValue(field.type, field.enumValues);
}
function sampleArrayItem(field, command, seenModels, depth) {
    if (field.modelName) {
        return sampleModelValue(field.modelName, command, seenModels, depth);
    }
    return sampleScalarValue(field.itemType || "string", field.enumValues);
}
function sampleModelValue(modelName, command, seenModels, depth) {
    if (depth >= MAX_BODY_EXAMPLE_DEPTH || seenModels.has(modelName)) {
        return {};
    }
    const model = command.bodyModels[modelName];
    if (!model) {
        return {};
    }
    const nextSeen = new Set(seenModels);
    nextSeen.add(modelName);
    return Object.fromEntries(model.fields.map((field) => [field.name, sampleBodyValue(field, command, nextSeen, depth + 1)]));
}
function sampleScalarValue(type, enumValues) {
    if (enumValues.length > 0) {
        return enumValues[0];
    }
    if (type === "boolean" || type === "bool") {
        return false;
    }
    if (["integer", "number", "int", "float"].includes(type)) {
        return 0;
    }
    if (type === "array") {
        return [];
    }
    if (type === "object") {
        return {};
    }
    return "<string>";
}
function bodyRequiredFields(command) {
    if (Object.keys(command.bodyModels).length === 0) {
        return command.bodyRequiredFields;
    }
    return collectRequiredBodyPaths(rootBodyFields(command), command, "", true, new Set(), 0);
}
function collectRequiredBodyPaths(fields, command, parentPath, parentRequired, seenModels, depth) {
    if (depth >= MAX_BODY_EXAMPLE_DEPTH) {
        return [];
    }
    const requiredPaths = [];
    for (const field of fields) {
        const path = parentPath ? `${parentPath}.${field.name}` : field.name;
        const required = parentRequired && field.required;
        if (required) {
            requiredPaths.push(path);
        }
        const nestedPath = field.type === "array" ? `${path}[]` : path;
        if (field.modelName && !seenModels.has(field.modelName) && command.bodyModels[field.modelName]) {
            const nextSeen = new Set(seenModels);
            nextSeen.add(field.modelName);
            requiredPaths.push(...collectRequiredBodyPaths(command.bodyModels[field.modelName].fields, command, nestedPath, required, nextSeen, depth + 1));
        }
    }
    return requiredPaths;
}
function bodyEnumFields(command) {
    if (Object.keys(command.bodyModels).length === 0) {
        return command.bodyEnumFields;
    }
    return collectBodyEnumPaths(rootBodyFields(command), command, "", new Set(), 0);
}
function collectBodyEnumPaths(fields, command, parentPath, seenModels, depth) {
    if (depth >= MAX_BODY_EXAMPLE_DEPTH) {
        return {};
    }
    const enumPaths = {};
    for (const field of fields) {
        const path = parentPath ? `${parentPath}.${field.name}` : field.name;
        if (field.enumValues.length > 0) {
            enumPaths[path] = field.enumValues;
        }
        const nestedPath = field.type === "array" ? `${path}[]` : path;
        if (field.modelName && !seenModels.has(field.modelName) && command.bodyModels[field.modelName]) {
            const nextSeen = new Set(seenModels);
            nextSeen.add(field.modelName);
            Object.assign(enumPaths, collectBodyEnumPaths(command.bodyModels[field.modelName].fields, command, nestedPath, nextSeen, depth + 1));
        }
    }
    return enumPaths;
}
