// aidp-cli/dist/json.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseJsonValue = parseJsonValue;
exports.parseJsonInput = parseJsonInput;
exports.parseJsonObject = parseJsonObject;
exports.readJsonArgument = readJsonArgument;
exports.readRawBodyArgument = readRawBodyArgument;
exports.readJsonFileOrStdin = readJsonFileOrStdin;
exports.isRecord = isRecord;
const fs_1 = require("fs");
const errors_1 = require("./errors");
function parseJsonValue(value) {
    const trimmed = value.trim();
    if (trimmed === "") {
        return "";
    }
    try {
        return JSON.parse(trimmed);
    }
    catch {
        return value;
    }
}
function parseJsonInput(raw, source) {
    let value;
    try {
        value = JSON.parse(raw);
    }
    catch (error) {
        const detail = error instanceof Error ? error.message : String(error);
        throw new errors_1.CliError(`${source} is not valid JSON: ${detail}`);
    }
    if (!isRecord(value) && !Array.isArray(value)) {
        throw new errors_1.CliError(`${source} must contain a JSON object or array.`);
    }
    return value;
}
function parseJsonObject(raw, source) {
    const value = parseJsonInput(raw, source);
    if (!isRecord(value)) {
        throw new errors_1.CliError(`${source} must contain a JSON object.`);
    }
    return value;
}
function readJsonArgument(value) {
    if (value === "-") {
        return (0, fs_1.readFileSync)(0, "utf8");
    }
    if (value.startsWith("@")) {
        return (0, fs_1.readFileSync)(value.slice(1), "utf8");
    }
    if (value.startsWith("file://")) {
        return (0, fs_1.readFileSync)(value.slice("file://".length), "utf8");
    }
    return value;
}
function readRawBodyArgument(value) {
    if (value === "-") {
        return (0, fs_1.readFileSync)(0);
    }
    if (value.startsWith("@")) {
        return (0, fs_1.readFileSync)(value.slice(1));
    }
    if (value.startsWith("file://")) {
        return (0, fs_1.readFileSync)(value.slice("file://".length));
    }
    return value;
}
function readJsonFileOrStdin(path) {
    return path === "-" ? (0, fs_1.readFileSync)(0, "utf8") : (0, fs_1.readFileSync)(path, "utf8");
}
function isRecord(value) {
    return typeof value === "object" && value !== null && !Array.isArray(value);
}
