// aidp-cli/dist/manifest.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadCommandManifest = loadCommandManifest;
const fs_1 = require("fs");
const path_1 = require("path");
const errors_1 = require("./errors");
function loadCommandManifest() {
    var _a, _b, _c;
    const manifestPath = (0, path_1.join)(__dirname, "operation_manifest.json");
    if (!(0, fs_1.existsSync)(manifestPath)) {
        throw new errors_1.CliError("AIDP command manifest is missing. Regenerate the CLI package before running aidp.");
    }
    const raw = JSON.parse((0, fs_1.readFileSync)(manifestPath, "utf8"));
    if (!Array.isArray(raw.commandGroups)) {
        throw new errors_1.CliError("AIDP command manifest is invalid. Regenerate the CLI package before running aidp.");
    }
    return {
        commandGroups: raw.commandGroups.map(commandGroupFromRaw),
        sourceSpec: String((_a = raw.sourceSpec) !== null && _a !== void 0 ? _a : ""),
        sourceSpecSha256: String((_b = raw.sourceSpecSha256) !== null && _b !== void 0 ? _b : ""),
        version: Number((_c = raw.version) !== null && _c !== void 0 ? _c : 0)
    };
}
function commandGroupFromRaw(raw) {
    var _a, _b, _c;
    return {
        clientClassName: requiredString(raw.clientClassName, "clientClassName"),
        commands: ((_a = raw.commands) !== null && _a !== void 0 ? _a : []).map(commandFromRaw),
        description: String((_b = raw.description) !== null && _b !== void 0 ? _b : ""),
        name: requiredString(raw.name, "name"),
        tag: String((_c = raw.tag) !== null && _c !== void 0 ? _c : "")
    };
}
function commandFromRaw(raw) {
    var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p;
    return {
        aliases: ((_a = raw.aliases) !== null && _a !== void 0 ? _a : []).map(String),
        bodyEnumFields: normalizeEnumMap(raw.bodyEnumFields),
        bodyField: raw.bodyField ? String(raw.bodyField) : undefined,
        bodyFields: ((_b = raw.bodyFields) !== null && _b !== void 0 ? _b : []).map(bodyFieldFromRaw),
        bodyModel: String((_c = raw.bodyModel) !== null && _c !== void 0 ? _c : ""),
        bodyModels: bodyModelsFromRaw((_d = raw.bodyModels) !== null && _d !== void 0 ? _d : {}),
        bodyRequiredFields: ((_e = raw.bodyRequiredFields) !== null && _e !== void 0 ? _e : []).map(String),
        deprecated: Boolean(raw.deprecated),
        description: String((_f = raw.description) !== null && _f !== void 0 ? _f : ""),
        fields: ((_g = raw.fields) !== null && _g !== void 0 ? _g : []).map(fieldFromRaw),
        httpMethod: String((_h = raw.httpMethod) !== null && _h !== void 0 ? _h : ""),
        name: requiredString(raw.name, "name"),
        operationId: String((_j = raw.operationId) !== null && _j !== void 0 ? _j : ""),
        path: String((_k = raw.path) !== null && _k !== void 0 ? _k : ""),
        responseBodyKey: String((_l = raw.responseBodyKey) !== null && _l !== void 0 ? _l : ""),
        responseHeaderFields: ((_m = raw.responseHeaderFields) !== null && _m !== void 0 ? _m : []).map(String),
        sdkMethodName: requiredString(raw.sdkMethodName, "sdkMethodName"),
        section: String((_o = raw.section) !== null && _o !== void 0 ? _o : "Available Commands"),
        summary: String((_p = raw.summary) !== null && _p !== void 0 ? _p : "")
    };
}
function fieldFromRaw(raw) {
    var _a, _b, _c, _d, _e, _f, _g;
    const name = requiredString(raw.name, "field.name");
    return {
        cliName: String((_a = raw.cliName) !== null && _a !== void 0 ? _a : name),
        description: String((_b = raw.description) !== null && _b !== void 0 ? _b : ""),
        enumValues: (_c = raw.enumValues) !== null && _c !== void 0 ? _c : [],
        in: String((_d = raw.in) !== null && _d !== void 0 ? _d : ""),
        modelName: String((_e = raw.modelName) !== null && _e !== void 0 ? _e : ""),
        name,
        originalName: String((_f = raw.originalName) !== null && _f !== void 0 ? _f : name),
        required: Boolean(raw.required),
        type: String((_g = raw.type) !== null && _g !== void 0 ? _g : "")
    };
}
function bodyFieldFromRaw(raw) {
    var _a, _b, _c, _d;
    return {
        enumValues: (_a = raw.enumValues) !== null && _a !== void 0 ? _a : [],
        itemType: String((_b = raw.itemType) !== null && _b !== void 0 ? _b : ""),
        modelName: String((_c = raw.modelName) !== null && _c !== void 0 ? _c : ""),
        name: requiredString(raw.name, "bodyField.name"),
        required: Boolean(raw.required),
        type: String((_d = raw.type) !== null && _d !== void 0 ? _d : "")
    };
}
function bodyModelsFromRaw(raw) {
    var _a, _b, _c;
    const models = {};
    for (const [name, model] of Object.entries(raw)) {
        models[name] = {
            enumFields: normalizeEnumMap(model.enumFields),
            fields: ((_a = model.fields) !== null && _a !== void 0 ? _a : []).map(bodyFieldFromRaw),
            requiredFields: ((_b = model.requiredFields) !== null && _b !== void 0 ? _b : []).map(String),
            variants: ((_c = model.variants) !== null && _c !== void 0 ? _c : []).map(bodyVariantFromRaw)
        };
    }
    return models;
}
function bodyVariantFromRaw(raw) {
    return {
        discriminatorField: requiredString(raw.discriminatorField, "bodyVariant.discriminatorField"),
        discriminatorValue: raw.discriminatorValue,
        modelName: requiredString(raw.modelName, "bodyVariant.modelName")
    };
}
function normalizeEnumMap(value) {
    const result = {};
    for (const [key, items] of Object.entries(value !== null && value !== void 0 ? value : {})) {
        result[key] = Array.isArray(items) ? items : [];
    }
    return result;
}
function requiredString(value, field) {
    if (typeof value !== "string" || value.length === 0) {
        throw new errors_1.CliError(`AIDP command manifest is invalid: missing ${field}.`);
    }
    return value;
}
