// aidp-cli/dist/names.js
"use strict";
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.wordsFromMixed = wordsFromMixed;
exports.serviceNameFromClientClass = serviceNameFromClientClass;
exports.camelToKebab = camelToKebab;
exports.camelToSnake = camelToSnake;
exports.cliNameToLowerCamel = cliNameToLowerCamel;
exports.lowerCamelToCliName = lowerCamelToCliName;
exports.normalizedLookupName = normalizedLookupName;
const WORD_PATTERN = /[A-Z]+(?=[A-Z][a-z]|\d|$)|[A-Z]?[a-z]+|\d+/g;
const COMMAND_GROUP_NAME_OVERRIDES = new Map([
    ["ml-ops", "mlops"]
]);
function wordsFromMixed(value) {
    var _a;
    const normalized = String(value || "").replace(/[^0-9A-Za-z]+/g, " ");
    const words = [];
    for (const token of normalized.split(/\s+/).filter(Boolean)) {
        words.push(...((_a = token.match(WORD_PATTERN)) !== null && _a !== void 0 ? _a : [token]));
    }
    return words;
}
function serviceNameFromClientClass(className) {
    var _a;
    const defaultName = wordsFromMixed(className.replace(/Client$/, ""))
        .map((word) => word.toLowerCase())
        .join("-");
    return (_a = COMMAND_GROUP_NAME_OVERRIDES.get(defaultName)) !== null && _a !== void 0 ? _a : defaultName;
}
function camelToKebab(value) {
    return wordsFromMixed(value)
        .map((word) => word.toLowerCase())
        .join("-");
}
function camelToSnake(value) {
    return wordsFromMixed(value)
        .map((word) => word.toLowerCase())
        .join("_");
}
function cliNameToLowerCamel(value) {
    const parts = value
        .trim()
        .split(/[-_]+/)
        .filter(Boolean)
        .map((part) => part.toLowerCase());
    if (parts.length === 0) {
        return "";
    }
    return parts
        .map((part, index) => (index === 0 ? part : part.charAt(0).toUpperCase() + part.slice(1)))
        .join("");
}
function lowerCamelToCliName(value) {
    return camelToKebab(value);
}
function normalizedLookupName(value) {
    return value.replace(/[-_]/g, "").toLowerCase();
}
