// aidp-cli/dist/output.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.printResponse = printResponse;
exports.printErrorResponse = printErrorResponse;
exports.stringify = stringify;
function printResponse(response, captured) {
    const payload = captured !== null && captured !== void 0 ? captured : responseEnvelope(response);
    console.log("Response:");
    console.log(stringify(payload));
}
function printErrorResponse(payload) {
    console.error("Response:");
    console.error(stringify(payload));
}
function stringify(value) {
    return JSON.stringify(value, (_key, item) => {
        if (typeof item === "bigint") {
            return item.toString();
        }
        if (item instanceof Uint8Array) {
            return Buffer.from(item).toString("base64");
        }
        return item;
    }, 2);
}
function responseEnvelope(response) {
    if (!isRecord(response)) {
        return { data: response, headers: {}, status: null };
    }
    const headers = {};
    const data = {};
    for (const [key, value] of Object.entries(response)) {
        if (isHeaderLikeKey(key)) {
            if (value !== undefined && value !== null) {
                headers[key] = String(value);
            }
        }
        else {
            data[key] = value;
        }
    }
    const dataEntries = Object.entries(data);
    return {
        data: dataEntries.length === 1 ? dataEntries[0][1] : data,
        headers,
        status: null
    };
}
function isHeaderLikeKey(key) {
    const normalized = key.toLowerCase();
    return normalized === "etag" || normalized === "lastmodified" || normalized.startsWith("opc");
}
function isRecord(value) {
    return typeof value === "object" && value !== null && !Array.isArray(value);
}
